// @vercel/remix/dist/index.js
/**
 * @vercel/remix v2.16.7
 *
 * Copyright (c) Vercel, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.md file in the root directory of this source tree.
 *
 * @license MIT
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var node = require('@remix-run/node');
var implementations = require('./implementations.js');
var entry_server = require('./entry.server.js');
var serverRuntime = require('@remix-run/server-runtime');



Object.defineProperty(exports, 'createCookie', {
  enumerable: true,
  get: function () { return node.createCookie; }
});
Object.defineProperty(exports, 'createCookieSessionStorage', {
  enumerable: true,
  get: function () { return node.createCookieSessionStorage; }
});
Object.defineProperty(exports, 'createSessionStorage', {
  enumerable: true,
  get: function () { return node.createSessionStorage; }
});
exports.createKvSessionStorage = implementations.createKvSessionStorage;
exports.handleRequest = entry_server.handleRequest;
Object.defineProperty(exports, 'MaxPartSizeExceededError', {
  enumerable: true,
  get: function () { return serverRuntime.MaxPartSizeExceededError; }
});
Object.defineProperty(exports, 'broadcastDevReady', {
  enumerable: true,
  get: function () { return serverRuntime.broadcastDevReady; }
});
Object.defineProperty(exports, 'createRequestHandler', {
  enumerable: true,
  get: function () { return serverRuntime.createRequestHandler; }
});
Object.defineProperty(exports, 'createSession', {
  enumerable: true,
  get: function () { return serverRuntime.createSession; }
});
Object.defineProperty(exports, 'data', {
  enumerable: true,
  get: function () { return serverRuntime.data; }
});
Object.defineProperty(exports, 'defer', {
  enumerable: true,
  get: function () { return serverRuntime.defer; }
});
Object.defineProperty(exports, 'isCookie', {
  enumerable: true,
  get: function () { return serverRuntime.isCookie; }
});
Object.defineProperty(exports, 'isSession', {
  enumerable: true,
  get: function () { return serverRuntime.isSession; }
});
Object.defineProperty(exports, 'json', {
  enumerable: true,
  get: function () { return serverRuntime.json; }
});
Object.defineProperty(exports, 'logDevReady', {
  enumerable: true,
  get: function () { return serverRuntime.logDevReady; }
});
Object.defineProperty(exports, 'redirect', {
  enumerable: true,
  get: function () { return serverRuntime.redirect; }
});
Object.defineProperty(exports, 'redirectDocument', {
  enumerable: true,
  get: function () { return serverRuntime.redirectDocument; }
});
Object.defineProperty(exports, 'replace', {
  enumerable: true,
  get: function () { return serverRuntime.replace; }
});
Object.defineProperty(exports, 'unstable_composeUploadHandlers', {
  enumerable: true,
  get: function () { return serverRuntime.unstable_composeUploadHandlers; }
});
Object.defineProperty(exports, 'unstable_createMemoryUploadHandler', {
  enumerable: true,
  get: function () { return serverRuntime.unstable_createMemoryUploadHandler; }
});
Object.defineProperty(exports, 'unstable_parseMultipartFormData', {
  enumerable: true,
  get: function () { return serverRuntime.unstable_parseMultipartFormData; }
});
