// @vercel/remix/dist/server.js
/**
 * @vercel/remix v2.16.7
 *
 * Copyright (c) Vercel, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.md file in the root directory of this source tree.
 *
 * @license MIT
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('./globals.js');
var node = require('@remix-run/node');

/**
 * A function that returns the value to use as `context` in route `loader` and
 * `action` functions.
 *
 * You can think of this as an escape hatch that allows you to pass
 * environment/platform-specific values through to your loader/action.
 */

/**
 * Returns a request handler for Vercel's Node.js runtime that serves the
 * response using Remix.
 */
function createRequestHandler({
  build,
  getLoadContext,
  mode = process.env.NODE_ENV
}) {
  let handleRequest = node.createRequestHandler(build, mode);
  return async (req, res) => {
    let request = createRemixRequest(req, res);
    let loadContext = getLoadContext === null || getLoadContext === void 0 ? void 0 : getLoadContext(request);
    let response = await handleRequest(request, loadContext);
    await sendRemixResponse(res, response);
  };
}
function createRemixHeaders(requestHeaders) {
  let headers = new Headers();
  for (let key in requestHeaders) {
    let header = requestHeaders[key];
    // set-cookie is an array (maybe others)
    if (Array.isArray(header)) {
      for (let value of header) {
        headers.append(key, value);
      }
    } else {
      headers.append(key, header);
    }
  }
  return headers;
}
function createRemixRequest(req, res) {
  let host = req.headers["x-forwarded-host"] || req.headers["host"];
  let protocol = req.headers["x-forwarded-proto"] || "https";
  let url = new URL(`${protocol}://${host}${req.url}`);

  // Abort action/loaders once we can no longer write a response
  let controller = new AbortController();
  res.on("close", () => controller.abort());
  let init = {
    method: req.method,
    headers: createRemixHeaders(req.headers),
    // Cast until reason/throwIfAborted added
    // https://github.com/mysticatea/abort-controller/issues/36
    signal: controller.signal
  };
  if (req.method !== "GET" && req.method !== "HEAD") {
    // @ts-expect-error
    init.body = req;
  }
  return new Request(url.href, init);
}
async function sendRemixResponse(res, nodeResponse) {
  res.statusMessage = nodeResponse.statusText;
  // @ts-expect-error
  let multiValueHeaders = nodeResponse.headers.raw();
  res.writeHead(nodeResponse.status, nodeResponse.statusText, multiValueHeaders);
  if (nodeResponse.body) {
    await node.writeReadableStreamToWritable(nodeResponse.body, res);
  } else {
    res.end();
  }
}

exports.createRequestHandler = createRequestHandler;
