// @shopify/cli/dist/meriyah-7MIMHTWQ.js
import {
  __esm,
  init_cjs_shims
} from "./chunk-PKR7KJ6P.js";

// ../../node_modules/.pnpm/prettier@3.7.4/node_modules/prettier/plugins/meriyah.mjs
function Se(e3) {
  return e3 <= 127 ? _2[e3] > 0 : Ct(e3);
}
function De(e3) {
  return e3 <= 127 ? Et[e3] > 0 : J2(e3) || e3 === 8204 || e3 === 8205;
}
function m(e3) {
  return e3.column++, e3.currentChar = e3.source.charCodeAt(++e3.index);
}
function Ke(e3) {
  let t = e3.currentChar;
  if ((t & 64512) !== 55296) return 0;
  let n = e3.source.charCodeAt(e3.index + 1);
  return (n & 64512) !== 56320 ? 0 : 65536 + ((t & 1023) << 10) + (n & 1023);
}
function $e(e3, t) {
  e3.currentChar = e3.source.charCodeAt(++e3.index), e3.flags |= 1, (t & 4) === 0 && (e3.column = 0, e3.line++);
}
function te(e3) {
  e3.flags |= 1, e3.currentChar = e3.source.charCodeAt(++e3.index), e3.column = 0, e3.line++;
}
function X2(e3) {
  return e3 === 160 || e3 === 65279 || e3 === 133 || e3 === 5760 || e3 >= 8192 && e3 <= 8203 || e3 === 8239 || e3 === 8287 || e3 === 12288 || e3 === 8201 || e3 === 65519;
}
function v(e3) {
  return e3 < 65 ? e3 - 48 : e3 - 65 + 10 & 15;
}
function j2(e3) {
  switch (e3) {
    case 134283266:
      return "NumericLiteral";
    case 134283267:
      return "StringLiteral";
    case 86021:
    case 86022:
      return "BooleanLiteral";
    case 86023:
      return "NullLiteral";
    case 65540:
      return "RegularExpression";
    case 67174408:
    case 67174409:
    case 131:
      return "TemplateLiteral";
    default:
      return (e3 & 143360) === 143360 ? "Identifier" : (e3 & 4096) === 4096 ? "Keyword" : "Punctuator";
  }
}
function H2(e3) {
  let { source: t } = e3;
  e3.currentChar === 35 && t.charCodeAt(e3.index + 1) === 33 && (m(e3), m(e3), We(e3, t, 0, 4, e3.tokenStart));
}
function rt(e3, t, n, u, o, i) {
  return u & 2 && e3.report(0), We(e3, t, n, o, i);
}
function We(e3, t, n, u, o) {
  let { index: i } = e3;
  for (e3.tokenIndex = e3.index, e3.tokenLine = e3.line, e3.tokenColumn = e3.column; e3.index < e3.end; ) {
    if (S[e3.currentChar] & 8) {
      let l = e3.currentChar === 13;
      te(e3), l && e3.index < e3.end && e3.currentChar === 10 && (e3.currentChar = t.charCodeAt(++e3.index));
      break;
    } else if ((e3.currentChar ^ 8232) <= 1) {
      te(e3);
      break;
    }
    m(e3), e3.tokenIndex = e3.index, e3.tokenLine = e3.line, e3.tokenColumn = e3.column;
  }
  if (e3.options.onComment) {
    let l = { start: { line: o.line, column: o.column }, end: { line: e3.tokenLine, column: e3.tokenColumn } };
    e3.options.onComment(wt[u & 255], t.slice(i, e3.tokenIndex), o.index, e3.tokenIndex, l);
  }
  return n | 1;
}
function z2(e3, t, n) {
  let { index: u } = e3;
  for (; e3.index < e3.end; ) if (e3.currentChar < 43) {
    let o = !1;
    for (; e3.currentChar === 42; ) if (o || (n &= -5, o = !0), m(e3) === 47) {
      if (m(e3), e3.options.onComment) {
        let i = { start: { line: e3.tokenLine, column: e3.tokenColumn }, end: { line: e3.line, column: e3.column } };
        e3.options.onComment(wt[1], t.slice(u, e3.index - 2), u - 2, e3.index, i);
      }
      return e3.tokenIndex = e3.index, e3.tokenLine = e3.line, e3.tokenColumn = e3.column, n;
    }
    if (o) continue;
    S[e3.currentChar] & 8 ? e3.currentChar === 13 ? (n |= 5, te(e3)) : ($e(e3, n), n = n & -5 | 1) : m(e3);
  } else (e3.currentChar ^ 8232) <= 1 ? (n = n & -5 | 1, te(e3)) : (n &= -5, m(e3));
  e3.report(18);
}
function St(e3, t) {
  return Object.prototype.hasOwnProperty.call(e3, t) ? e3[t] : void 0;
}
function ht(e3, t, n) {
  for (; Et[m(e3)]; ) ;
  return e3.tokenValue = e3.source.slice(e3.tokenIndex, e3.index), e3.currentChar !== 92 && e3.currentChar <= 126 ? St(Bt, e3.tokenValue) ?? 208897 : Ye(e3, t, 0, n);
}
function $2(e3, t) {
  let n = Ft(e3);
  return Se(n) || e3.report(5), e3.tokenValue = String.fromCodePoint(n), Ye(e3, t, 1, S[n] & 4);
}
function Ye(e3, t, n, u) {
  let o = e3.index;
  for (; e3.index < e3.end; ) if (e3.currentChar === 92) {
    e3.tokenValue += e3.source.slice(o, e3.index), n = 1;
    let l = Ft(e3);
    De(l) || e3.report(5), u = u && S[l] & 4, e3.tokenValue += String.fromCodePoint(l), o = e3.index;
  } else {
    let l = Ke(e3);
    if (l > 0) De(l) || e3.report(20, String.fromCodePoint(l)), e3.currentChar = l, e3.index++, e3.column++;
    else if (!De(e3.currentChar)) break;
    m(e3);
  }
  e3.index <= e3.end && (e3.tokenValue += e3.source.slice(o, e3.index));
  let { length: i } = e3.tokenValue;
  if (u && i >= 2 && i <= 11) {
    let l = St(Bt, e3.tokenValue);
    return l === void 0 ? 208897 | (n ? -2147483648 : 0) : n ? l === 209006 ? (t & 2050) === 0 ? l | -2147483648 : -2147483528 : t & 1 ? l === 36970 || (l & 36864) === 36864 ? -2147483527 : (l & 20480) === 20480 ? t & 262144 && (t & 8) === 0 ? l | -2147483648 : -2147483528 : -2147274630 : t & 262144 && (t & 8) === 0 && (l & 20480) === 20480 ? l | -2147483648 : l === 241771 ? t & 262144 ? -2147274630 : t & 1024 ? -2147483528 : l | -2147483648 : l === 209005 ? -2147274630 : (l & 36864) === 36864 ? l | 12288 | -2147483648 : -2147483528 : l;
  }
  return 208897 | (n ? -2147483648 : 0);
}
function W2(e3) {
  let t = m(e3);
  if (t === 92) return 130;
  let n = Ke(e3);
  return n && (t = n), Se(t) || e3.report(96), 130;
}
function Ft(e3) {
  return e3.source.charCodeAt(e3.index + 1) !== 117 && e3.report(5), e3.currentChar = e3.source.charCodeAt(e3.index += 2), e3.column += 2, Y2(e3);
}
function Y2(e3) {
  let t = 0, n = e3.currentChar;
  if (n === 123) {
    let l = e3.index - 2;
    for (; S[m(e3)] & 64; ) if (t = t << 4 | v(e3.currentChar), t > 1114111) throw new q({ index: l, line: e3.line, column: e3.column }, e3.currentLocation, 104);
    if (e3.currentChar !== 125) throw new q({ index: l, line: e3.line, column: e3.column }, e3.currentLocation, 7);
    return m(e3), t;
  }
  (S[n] & 64) === 0 && e3.report(7);
  let u = e3.source.charCodeAt(e3.index + 1);
  (S[u] & 64) === 0 && e3.report(7);
  let o = e3.source.charCodeAt(e3.index + 2);
  (S[o] & 64) === 0 && e3.report(7);
  let i = e3.source.charCodeAt(e3.index + 3);
  return (S[i] & 64) === 0 && e3.report(7), t = v(n) << 12 | v(u) << 8 | v(o) << 4 | v(i), e3.currentChar = e3.source.charCodeAt(e3.index += 4), e3.column += 4, t;
}
function kt(e3, t, n) {
  let u = e3.currentChar, o = 0, i = 9, l = n & 64 ? 0 : 1, f = 0, c = 0;
  if (n & 64) o = "." + Te(e3, u), u = e3.currentChar, u === 110 && e3.report(12);
  else {
    if (u === 48) if (u = m(e3), (u | 32) === 120) {
      for (n = 136, u = m(e3); S[u] & 4160; ) {
        if (u === 95) {
          c || e3.report(152), c = 0, u = m(e3);
          continue;
        }
        c = 1, o = o * 16 + v(u), f++, u = m(e3);
      }
      (f === 0 || !c) && e3.report(f === 0 ? 21 : 153);
    } else if ((u | 32) === 111) {
      for (n = 132, u = m(e3); S[u] & 4128; ) {
        if (u === 95) {
          c || e3.report(152), c = 0, u = m(e3);
          continue;
        }
        c = 1, o = o * 8 + (u - 48), f++, u = m(e3);
      }
      (f === 0 || !c) && e3.report(f === 0 ? 0 : 153);
    } else if ((u | 32) === 98) {
      for (n = 130, u = m(e3); S[u] & 4224; ) {
        if (u === 95) {
          c || e3.report(152), c = 0, u = m(e3);
          continue;
        }
        c = 1, o = o * 2 + (u - 48), f++, u = m(e3);
      }
      (f === 0 || !c) && e3.report(f === 0 ? 0 : 153);
    } else if (S[u] & 32) for (t & 1 && e3.report(1), n = 1; S[u] & 16; ) {
      if (S[u] & 512) {
        n = 32, l = 0;
        break;
      }
      o = o * 8 + (u - 48), u = m(e3);
    }
    else S[u] & 512 ? (t & 1 && e3.report(1), e3.flags |= 64, n = 32) : u === 95 && e3.report(0);
    if (n & 48) {
      if (l) {
        for (; i >= 0 && S[u] & 4112; ) {
          if (u === 95) {
            if (u = m(e3), u === 95 || n & 32) throw new q(e3.currentLocation, { index: e3.index + 1, line: e3.line, column: e3.column }, 152);
            c = 1;
            continue;
          }
          c = 0, o = 10 * o + (u - 48), u = m(e3), --i;
        }
        if (c) throw new q(e3.currentLocation, { index: e3.index + 1, line: e3.line, column: e3.column }, 153);
        if (i >= 0 && !Se(u) && u !== 46) return e3.tokenValue = o, e3.options.raw && (e3.tokenRaw = e3.source.slice(e3.tokenIndex, e3.index)), 134283266;
      }
      o += Te(e3, u), u = e3.currentChar, u === 46 && (m(e3) === 95 && e3.report(0), n = 64, o += "." + Te(e3, e3.currentChar), u = e3.currentChar);
    }
  }
  let g = e3.index, d = 0;
  if (u === 110 && n & 128) d = 1, u = m(e3);
  else if ((u | 32) === 101) {
    u = m(e3), S[u] & 256 && (u = m(e3));
    let { index: a } = e3;
    (S[u] & 16) === 0 && e3.report(11), o += e3.source.substring(g, a) + Te(e3, u), u = e3.currentChar;
  }
  return (e3.index < e3.end && S[u] & 16 || Se(u)) && e3.report(13), d ? (e3.tokenRaw = e3.source.slice(e3.tokenIndex, e3.index), e3.tokenValue = BigInt(p(0, e3.tokenRaw.slice(0, -1), "_", "")), 134283388) : (e3.tokenValue = n & 15 ? o : n & 32 ? parseFloat(e3.source.substring(e3.tokenIndex, e3.index)) : +o, e3.options.raw && (e3.tokenRaw = e3.source.slice(e3.tokenIndex, e3.index)), 134283266);
}
function Te(e3, t) {
  let n = 0, u = e3.index, o = "";
  for (; S[t] & 4112; ) {
    if (t === 95) {
      let { index: i } = e3;
      if (t = m(e3), t === 95) throw new q(e3.currentLocation, { index: e3.index + 1, line: e3.line, column: e3.column }, 152);
      n = 1, o += e3.source.substring(u, i), u = e3.index;
      continue;
    }
    n = 0, t = m(e3);
  }
  if (n) throw new q(e3.currentLocation, { index: e3.index + 1, line: e3.line, column: e3.column }, 153);
  return o + e3.source.substring(u, e3.index);
}
function Q2(e3) {
  let t = e3.index, n = Z.Empty;
  e: for (; ; ) {
    let g = e3.currentChar;
    if (m(e3), n & Z.Escape) n &= ~Z.Escape;
    else switch (g) {
      case 47:
        if (n) break;
        break e;
      case 92:
        n |= Z.Escape;
        break;
      case 91:
        n |= Z.Class;
        break;
      case 93:
        n &= Z.Escape;
        break;
    }
    if ((g === 13 || g === 10 || g === 8232 || g === 8233) && e3.report(34), e3.index >= e3.source.length) return e3.report(34);
  }
  let u = e3.index - 1, o = P.Empty, i = e3.currentChar, { index: l } = e3;
  for (; De(i); ) {
    switch (i) {
      case 103:
        o & P.Global && e3.report(36, "g"), o |= P.Global;
        break;
      case 105:
        o & P.IgnoreCase && e3.report(36, "i"), o |= P.IgnoreCase;
        break;
      case 109:
        o & P.Multiline && e3.report(36, "m"), o |= P.Multiline;
        break;
      case 117:
        o & P.Unicode && e3.report(36, "u"), o & P.UnicodeSets && e3.report(36, "vu"), o |= P.Unicode;
        break;
      case 118:
        o & P.Unicode && e3.report(36, "uv"), o & P.UnicodeSets && e3.report(36, "v"), o |= P.UnicodeSets;
        break;
      case 121:
        o & P.Sticky && e3.report(36, "y"), o |= P.Sticky;
        break;
      case 115:
        o & P.DotAll && e3.report(36, "s"), o |= P.DotAll;
        break;
      case 100:
        o & P.Indices && e3.report(36, "d"), o |= P.Indices;
        break;
      default:
        e3.report(35);
    }
    i = m(e3);
  }
  let f = e3.source.slice(l, e3.index), c = e3.source.slice(t, u);
  return e3.tokenRegExp = { pattern: c, flags: f }, e3.options.raw && (e3.tokenRaw = e3.source.slice(e3.tokenIndex, e3.index)), e3.tokenValue = Z2(e3, c, f), 65540;
}
function Z2(e3, t, n) {
  try {
    return new RegExp(t, n);
  } catch {
    if (!e3.options.validateRegex) return null;
    e3.report(34);
  }
}
function G2(e3, t, n) {
  let { index: u } = e3, o = "", i = m(e3), l = e3.index;
  for (; (S[i] & 8) === 0; ) {
    if (i === n) return o += e3.source.slice(l, e3.index), m(e3), e3.options.raw && (e3.tokenRaw = e3.source.slice(u, e3.index)), e3.tokenValue = o, 134283267;
    if ((i & 8) === 8 && i === 92) {
      if (o += e3.source.slice(l, e3.index), i = m(e3), i < 127 || i === 8232 || i === 8233) {
        let f = Nt(e3, t, i);
        f >= 0 ? o += String.fromCodePoint(f) : Lt(e3, f, 0);
      } else o += String.fromCodePoint(i);
      l = e3.index + 1;
    } else (i === 8232 || i === 8233) && (e3.column = -1, e3.line++);
    e3.index >= e3.end && e3.report(16), i = m(e3);
  }
  e3.report(16);
}
function Nt(e3, t, n, u = 0) {
  switch (n) {
    case 98:
      return 8;
    case 102:
      return 12;
    case 114:
      return 13;
    case 110:
      return 10;
    case 116:
      return 9;
    case 118:
      return 11;
    case 13:
      if (e3.index < e3.end) {
        let o = e3.source.charCodeAt(e3.index + 1);
        o === 10 && (e3.index = e3.index + 1, e3.currentChar = o);
      }
    case 10:
    case 8232:
    case 8233:
      return e3.column = -1, e3.line++, -1;
    case 48:
    case 49:
    case 50:
    case 51: {
      let o = n - 48, i = e3.index + 1, l = e3.column + 1;
      if (i < e3.end) {
        let f = e3.source.charCodeAt(i);
        if ((S[f] & 32) === 0) {
          if (o !== 0 || S[f] & 512) {
            if (t & 1 || u) return -2;
            e3.flags |= 64;
          }
        } else {
          if (t & 1 || u) return -2;
          if (e3.currentChar = f, o = o << 3 | f - 48, i++, l++, i < e3.end) {
            let c = e3.source.charCodeAt(i);
            S[c] & 32 && (e3.currentChar = c, o = o << 3 | c - 48, i++, l++);
          }
          e3.flags |= 64;
        }
        e3.index = i - 1, e3.column = l - 1;
      }
      return o;
    }
    case 52:
    case 53:
    case 54:
    case 55: {
      if (u || t & 1) return -2;
      let o = n - 48, i = e3.index + 1, l = e3.column + 1;
      if (i < e3.end) {
        let f = e3.source.charCodeAt(i);
        S[f] & 32 && (o = o << 3 | f - 48, e3.currentChar = f, e3.index = i, e3.column = l);
      }
      return e3.flags |= 64, o;
    }
    case 120: {
      let o = m(e3);
      if ((S[o] & 64) === 0) return -4;
      let i = v(o), l = m(e3);
      if ((S[l] & 64) === 0) return -4;
      let f = v(l);
      return i << 4 | f;
    }
    case 117: {
      let o = m(e3);
      if (e3.currentChar === 123) {
        let i = 0;
        for (; (S[m(e3)] & 64) !== 0; ) if (i = i << 4 | v(e3.currentChar), i > 1114111) return -5;
        return e3.currentChar < 1 || e3.currentChar !== 125 ? -4 : i;
      } else {
        if ((S[o] & 64) === 0) return -4;
        let i = e3.source.charCodeAt(e3.index + 1);
        if ((S[i] & 64) === 0) return -4;
        let l = e3.source.charCodeAt(e3.index + 2);
        if ((S[l] & 64) === 0) return -4;
        let f = e3.source.charCodeAt(e3.index + 3);
        return (S[f] & 64) === 0 ? -4 : (e3.index += 3, e3.column += 3, e3.currentChar = e3.source.charCodeAt(e3.index), v(o) << 12 | v(i) << 8 | v(l) << 4 | v(f));
      }
    }
    case 56:
    case 57:
      if (u || !e3.options.webcompat || t & 1) return -3;
      e3.flags |= 4096;
    default:
      return n;
  }
}
function Lt(e3, t, n) {
  switch (t) {
    case -1:
      return;
    case -2:
      e3.report(n ? 2 : 1);
    case -3:
      e3.report(n ? 3 : 14);
    case -4:
      e3.report(7);
    case -5:
      e3.report(104);
  }
}
function It(e3, t) {
  let { index: n } = e3, u = 67174409, o = "", i = m(e3);
  for (; i !== 96; ) {
    if (i === 36 && e3.source.charCodeAt(e3.index + 1) === 123) {
      m(e3), u = 67174408;
      break;
    } else if (i === 92) if (i = m(e3), i > 126) o += String.fromCodePoint(i);
    else {
      let { index: l, line: f, column: c } = e3, g = Nt(e3, t | 1, i, 1);
      if (g >= 0) o += String.fromCodePoint(g);
      else if (g !== -1 && t & 64) {
        e3.index = l, e3.line = f, e3.column = c, o = null, i = x2(e3, i), i < 0 && (u = 67174408);
        break;
      } else Lt(e3, g, 1);
    }
    else e3.index < e3.end && (i === 13 && e3.source.charCodeAt(e3.index) === 10 && (o += String.fromCodePoint(i), e3.currentChar = e3.source.charCodeAt(++e3.index)), ((i & 83) < 3 && i === 10 || (i ^ 8232) <= 1) && (e3.column = -1, e3.line++), o += String.fromCodePoint(i));
    e3.index >= e3.end && e3.report(17), i = m(e3);
  }
  return m(e3), e3.tokenValue = o, e3.tokenRaw = e3.source.slice(n + 1, e3.index - (u === 67174409 ? 1 : 2)), u;
}
function x2(e3, t) {
  for (; t !== 96; ) {
    switch (t) {
      case 36: {
        let n = e3.index + 1;
        if (n < e3.end && e3.source.charCodeAt(n) === 123) return e3.index = n, e3.column++, -t;
        break;
      }
      case 10:
      case 8232:
      case 8233:
        e3.column = -1, e3.line++;
    }
    e3.index >= e3.end && e3.report(17), t = m(e3);
  }
  return t;
}
function p2(e3, t) {
  return e3.index >= e3.end && e3.report(0), e3.index--, e3.column--, It(e3, t);
}
function r(e3, t) {
  e3.flags = (e3.flags | 1) ^ 1, e3.startIndex = e3.index, e3.startColumn = e3.column, e3.startLine = e3.line, e3.setToken(qt(e3, t, 0));
}
function qt(e3, t, n) {
  let u = e3.index === 0, { source: o } = e3;
  for (; e3.index < e3.end; ) {
    e3.tokenIndex = e3.index, e3.tokenColumn = e3.column, e3.tokenLine = e3.line;
    let i = e3.currentChar;
    if (i <= 126) {
      let l = en[i];
      switch (l) {
        case 67174411:
        case 16:
        case 2162700:
        case 1074790415:
        case 69271571:
        case 20:
        case 21:
        case 1074790417:
        case 18:
        case 16842799:
        case 132:
        case 128:
          return m(e3), l;
        case 208897:
          return ht(e3, t, 0);
        case 4096:
          return ht(e3, t, 1);
        case 134283266:
          return kt(e3, t, 144);
        case 134283267:
          return G2(e3, t, i);
        case 131:
          return It(e3, t);
        case 136:
          return $2(e3, t);
        case 130:
          return W2(e3);
        case 127:
          m(e3);
          break;
        case 129:
          n |= 5, te(e3);
          break;
        case 135:
          $e(e3, n), n = n & -5 | 1;
          break;
        case 8456256: {
          let f = m(e3);
          if (e3.index < e3.end) {
            if (f === 60) return e3.index < e3.end && m(e3) === 61 ? (m(e3), 4194332) : 8390978;
            if (f === 61) return m(e3), 8390718;
            if (f === 33) {
              let c = e3.index + 1;
              if (c + 1 < e3.end && o.charCodeAt(c) === 45 && o.charCodeAt(c + 1) == 45) {
                e3.column += 3, e3.currentChar = o.charCodeAt(e3.index += 3), n = rt(e3, o, n, t, 2, e3.tokenStart);
                continue;
              }
              return 8456256;
            }
          }
          return 8456256;
        }
        case 1077936155: {
          m(e3);
          let f = e3.currentChar;
          return f === 61 ? m(e3) === 61 ? (m(e3), 8390458) : 8390460 : f === 62 ? (m(e3), 10) : 1077936155;
        }
        case 16842798:
          return m(e3) !== 61 ? 16842798 : m(e3) !== 61 ? 8390461 : (m(e3), 8390459);
        case 8391477:
          return m(e3) !== 61 ? 8391477 : (m(e3), 4194340);
        case 8391476: {
          if (m(e3), e3.index >= e3.end) return 8391476;
          let f = e3.currentChar;
          return f === 61 ? (m(e3), 4194338) : f !== 42 ? 8391476 : m(e3) !== 61 ? 8391735 : (m(e3), 4194335);
        }
        case 8389959:
          return m(e3) !== 61 ? 8389959 : (m(e3), 4194341);
        case 25233968: {
          m(e3);
          let f = e3.currentChar;
          return f === 43 ? (m(e3), 33619993) : f === 61 ? (m(e3), 4194336) : 25233968;
        }
        case 25233969: {
          m(e3);
          let f = e3.currentChar;
          if (f === 45) {
            if (m(e3), (n & 1 || u) && e3.currentChar === 62) {
              e3.options.webcompat || e3.report(112), m(e3), n = rt(e3, o, n, t, 3, e3.tokenStart);
              continue;
            }
            return 33619994;
          }
          return f === 61 ? (m(e3), 4194337) : 25233969;
        }
        case 8457014: {
          if (m(e3), e3.index < e3.end) {
            let f = e3.currentChar;
            if (f === 47) {
              m(e3), n = We(e3, o, n, 0, e3.tokenStart);
              continue;
            }
            if (f === 42) {
              m(e3), n = z2(e3, o, n);
              continue;
            }
            if (t & 32) return Q2(e3);
            if (f === 61) return m(e3), 4259875;
          }
          return 8457014;
        }
        case 67108877: {
          let f = m(e3);
          if (f >= 48 && f <= 57) return kt(e3, t, 80);
          if (f === 46) {
            let c = e3.index + 1;
            if (c < e3.end && o.charCodeAt(c) === 46) return e3.column += 2, e3.currentChar = o.charCodeAt(e3.index += 2), 14;
          }
          return 67108877;
        }
        case 8389702: {
          m(e3);
          let f = e3.currentChar;
          return f === 124 ? (m(e3), e3.currentChar === 61 ? (m(e3), 4194344) : 8913465) : f === 61 ? (m(e3), 4194342) : 8389702;
        }
        case 8390721: {
          m(e3);
          let f = e3.currentChar;
          if (f === 61) return m(e3), 8390719;
          if (f !== 62) return 8390721;
          if (m(e3), e3.index < e3.end) {
            let c = e3.currentChar;
            if (c === 62) return m(e3) === 61 ? (m(e3), 4194334) : 8390980;
            if (c === 61) return m(e3), 4194333;
          }
          return 8390979;
        }
        case 8390213: {
          m(e3);
          let f = e3.currentChar;
          return f === 38 ? (m(e3), e3.currentChar === 61 ? (m(e3), 4194345) : 8913720) : f === 61 ? (m(e3), 4194343) : 8390213;
        }
        case 22: {
          let f = m(e3);
          if (f === 63) return m(e3), e3.currentChar === 61 ? (m(e3), 4194346) : 276824445;
          if (f === 46) {
            let c = e3.index + 1;
            if (c < e3.end && (f = o.charCodeAt(c), !(f >= 48 && f <= 57))) return m(e3), 67108990;
          }
          return 22;
        }
      }
    } else {
      if ((i ^ 8232) <= 1) {
        n = n & -5 | 1, te(e3);
        continue;
      }
      let l = Ke(e3);
      if (l > 0 && (i = l), Ct(i)) return e3.tokenValue = "", Ye(e3, t, 0, 0);
      if (X2(i)) {
        m(e3);
        continue;
      }
      e3.report(20, String.fromCodePoint(i));
    }
  }
  return 1048576;
}
function M(e3, t) {
  (e3.flags & 1) === 0 && (e3.getToken() & 1048576) !== 1048576 && e3.report(30, B[e3.getToken() & 255]), C(e3, t, 1074790417) || e3.options.onInsertedSemicolon?.(e3.startIndex);
}
function Pt(e3, t, n, u) {
  return t - n < 13 && u === "use strict" && ((e3.getToken() & 1048576) === 1048576 || e3.flags & 1) ? 1 : 0;
}
function Qe(e3, t, n) {
  return e3.getToken() !== n ? 0 : (r(e3, t), 1);
}
function C(e3, t, n) {
  return e3.getToken() !== n ? !1 : (r(e3, t), !0);
}
function y(e3, t, n) {
  e3.getToken() !== n && e3.report(25, B[n & 255]), r(e3, t);
}
function K(e3, t) {
  switch (t.type) {
    case "ArrayExpression": {
      t.type = "ArrayPattern";
      let { elements: n } = t;
      for (let u = 0, o = n.length; u < o; ++u) {
        let i = n[u];
        i && K(e3, i);
      }
      return;
    }
    case "ObjectExpression": {
      t.type = "ObjectPattern";
      let { properties: n } = t;
      for (let u = 0, o = n.length; u < o; ++u) K(e3, n[u]);
      return;
    }
    case "AssignmentExpression":
      t.type = "AssignmentPattern", t.operator !== "=" && e3.report(71), delete t.operator, K(e3, t.left);
      return;
    case "Property":
      K(e3, t.value);
      return;
    case "SpreadElement":
      t.type = "RestElement", K(e3, t.argument);
  }
}
function Be(e3, t, n, u, o) {
  t & 1 && ((u & 36864) === 36864 && e3.report(118), !o && (u & 537079808) === 537079808 && e3.report(119)), ((u & 20480) === 20480 || u === -2147483528) && e3.report(102), n & 24 && (u & 255) === 73 && e3.report(100), t & 2050 && u === 209006 && e3.report(110), t & 1025 && u === 241771 && e3.report(97, "yield");
}
function Ot(e3, t, n) {
  t & 1 && ((n & 36864) === 36864 && e3.report(118), (n & 537079808) === 537079808 && e3.report(119), n === -2147483527 && e3.report(95), n === -2147483528 && e3.report(95)), (n & 20480) === 20480 && e3.report(102), t & 2050 && n === 209006 && e3.report(110), t & 1025 && n === 241771 && e3.report(97, "yield");
}
function Vt(e3, t, n) {
  return n === 209006 && (t & 2050 && e3.report(110), e3.destructible |= 128), n === 241771 && t & 1024 && e3.report(97, "yield"), (n & 20480) === 20480 || (n & 36864) === 36864 || n == -2147483527;
}
function tn(e3) {
  return e3.property ? e3.property.type === "PrivateIdentifier" : !1;
}
function Rt(e3, t, n, u) {
  for (; t; ) {
    if (t["$" + n]) return u && e3.report(137), 1;
    u && t.loop && (u = 0), t = t.$;
  }
  return 0;
}
function nn(e3, t, n) {
  let u = t;
  for (; u; ) u["$" + n] && e3.report(136, n), u = u.$;
  t["$" + n] = 1;
}
function Fe(e3) {
  switch (e3.type) {
    case "JSXIdentifier":
      return e3.name;
    case "JSXNamespacedName":
      return e3.namespace + ":" + e3.name;
    case "JSXMemberExpression":
      return Fe(e3.object) + "." + Fe(e3.property);
  }
}
function ge(e3, t) {
  return e3 & 1025 ? e3 & 2 && t === 209006 || e3 & 1024 && t === 241771 ? !1 : (t & 12288) === 12288 : (t & 12288) === 12288 || (t & 36864) === 36864;
}
function Ie(e3, t, n) {
  (n & 537079808) === 537079808 && (t & 1 && e3.report(119), e3.flags |= 512), ge(t, n) || e3.report(0);
}
function un(e3, t) {
  return e3.startIndex = e3.tokenIndex = e3.index, e3.startColumn = e3.tokenColumn = e3.column, e3.startLine = e3.tokenLine = e3.line, e3.setToken(S[e3.currentChar] & 8192 ? on(e3) : qt(e3, t, 0)), e3.getToken();
}
function on(e3) {
  let t = e3.currentChar, n = m(e3), u = e3.index;
  for (; n !== t; ) e3.index >= e3.end && e3.report(16), n = m(e3);
  return n !== t && e3.report(16), e3.tokenValue = e3.source.slice(u, e3.index), m(e3), e3.options.raw && (e3.tokenRaw = e3.source.slice(e3.tokenIndex, e3.index)), 134283267;
}
function me(e3) {
  if (e3.startIndex = e3.tokenIndex = e3.index, e3.startColumn = e3.tokenColumn = e3.column, e3.startLine = e3.tokenLine = e3.line, e3.index >= e3.end) {
    e3.setToken(1048576);
    return;
  }
  if (e3.currentChar === 60) {
    m(e3), e3.setToken(8456256);
    return;
  }
  if (e3.currentChar === 123) {
    m(e3), e3.setToken(2162700);
    return;
  }
  let t = 0;
  for (; e3.index < e3.end; ) {
    let u = S[e3.source.charCodeAt(e3.index)];
    if (u & 1024 ? (t |= 5, te(e3)) : u & 2048 ? ($e(e3, t), t = t & -5 | 1) : m(e3), S[e3.currentChar] & 16384) break;
  }
  e3.tokenIndex === e3.index && e3.report(0);
  let n = e3.source.slice(e3.tokenIndex, e3.index);
  e3.options.raw && (e3.tokenRaw = n), e3.tokenValue = n, e3.setToken(137);
}
function Ue(e3) {
  if ((e3.getToken() & 143360) === 143360) {
    let { index: t } = e3, n = e3.currentChar;
    for (; S[n] & 32770; ) n = m(e3);
    e3.tokenValue += e3.source.slice(t, e3.index), e3.setToken(208897, !0);
  }
  return e3.getToken();
}
function ln(e3) {
  let t = { validateRegex: !0, ...e3 };
  return t.module && !t.sourceType && (t.sourceType = "module"), t.globalReturn && (!t.sourceType || t.sourceType === "script") && (t.sourceType = "commonjs"), t;
}
function qe(e3, t, n) {
  let u = e3.createScope().createChildScope(512);
  return u.addBlockName(t, n, 1, 0), u;
}
function fn(e3, t) {
  return function(n, u, o, i, l) {
    let f = { type: n, value: u };
    t.ranges && (f.start = o, f.end = i, f.range = [o, i]), t.loc && (f.loc = l), e3.push(f);
  };
}
function cn(e3, t) {
  return function(n, u, o, i) {
    let l = { token: n };
    t.ranges && (l.start = u, l.end = o, l.range = [u, o]), t.loc && (l.loc = i), e3.push(l);
  };
}
function sn(e3, t = {}, n = 0) {
  let u = new Xe(e3, t);
  u.options.sourceType === "module" && (n |= 3), u.options.sourceType === "commonjs" && (n |= 69632), u.options.impliedStrict && (n |= 1), H2(u);
  let o = u.createScopeIfLexical(), i = [], l = "script";
  if (n & 2) {
    if (l = "module", i = an(u, n | 8, o), o) for (let f of u.exportedBindings) o.hasVariable(f) || u.report(148, f);
  } else i = dn(u, n | 8, o);
  return u.finishNode({ type: "Program", sourceType: l, body: i }, { index: 0, line: 1, column: 0 }, u.currentLocation);
}
function dn(e3, t, n) {
  r(e3, t | 32 | 262144);
  let u = [];
  for (; e3.getToken() === 134283267; ) {
    let { index: o, tokenValue: i, tokenStart: l, tokenIndex: f } = e3, c = e3.getToken(), g = O(e3, t);
    if (Pt(e3, o, f, i)) {
      if (t |= 1, e3.flags & 64) throw new q(e3.tokenStart, e3.currentLocation, 9);
      if (e3.flags & 4096) throw new q(e3.tokenStart, e3.currentLocation, 15);
    }
    u.push(Ge(e3, t, g, c, l));
  }
  for (; e3.getToken() !== 1048576; ) u.push(re(e3, t, n, void 0, 4, {}));
  return u;
}
function an(e3, t, n) {
  r(e3, t | 32);
  let u = [];
  for (; e3.getToken() === 134283267; ) {
    let { tokenStart: o } = e3, i = e3.getToken();
    u.push(Ge(e3, t, O(e3, t), i, o));
  }
  for (; e3.getToken() !== 1048576; ) u.push(gn(e3, t, n));
  return u;
}
function gn(e3, t, n) {
  e3.getToken() === 132 && Object.assign(e3.leadingDecorators, { start: e3.tokenStart, decorators: Ve(e3, t, void 0) });
  let u;
  switch (e3.getToken()) {
    case 20564:
      u = Pn(e3, t, n);
      break;
    case 86106:
      u = In(e3, t, n);
      break;
    default:
      u = re(e3, t, n, void 0, 4, {});
  }
  return e3.leadingDecorators?.decorators.length && e3.report(170), u;
}
function re(e3, t, n, u, o, i) {
  let l = e3.tokenStart;
  switch (e3.getToken()) {
    case 86104:
      return x(e3, t, n, u, o, 1, 0, 0, l);
    case 132:
    case 86094:
      return ze(e3, t, n, u, 0);
    case 86090:
      return je(e3, t, n, u, 16, 0);
    case 241737:
      return Nn(e3, t, n, u, o);
    case 20564:
      e3.report(103, "export");
    case 86106:
      switch (r(e3, t), e3.getToken()) {
        case 67174411:
          return _t(e3, t, u, l);
        case 67108877:
          return Jt(e3, t, l);
        default:
          e3.report(103, "import");
      }
    case 209005:
      return Mt(e3, t, n, u, o, i, 1);
    default:
      return he(e3, t, n, u, o, i, 1);
  }
}
function he(e3, t, n, u, o, i, l) {
  switch (e3.getToken()) {
    case 86088:
      return vt(e3, t, n, u, 0);
    case 20572:
      return rn(e3, t, u);
    case 20569:
      return yn(e3, t, n, u, i);
    case 20567:
      return Ln(e3, t, n, u, i);
    case 20562:
      return Fn(e3, t, n, u, i);
    case 20578:
      return Tn(e3, t, n, u, i);
    case 86110:
      return An(e3, t, n, u, i);
    case 1074790417:
      return hn(e3, t);
    case 2162700:
      return ae(e3, t, n?.createChildScope(), u, i, e3.tokenStart);
    case 86112:
      return kn(e3, t, u);
    case 20555:
      return Dn(e3, t, i);
    case 20559:
      return bn(e3, t, i);
    case 20577:
      return wn(e3, t, n, u, i);
    case 20579:
      return Cn(e3, t, n, u, i);
    case 20560:
      return En(e3, t);
    case 209005:
      return Mt(e3, t, n, u, o, i, 0);
    case 20557:
      e3.report(162);
    case 20566:
      e3.report(163);
    case 86104:
      e3.report(t & 1 ? 76 : e3.options.webcompat ? 77 : 78);
    case 86094:
      e3.report(79);
    default:
      return mn(e3, t, n, u, o, i, l);
  }
}
function mn(e3, t, n, u, o, i, l) {
  let { tokenValue: f, tokenStart: c } = e3, g = e3.getToken(), d;
  switch (g) {
    case 241737:
      d = N(e3, t), t & 1 && e3.report(85), e3.getToken() === 69271571 && e3.report(84);
      break;
    default:
      d = U(e3, t, u, 2, 0, 1, 0, 1, e3.tokenStart);
  }
  return g & 143360 && e3.getToken() === 21 ? Ze(e3, t, n, u, o, i, f, d, g, l, c) : (d = F(e3, t, u, d, 0, 0, c), d = I(e3, t, u, 0, 0, c, d), e3.getToken() === 18 && (d = W(e3, t, u, 0, c, d)), ie(e3, t, d, c));
}
function ae(e3, t, n, u, o, i = e3.tokenStart, l = "BlockStatement") {
  let f = [];
  for (y(e3, t | 32, 2162700); e3.getToken() !== 1074790415; ) f.push(re(e3, t, n, u, 2, { $: o }));
  return y(e3, t | 32, 1074790415), e3.finishNode({ type: l, body: f }, i);
}
function rn(e3, t, n) {
  (t & 4096) === 0 && e3.report(92);
  let u = e3.tokenStart;
  r(e3, t | 32);
  let o = e3.flags & 1 || e3.getToken() & 1048576 ? null : V(e3, t, n, 0, 1, e3.tokenStart);
  return M(e3, t | 32), e3.finishNode({ type: "ReturnStatement", argument: o }, u);
}
function ie(e3, t, n, u) {
  return M(e3, t | 32), e3.finishNode({ type: "ExpressionStatement", expression: n }, u);
}
function Ze(e3, t, n, u, o, i, l, f, c, g, d) {
  Be(e3, t, 0, c, 1), nn(e3, i, l), r(e3, t | 32);
  let a = g && (t & 1) === 0 && e3.options.webcompat && e3.getToken() === 86104 ? x(e3, t, n?.createChildScope(), u, o, 0, 0, 0, e3.tokenStart) : he(e3, t, n, u, o, i, g);
  return e3.finishNode({ type: "LabeledStatement", label: f, body: a }, d);
}
function Mt(e3, t, n, u, o, i, l) {
  let { tokenValue: f, tokenStart: c } = e3, g = e3.getToken(), d = N(e3, t);
  if (e3.getToken() === 21) return Ze(e3, t, n, u, o, i, f, d, g, 1, c);
  let a = e3.flags & 1;
  if (!a) {
    if (e3.getToken() === 86104) return l || e3.report(123), x(e3, t, n, u, o, 1, 0, 1, c);
    if (ge(t, e3.getToken())) return d = Wt(e3, t, u, 1, c), e3.getToken() === 18 && (d = W(e3, t, u, 0, c, d)), ie(e3, t, d, c);
  }
  return e3.getToken() === 67174411 ? d = ut(e3, t, u, d, 1, 1, 0, a, c) : (e3.getToken() === 10 && (Ie(e3, t, g), (g & 36864) === 36864 && (e3.flags |= 256), d = Oe(e3, t | 2048, u, e3.tokenValue, d, 0, 1, 0, c)), e3.assignable = 1), d = F(e3, t, u, d, 0, 0, c), d = I(e3, t, u, 0, 0, c, d), e3.assignable = 1, e3.getToken() === 18 && (d = W(e3, t, u, 0, c, d)), ie(e3, t, d, c);
}
function Ge(e3, t, n, u, o) {
  let i = e3.startIndex;
  u !== 1074790417 && (e3.assignable = 2, n = F(e3, t, void 0, n, 0, 0, o), e3.getToken() !== 1074790417 && (n = I(e3, t, void 0, 0, 0, o, n), e3.getToken() === 18 && (n = W(e3, t, void 0, 0, o, n))), M(e3, t | 32));
  let l = { type: "ExpressionStatement", expression: n };
  return n.type === "Literal" && typeof n.value == "string" && (l.directive = e3.source.slice(o.index + 1, i - 1)), e3.finishNode(l, o);
}
function hn(e3, t) {
  let n = e3.tokenStart;
  return r(e3, t | 32), e3.finishNode({ type: "EmptyStatement" }, n);
}
function kn(e3, t, n) {
  let u = e3.tokenStart;
  r(e3, t | 32), e3.flags & 1 && e3.report(90);
  let o = V(e3, t, n, 0, 1, e3.tokenStart);
  return M(e3, t | 32), e3.finishNode({ type: "ThrowStatement", argument: o }, u);
}
function yn(e3, t, n, u, o) {
  let i = e3.tokenStart;
  r(e3, t), y(e3, t | 32, 67174411), e3.assignable = 1;
  let l = V(e3, t, u, 0, 1, e3.tokenStart);
  y(e3, t | 32, 16);
  let f = yt(e3, t, n, u, o), c = null;
  return e3.getToken() === 20563 && (r(e3, t | 32), c = yt(e3, t, n, u, o)), e3.finishNode({ type: "IfStatement", test: l, consequent: f, alternate: c }, i);
}
function yt(e3, t, n, u, o) {
  let { tokenStart: i } = e3;
  return t & 1 || !e3.options.webcompat || e3.getToken() !== 86104 ? he(e3, t, n, u, 0, { $: o }, 0) : x(e3, t, n?.createChildScope(), u, 0, 0, 0, 0, i);
}
function An(e3, t, n, u, o) {
  let i = e3.tokenStart;
  r(e3, t), y(e3, t | 32, 67174411);
  let l = V(e3, t, u, 0, 1, e3.tokenStart);
  y(e3, t, 16), y(e3, t, 2162700);
  let f = [], c = 0;
  for (n = n?.createChildScope(8); e3.getToken() !== 1074790415; ) {
    let { tokenStart: g } = e3, d = null, a = [];
    for (C(e3, t | 32, 20556) ? d = V(e3, t, u, 0, 1, e3.tokenStart) : (y(e3, t | 32, 20561), c && e3.report(89), c = 1), y(e3, t | 32, 21); e3.getToken() !== 20556 && e3.getToken() !== 1074790415 && e3.getToken() !== 20561; ) a.push(re(e3, t | 4, n, u, 2, { $: o }));
    f.push(e3.finishNode({ type: "SwitchCase", test: d, consequent: a }, g));
  }
  return y(e3, t | 32, 1074790415), e3.finishNode({ type: "SwitchStatement", discriminant: l, cases: f }, i);
}
function Tn(e3, t, n, u, o) {
  let i = e3.tokenStart;
  r(e3, t), y(e3, t | 32, 67174411);
  let l = V(e3, t, u, 0, 1, e3.tokenStart);
  y(e3, t | 32, 16);
  let f = de(e3, t, n, u, o);
  return e3.finishNode({ type: "WhileStatement", test: l, body: f }, i);
}
function de(e3, t, n, u, o) {
  return he(e3, (t | 131072) ^ 131072 | 128, n, u, 0, { loop: 1, $: o }, 0);
}
function bn(e3, t, n) {
  (t & 128) === 0 && e3.report(68);
  let u = e3.tokenStart;
  r(e3, t);
  let o = null;
  if ((e3.flags & 1) === 0 && e3.getToken() & 143360) {
    let { tokenValue: i } = e3;
    o = N(e3, t | 32), Rt(e3, n, i, 1) || e3.report(138, i);
  }
  return M(e3, t | 32), e3.finishNode({ type: "ContinueStatement", label: o }, u);
}
function Dn(e3, t, n) {
  let u = e3.tokenStart;
  r(e3, t | 32);
  let o = null;
  if ((e3.flags & 1) === 0 && e3.getToken() & 143360) {
    let { tokenValue: i } = e3;
    o = N(e3, t | 32), Rt(e3, n, i, 0) || e3.report(138, i);
  } else (t & 132) === 0 && e3.report(69);
  return M(e3, t | 32), e3.finishNode({ type: "BreakStatement", label: o }, u);
}
function Cn(e3, t, n, u, o) {
  let i = e3.tokenStart;
  r(e3, t), t & 1 && e3.report(91), y(e3, t | 32, 67174411);
  let l = V(e3, t, u, 0, 1, e3.tokenStart);
  y(e3, t | 32, 16);
  let f = he(e3, t, n, u, 2, o, 0);
  return e3.finishNode({ type: "WithStatement", object: l, body: f }, i);
}
function En(e3, t) {
  let n = e3.tokenStart;
  return r(e3, t | 32), M(e3, t | 32), e3.finishNode({ type: "DebuggerStatement" }, n);
}
function wn(e3, t, n, u, o) {
  let i = e3.tokenStart;
  r(e3, t | 32);
  let l = n?.createChildScope(16), f = ae(e3, t, l, u, { $: o }), { tokenStart: c } = e3, g = C(e3, t | 32, 20557) ? Sn(e3, t, n, u, o, c) : null, d = null;
  if (e3.getToken() === 20566) {
    r(e3, t | 32);
    let a = n?.createChildScope(4);
    d = ae(e3, t, a, u, { $: o });
  }
  return !g && !d && e3.report(88), e3.finishNode({ type: "TryStatement", block: f, handler: g, finalizer: d }, i);
}
function Sn(e3, t, n, u, o, i) {
  let l = null, f = n;
  C(e3, t, 67174411) && (n = n?.createChildScope(4), l = Zt(e3, t, n, u, (e3.getToken() & 2097152) === 2097152 ? 256 : 512, 0), e3.getToken() === 18 ? e3.report(86) : e3.getToken() === 1077936155 && e3.report(87), y(e3, t | 32, 16)), f = n?.createChildScope(32);
  let c = ae(e3, t, f, u, { $: o });
  return e3.finishNode({ type: "CatchClause", param: l, body: c }, i);
}
function Bn(e3, t, n, u, o) {
  n = n?.createChildScope();
  let i = 5764;
  return t = (t | i) ^ i | 256 | 2048 | 524288 | 65536, ae(e3, t, n, u, {}, o, "StaticBlock");
}
function Fn(e3, t, n, u, o) {
  let i = e3.tokenStart;
  r(e3, t | 32);
  let l = de(e3, t, n, u, o);
  y(e3, t, 20578), y(e3, t | 32, 67174411);
  let f = V(e3, t, u, 0, 1, e3.tokenStart);
  return y(e3, t | 32, 16), C(e3, t | 32, 1074790417), e3.finishNode({ type: "DoWhileStatement", body: l, test: f }, i);
}
function Nn(e3, t, n, u, o) {
  let { tokenValue: i, tokenStart: l } = e3, f = e3.getToken(), c = N(e3, t);
  if (e3.getToken() & 2240512) {
    let g = ue(e3, t, n, u, 8, 0);
    return M(e3, t | 32), e3.finishNode({ type: "VariableDeclaration", kind: "let", declarations: g }, l);
  }
  if (e3.assignable = 1, t & 1 && e3.report(85), e3.getToken() === 21) return Ze(e3, t, n, u, o, {}, i, c, f, 0, l);
  if (e3.getToken() === 10) {
    let g;
    e3.options.lexical && (g = qe(e3, t, i)), e3.flags = (e3.flags | 128) ^ 128, c = ke(e3, t, g, u, [c], 0, l);
  } else c = F(e3, t, u, c, 0, 0, l), c = I(e3, t, u, 0, 0, l, c);
  return e3.getToken() === 18 && (c = W(e3, t, u, 0, l, c)), ie(e3, t, c, l);
}
function je(e3, t, n, u, o, i) {
  let l = e3.tokenStart;
  r(e3, t);
  let f = ue(e3, t, n, u, o, i);
  return M(e3, t | 32), e3.finishNode({ type: "VariableDeclaration", kind: o & 8 ? "let" : "const", declarations: f }, l);
}
function vt(e3, t, n, u, o) {
  let i = e3.tokenStart;
  r(e3, t);
  let l = ue(e3, t, n, u, 4, o);
  return M(e3, t | 32), e3.finishNode({ type: "VariableDeclaration", kind: "var", declarations: l }, i);
}
function ue(e3, t, n, u, o, i) {
  let l = 1, f = [At(e3, t, n, u, o, i)];
  for (; C(e3, t, 18); ) l++, f.push(At(e3, t, n, u, o, i));
  return l > 1 && i & 32 && e3.getToken() & 262144 && e3.report(61, B[e3.getToken() & 255]), f;
}
function At(e3, t, n, u, o, i) {
  let { tokenStart: l } = e3, f = e3.getToken(), c = null, g = Zt(e3, t, n, u, o, i);
  if (e3.getToken() === 1077936155) {
    if (r(e3, t | 32), c = L(e3, t, u, 1, 0, e3.tokenStart), (i & 32 || (f & 2097152) === 0) && (e3.getToken() === 471156 || e3.getToken() === 8673330 && (f & 2097152 || (o & 4) === 0 || t & 1))) throw new q(l, e3.currentLocation, 60, e3.getToken() === 471156 ? "of" : "in");
  } else (o & 16 || (f & 2097152) > 0) && (e3.getToken() & 262144) !== 262144 && e3.report(59, o & 16 ? "const" : "destructuring");
  return e3.finishNode({ type: "VariableDeclarator", id: g, init: c }, l);
}
function Ln(e3, t, n, u, o) {
  let i = e3.tokenStart;
  r(e3, t);
  let l = ((t & 2048) > 0 || (t & 2) > 0 && (t & 8) > 0) && C(e3, t, 209006);
  y(e3, t | 32, 67174411), n = n?.createChildScope(1);
  let f = null, c = null, g = 0, d = null, a = e3.getToken() === 86088 || e3.getToken() === 241737 || e3.getToken() === 86090, h, { tokenStart: A } = e3, b = e3.getToken();
  if (a) b === 241737 ? (d = N(e3, t), e3.getToken() & 2240512 ? (e3.getToken() === 8673330 ? t & 1 && e3.report(67) : d = e3.finishNode({ type: "VariableDeclaration", kind: "let", declarations: ue(e3, t | 131072, n, u, 8, 32) }, A), e3.assignable = 1) : t & 1 ? e3.report(67) : (a = !1, e3.assignable = 1, d = F(e3, t, u, d, 0, 0, A), e3.getToken() === 471156 && e3.report(115))) : (r(e3, t), d = e3.finishNode(b === 86088 ? { type: "VariableDeclaration", kind: "var", declarations: ue(e3, t | 131072, n, u, 4, 32) } : { type: "VariableDeclaration", kind: "const", declarations: ue(e3, t | 131072, n, u, 16, 32) }, A), e3.assignable = 1);
  else if (b === 1074790417) l && e3.report(82);
  else if ((b & 2097152) === 2097152) {
    let T = e3.tokenStart;
    d = b === 2162700 ? j(e3, t, void 0, u, 1, 0, 0, 2, 32) : X(e3, t, void 0, u, 1, 0, 0, 2, 32), g = e3.destructible, g & 64 && e3.report(63), e3.assignable = g & 16 ? 2 : 1, d = F(e3, t | 131072, u, d, 0, 0, T);
  } else d = _(e3, t | 131072, u, 1, 0, 1);
  if ((e3.getToken() & 262144) === 262144) {
    if (e3.getToken() === 471156) {
      e3.assignable & 2 && e3.report(80, l ? "await" : "of"), K(e3, d), r(e3, t | 32), h = L(e3, t, u, 1, 0, e3.tokenStart), y(e3, t | 32, 16);
      let D = de(e3, t, n, u, o);
      return e3.finishNode({ type: "ForOfStatement", left: d, right: h, body: D, await: l }, i);
    }
    e3.assignable & 2 && e3.report(80, "in"), K(e3, d), r(e3, t | 32), l && e3.report(82), h = V(e3, t, u, 0, 1, e3.tokenStart), y(e3, t | 32, 16);
    let T = de(e3, t, n, u, o);
    return e3.finishNode({ type: "ForInStatement", body: T, left: d, right: h }, i);
  }
  l && e3.report(82), a || (g & 8 && e3.getToken() !== 1077936155 && e3.report(80, "loop"), d = I(e3, t | 131072, u, 0, 0, A, d)), e3.getToken() === 18 && (d = W(e3, t, u, 0, A, d)), y(e3, t | 32, 1074790417), e3.getToken() !== 1074790417 && (f = V(e3, t, u, 0, 1, e3.tokenStart)), y(e3, t | 32, 1074790417), e3.getToken() !== 16 && (c = V(e3, t, u, 0, 1, e3.tokenStart)), y(e3, t | 32, 16);
  let w = de(e3, t, n, u, o);
  return e3.finishNode({ type: "ForStatement", init: d, test: f, update: c, body: w }, i);
}
function Ut(e3, t, n) {
  return ge(t, e3.getToken()) || e3.report(118), (e3.getToken() & 537079808) === 537079808 && e3.report(119), n?.addBlockName(t, e3.tokenValue, 8, 0), N(e3, t);
}
function In(e3, t, n) {
  let u = e3.tokenStart;
  r(e3, t);
  let o = null, { tokenStart: i } = e3, l = [];
  if (e3.getToken() === 134283267) o = O(e3, t);
  else {
    if (e3.getToken() & 143360) {
      let g = Ut(e3, t, n);
      if (l = [e3.finishNode({ type: "ImportDefaultSpecifier", local: g }, i)], C(e3, t, 18)) switch (e3.getToken()) {
        case 8391476:
          l.push(Tt(e3, t, n));
          break;
        case 2162700:
          bt(e3, t, n, l);
          break;
        default:
          e3.report(107);
      }
    } else switch (e3.getToken()) {
      case 8391476:
        l = [Tt(e3, t, n)];
        break;
      case 2162700:
        bt(e3, t, n, l);
        break;
      case 67174411:
        return _t(e3, t, void 0, u);
      case 67108877:
        return Jt(e3, t, u);
      default:
        e3.report(30, B[e3.getToken() & 255]);
    }
    o = qn(e3, t);
  }
  let f = He(e3, t), c = { type: "ImportDeclaration", specifiers: l, source: o, attributes: f };
  return M(e3, t | 32), e3.finishNode(c, u);
}
function Tt(e3, t, n) {
  let { tokenStart: u } = e3;
  if (r(e3, t), y(e3, t, 77932), (e3.getToken() & 134217728) === 134217728) throw new q(u, e3.currentLocation, 30, B[e3.getToken() & 255]);
  return e3.finishNode({ type: "ImportNamespaceSpecifier", local: Ut(e3, t, n) }, u);
}
function qn(e3, t) {
  return y(e3, t, 209011), e3.getToken() !== 134283267 && e3.report(105, "Import"), O(e3, t);
}
function bt(e3, t, n, u) {
  for (r(e3, t); e3.getToken() & 143360 || e3.getToken() === 134283267; ) {
    let { tokenValue: o, tokenStart: i } = e3, l = e3.getToken(), f = Ce(e3, t), c;
    C(e3, t, 77932) ? ((e3.getToken() & 134217728) === 134217728 || e3.getToken() === 18 ? e3.report(106) : Be(e3, t, 16, e3.getToken(), 0), o = e3.tokenValue, c = N(e3, t)) : f.type === "Identifier" ? (Be(e3, t, 16, l, 0), c = e3.cloneIdentifier(f)) : e3.report(25, B[108]), n?.addBlockName(t, o, 8, 0), u.push(e3.finishNode({ type: "ImportSpecifier", local: c, imported: f }, i)), e3.getToken() !== 1074790415 && y(e3, t, 18);
  }
  return y(e3, t, 1074790415), u;
}
function Jt(e3, t, n) {
  let u = Xt(e3, t, e3.finishNode({ type: "Identifier", name: "import" }, n), n);
  return u = F(e3, t, void 0, u, 0, 0, n), u = I(e3, t, void 0, 0, 0, n, u), e3.getToken() === 18 && (u = W(e3, t, void 0, 0, n, u)), ie(e3, t, u, n);
}
function _t(e3, t, n, u) {
  let o = jt(e3, t, n, 0, u);
  return o = F(e3, t, n, o, 0, 0, u), e3.getToken() === 18 && (o = W(e3, t, n, 0, u, o)), ie(e3, t, o, u);
}
function Pn(e3, t, n) {
  let u = e3.leadingDecorators.decorators.length ? e3.leadingDecorators.start : e3.tokenStart;
  r(e3, t | 32);
  let o = [], i = null, l = null, f = [];
  if (C(e3, t | 32, 20561)) {
    switch (e3.getToken()) {
      case 86104: {
        i = x(e3, t, n, void 0, 4, 1, 1, 0, e3.tokenStart);
        break;
      }
      case 132:
      case 86094:
        i = ze(e3, t, n, void 0, 1);
        break;
      case 209005: {
        let { tokenStart: g } = e3;
        i = N(e3, t);
        let { flags: d } = e3;
        (d & 1) === 0 && (e3.getToken() === 86104 ? i = x(e3, t, n, void 0, 4, 1, 1, 1, g) : e3.getToken() === 67174411 ? (i = ut(e3, t, void 0, i, 1, 1, 0, d, g), i = F(e3, t, void 0, i, 0, 0, g), i = I(e3, t, void 0, 0, 0, g, i)) : e3.getToken() & 143360 && (n && (n = qe(e3, t, e3.tokenValue)), i = N(e3, t), i = ke(e3, t, n, void 0, [i], 1, g)));
        break;
      }
      default:
        i = L(e3, t, void 0, 1, 0, e3.tokenStart), M(e3, t | 32);
    }
    return n && e3.declareUnboundVariable("default"), e3.finishNode({ type: "ExportDefaultDeclaration", declaration: i }, u);
  }
  switch (e3.getToken()) {
    case 8391476: {
      r(e3, t);
      let g = null;
      C(e3, t, 77932) && (n && e3.declareUnboundVariable(e3.tokenValue), g = Ce(e3, t)), y(e3, t, 209011), e3.getToken() !== 134283267 && e3.report(105, "Export"), l = O(e3, t);
      let a = He(e3, t), h = { type: "ExportAllDeclaration", source: l, exported: g, attributes: a };
      return M(e3, t | 32), e3.finishNode(h, u);
    }
    case 2162700: {
      r(e3, t);
      let g = [], d = [], a = 0;
      for (; e3.getToken() & 143360 || e3.getToken() === 134283267; ) {
        let { tokenStart: h, tokenValue: A } = e3, b = Ce(e3, t);
        b.type === "Literal" && (a = 1);
        let w;
        e3.getToken() === 77932 ? (r(e3, t), (e3.getToken() & 143360) === 0 && e3.getToken() !== 134283267 && e3.report(106), n && (g.push(e3.tokenValue), d.push(A)), w = Ce(e3, t)) : (n && (g.push(e3.tokenValue), d.push(e3.tokenValue)), w = b.type === "Literal" ? e3.cloneStringLiteral(b) : e3.cloneIdentifier(b)), o.push(e3.finishNode({ type: "ExportSpecifier", local: b, exported: w }, h)), e3.getToken() !== 1074790415 && y(e3, t, 18);
      }
      y(e3, t, 1074790415), C(e3, t, 209011) ? (e3.getToken() !== 134283267 && e3.report(105, "Export"), l = O(e3, t), f = He(e3, t), n && g.forEach((h) => e3.declareUnboundVariable(h))) : (a && e3.report(172), n && (g.forEach((h) => e3.declareUnboundVariable(h)), d.forEach((h) => e3.addBindingToExports(h)))), M(e3, t | 32);
      break;
    }
    case 132:
    case 86094:
      i = ze(e3, t, n, void 0, 2);
      break;
    case 86104:
      i = x(e3, t, n, void 0, 4, 1, 2, 0, e3.tokenStart);
      break;
    case 241737:
      i = je(e3, t, n, void 0, 8, 64);
      break;
    case 86090:
      i = je(e3, t, n, void 0, 16, 64);
      break;
    case 86088:
      i = vt(e3, t, n, void 0, 64);
      break;
    case 209005: {
      let { tokenStart: g } = e3;
      if (r(e3, t), (e3.flags & 1) === 0 && e3.getToken() === 86104) {
        i = x(e3, t, n, void 0, 4, 1, 2, 1, g);
        break;
      }
    }
    default:
      e3.report(30, B[e3.getToken() & 255]);
  }
  let c = { type: "ExportNamedDeclaration", declaration: i, specifiers: o, source: l, attributes: f };
  return e3.finishNode(c, u);
}
function L(e3, t, n, u, o, i) {
  let l = U(e3, t, n, 2, 0, u, o, 1, i);
  return l = F(e3, t, n, l, o, 0, i), I(e3, t, n, o, 0, i, l);
}
function W(e3, t, n, u, o, i) {
  let l = [i];
  for (; C(e3, t | 32, 18); ) l.push(L(e3, t, n, 1, u, e3.tokenStart));
  return e3.finishNode({ type: "SequenceExpression", expressions: l }, o);
}
function V(e3, t, n, u, o, i) {
  let l = L(e3, t, n, o, u, i);
  return e3.getToken() === 18 ? W(e3, t, n, u, i, l) : l;
}
function I(e3, t, n, u, o, i, l) {
  let f = e3.getToken();
  if ((f & 4194304) === 4194304) {
    e3.assignable & 2 && e3.report(26), (!o && f === 1077936155 && l.type === "ArrayExpression" || l.type === "ObjectExpression") && K(e3, l), r(e3, t | 32);
    let c = L(e3, t, n, 1, u, e3.tokenStart);
    return e3.assignable = 2, e3.finishNode(o ? { type: "AssignmentPattern", left: l, right: c } : { type: "AssignmentExpression", left: l, operator: B[f & 255], right: c }, i);
  }
  return (f & 8388608) === 8388608 && (l = G(e3, t, n, u, i, 4, f, l)), C(e3, t | 32, 22) && (l = ee(e3, t, n, l, i)), l;
}
function be(e3, t, n, u, o, i, l) {
  let f = e3.getToken();
  r(e3, t | 32);
  let c = L(e3, t, n, 1, u, e3.tokenStart);
  return l = e3.finishNode(o ? { type: "AssignmentPattern", left: l, right: c } : { type: "AssignmentExpression", left: l, operator: B[f & 255], right: c }, i), e3.assignable = 2, l;
}
function ee(e3, t, n, u, o) {
  let i = L(e3, (t | 131072) ^ 131072, n, 1, 0, e3.tokenStart);
  y(e3, t | 32, 21), e3.assignable = 1;
  let l = L(e3, t, n, 1, 0, e3.tokenStart);
  return e3.assignable = 2, e3.finishNode({ type: "ConditionalExpression", test: u, consequent: i, alternate: l }, o);
}
function G(e3, t, n, u, o, i, l, f) {
  let c = -((t & 131072) > 0) & 8673330, g, d;
  for (e3.assignable = 2; e3.getToken() & 8388608 && (g = e3.getToken(), d = g & 3840, (g & 524288 && l & 268435456 || l & 524288 && g & 268435456) && e3.report(165), !(d + ((g === 8391735) << 8) - ((c === g) << 12) <= i)); ) r(e3, t | 32), f = e3.finishNode({ type: g & 524288 || g & 268435456 ? "LogicalExpression" : "BinaryExpression", left: f, right: G(e3, t, n, u, e3.tokenStart, d, g, _(e3, t, n, 0, u, 1)), operator: B[g & 255] }, o);
  return e3.getToken() === 1077936155 && e3.report(26), f;
}
function On(e3, t, n, u, o) {
  u || e3.report(0);
  let { tokenStart: i } = e3, l = e3.getToken();
  r(e3, t | 32);
  let f = _(e3, t, n, 0, o, 1);
  return e3.getToken() === 8391735 && e3.report(33), t & 1 && l === 16863276 && (f.type === "Identifier" ? e3.report(121) : tn(f) && e3.report(127)), e3.assignable = 2, e3.finishNode({ type: "UnaryExpression", operator: B[l & 255], argument: f, prefix: !0 }, i);
}
function Vn(e3, t, n, u, o, i, l, f) {
  let c = e3.getToken(), g = N(e3, t), { flags: d } = e3;
  if ((d & 1) === 0) {
    if (e3.getToken() === 86104) return zt(e3, t, n, 1, u, f);
    if (ge(t, e3.getToken())) return o || e3.report(0), (e3.getToken() & 36864) === 36864 && (e3.flags |= 256), Wt(e3, t, n, i, f);
  }
  return !l && e3.getToken() === 67174411 ? ut(e3, t, n, g, i, 1, 0, d, f) : e3.getToken() === 10 ? (Ie(e3, t, c), l && e3.report(51), (c & 36864) === 36864 && (e3.flags |= 256), Oe(e3, t, n, e3.tokenValue, g, l, i, 0, f)) : (e3.assignable = 1, g);
}
function Rn(e3, t, n, u, o, i) {
  if (u && (e3.destructible |= 256), t & 1024) {
    r(e3, t | 32), t & 8192 && e3.report(32), o || e3.report(26), e3.getToken() === 22 && e3.report(124);
    let l = null, f = !1;
    return (e3.flags & 1) === 0 ? (f = C(e3, t | 32, 8391476), (e3.getToken() & 77824 || f) && (l = L(e3, t, n, 1, 0, e3.tokenStart))) : e3.getToken() === 8391476 && e3.report(30, B[e3.getToken() & 255]), e3.assignable = 2, e3.finishNode({ type: "YieldExpression", argument: l, delegate: f }, i);
  }
  return t & 1 && e3.report(97, "yield"), nt(e3, t, n);
}
function Mn(e3, t, n, u, o, i) {
  o && (e3.destructible |= 128), t & 524288 && e3.report(177);
  let l = nt(e3, t, n);
  if (l.type === "ArrowFunctionExpression" || (e3.getToken() & 65536) === 0) {
    if (t & 2048) throw new q(i, { index: e3.startIndex, line: e3.startLine, column: e3.startColumn }, 176);
    if (t & 2) throw new q(i, { index: e3.startIndex, line: e3.startLine, column: e3.startColumn }, 110);
    if (t & 8192 && t & 2048) throw new q(i, { index: e3.startIndex, line: e3.startLine, column: e3.startColumn }, 110);
    return l;
  }
  if (t & 8192) throw new q(i, { index: e3.startIndex, line: e3.startLine, column: e3.startColumn }, 31);
  if (t & 2048 || t & 2 && t & 8) {
    if (u) throw new q(i, { index: e3.startIndex, line: e3.startLine, column: e3.startColumn }, 0);
    let c = _(e3, t, n, 0, 0, 1);
    return e3.getToken() === 8391735 && e3.report(33), e3.assignable = 2, e3.finishNode({ type: "AwaitExpression", argument: c }, i);
  }
  if (t & 2) throw new q(i, { index: e3.startIndex, line: e3.startLine, column: e3.startColumn }, 98);
  return l;
}
function Pe(e3, t, n, u, o, i, l) {
  let { tokenStart: f } = e3;
  y(e3, t | 32, 2162700);
  let c = [];
  if (e3.getToken() !== 1074790415) {
    for (; e3.getToken() === 134283267; ) {
      let { index: g, tokenStart: d, tokenIndex: a, tokenValue: h } = e3, A = e3.getToken(), b = O(e3, t);
      if (Pt(e3, g, a, h)) {
        if (t |= 1, e3.flags & 128) throw new q(d, e3.currentLocation, 66);
        if (e3.flags & 64) throw new q(d, e3.currentLocation, 9);
        if (e3.flags & 4096) throw new q(d, e3.currentLocation, 15);
        l?.reportScopeError();
      }
      c.push(Ge(e3, t, b, A, d));
    }
    t & 1 && (i && ((i & 537079808) === 537079808 && e3.report(119), (i & 36864) === 36864 && e3.report(40)), e3.flags & 512 && e3.report(119), e3.flags & 256 && e3.report(118));
  }
  for (e3.flags = (e3.flags | 512 | 256 | 64 | 4096) ^ 4928, e3.destructible = (e3.destructible | 256) ^ 256; e3.getToken() !== 1074790415; ) c.push(re(e3, t, n, u, 4, {}));
  return y(e3, o & 24 ? t | 32 : t, 1074790415), e3.flags &= -4289, e3.getToken() === 1077936155 && e3.report(26), e3.finishNode({ type: "BlockStatement", body: c }, f);
}
function vn(e3, t) {
  let { tokenStart: n } = e3;
  switch (r(e3, t), e3.getToken()) {
    case 67108990:
      e3.report(167);
    case 67174411: {
      (t & 512) === 0 && e3.report(28), e3.assignable = 2;
      break;
    }
    case 69271571:
    case 67108877: {
      (t & 256) === 0 && e3.report(29), e3.assignable = 1;
      break;
    }
    default:
      e3.report(30, "super");
  }
  return e3.finishNode({ type: "Super" }, n);
}
function _(e3, t, n, u, o, i) {
  let l = e3.tokenStart, f = U(e3, t, n, 2, 0, u, o, i, l);
  return F(e3, t, n, f, o, 0, l);
}
function Un(e3, t, n, u) {
  e3.assignable & 2 && e3.report(55);
  let o = e3.getToken();
  return r(e3, t), e3.assignable = 2, e3.finishNode({ type: "UpdateExpression", argument: n, operator: B[o & 255], prefix: !1 }, u);
}
function F(e3, t, n, u, o, i, l) {
  if ((e3.getToken() & 33619968) === 33619968 && (e3.flags & 1) === 0) u = Un(e3, t, u, l);
  else if ((e3.getToken() & 67108864) === 67108864) {
    switch (t = (t | 131072) ^ 131072, e3.getToken()) {
      case 67108877: {
        r(e3, (t | 262144 | 8) ^ 8), t & 16 && e3.getToken() === 130 && e3.tokenValue === "super" && e3.report(173), e3.assignable = 1;
        let f = xe(e3, t | 64, n);
        u = e3.finishNode({ type: "MemberExpression", object: u, computed: !1, property: f, optional: !1 }, l);
        break;
      }
      case 69271571: {
        let f = !1;
        (e3.flags & 2048) === 2048 && (f = !0, e3.flags = (e3.flags | 2048) ^ 2048), r(e3, t | 32);
        let { tokenStart: c } = e3, g = V(e3, t, n, o, 1, c);
        y(e3, t, 20), e3.assignable = 1, u = e3.finishNode({ type: "MemberExpression", object: u, computed: !0, property: g, optional: !1 }, l), f && (e3.flags |= 2048);
        break;
      }
      case 67174411: {
        if ((e3.flags & 1024) === 1024) return e3.flags = (e3.flags | 1024) ^ 1024, u;
        let f = !1;
        (e3.flags & 2048) === 2048 && (f = !0, e3.flags = (e3.flags | 2048) ^ 2048);
        let c = tt(e3, t, n, o);
        e3.assignable = 2, u = e3.finishNode({ type: "CallExpression", callee: u, arguments: c, optional: !1 }, l), f && (e3.flags |= 2048);
        break;
      }
      case 67108990: {
        r(e3, (t | 262144 | 8) ^ 8), e3.flags |= 2048, e3.assignable = 2, u = Jn(e3, t, n, u, l);
        break;
      }
      default:
        (e3.flags & 2048) === 2048 && e3.report(166), e3.assignable = 2, u = e3.finishNode({ type: "TaggedTemplateExpression", tag: u, quasi: e3.getToken() === 67174408 ? et(e3, t | 64, n) : pe(e3, t) }, l);
    }
    u = F(e3, t, n, u, 0, 1, l);
  }
  return i === 0 && (e3.flags & 2048) === 2048 && (e3.flags = (e3.flags | 2048) ^ 2048, u = e3.finishNode({ type: "ChainExpression", expression: u }, l)), u;
}
function Jn(e3, t, n, u, o) {
  let i = !1, l;
  if ((e3.getToken() === 69271571 || e3.getToken() === 67174411) && (e3.flags & 2048) === 2048 && (i = !0, e3.flags = (e3.flags | 2048) ^ 2048), e3.getToken() === 69271571) {
    r(e3, t | 32);
    let { tokenStart: f } = e3, c = V(e3, t, n, 0, 1, f);
    y(e3, t, 20), e3.assignable = 2, l = e3.finishNode({ type: "MemberExpression", object: u, computed: !0, optional: !0, property: c }, o);
  } else if (e3.getToken() === 67174411) {
    let f = tt(e3, t, n, 0);
    e3.assignable = 2, l = e3.finishNode({ type: "CallExpression", callee: u, arguments: f, optional: !0 }, o);
  } else {
    let f = xe(e3, t, n);
    e3.assignable = 2, l = e3.finishNode({ type: "MemberExpression", object: u, computed: !1, optional: !0, property: f }, o);
  }
  return i && (e3.flags |= 2048), l;
}
function xe(e3, t, n) {
  return (e3.getToken() & 143360) === 0 && e3.getToken() !== -2147483528 && e3.getToken() !== -2147483527 && e3.getToken() !== 130 && e3.report(160), e3.getToken() === 130 ? Le(e3, t, n, 0) : N(e3, t);
}
function _n(e3, t, n, u, o, i) {
  u && e3.report(56), o || e3.report(0);
  let l = e3.getToken();
  r(e3, t | 32);
  let f = _(e3, t, n, 0, 0, 1);
  return e3.assignable & 2 && e3.report(55), e3.assignable = 2, e3.finishNode({ type: "UpdateExpression", argument: f, operator: B[l & 255], prefix: !0 }, i);
}
function U(e3, t, n, u, o, i, l, f, c) {
  if ((e3.getToken() & 143360) === 143360) {
    switch (e3.getToken()) {
      case 209006:
        return Mn(e3, t, n, o, l, c);
      case 241771:
        return Rn(e3, t, n, l, i, c);
      case 209005:
        return Vn(e3, t, n, l, f, i, o, c);
    }
    let { tokenValue: g } = e3, d = e3.getToken(), a = N(e3, t | 64);
    return e3.getToken() === 10 ? (f || e3.report(0), Ie(e3, t, d), (d & 36864) === 36864 && (e3.flags |= 256), Oe(e3, t, n, g, a, o, i, 0, c)) : (t & 16 && !(t & 32768) && !(t & 8192) && e3.tokenValue === "arguments" && e3.report(130), (d & 255) === 73 && (t & 1 && e3.report(113), u & 24 && e3.report(100)), e3.assignable = t & 1 && (d & 537079808) === 537079808 ? 2 : 1, a);
  }
  if ((e3.getToken() & 134217728) === 134217728) return O(e3, t);
  switch (e3.getToken()) {
    case 33619993:
    case 33619994:
      return _n(e3, t, n, o, f, c);
    case 16863276:
    case 16842798:
    case 16842799:
    case 25233968:
    case 25233969:
    case 16863275:
    case 16863277:
      return On(e3, t, n, f, l);
    case 86104:
      return zt(e3, t, n, 0, l, c);
    case 2162700:
      return Yn(e3, t, n, i ? 0 : 1, l);
    case 69271571:
      return Wn(e3, t, n, i ? 0 : 1, l);
    case 67174411:
      return Zn(e3, t | 64, n, i, 1, 0, c);
    case 86021:
    case 86022:
    case 86023:
      return Kn(e3, t);
    case 86111:
      return $n(e3, t);
    case 65540:
      return pn(e3, t);
    case 132:
    case 86094:
      return eu(e3, t, n, l, c);
    case 86109:
      return vn(e3, t);
    case 67174409:
      return pe(e3, t);
    case 67174408:
      return et(e3, t, n);
    case 86107:
      return Gn(e3, t, n, l);
    case 134283388:
      return Ht(e3, t);
    case 130:
      return Le(e3, t, n, 0);
    case 86106:
      return Xn(e3, t, n, o, l, c);
    case 8456256:
      if (e3.options.jsx) return Re(e3, t, n, 0, e3.tokenStart);
    default:
      if (ge(t, e3.getToken())) return nt(e3, t, n);
      e3.report(30, B[e3.getToken() & 255]);
  }
}
function Xn(e3, t, n, u, o, i) {
  let l = N(e3, t);
  return e3.getToken() === 67108877 ? Xt(e3, t, l, i) : (u && e3.report(142), l = jt(e3, t, n, o, i), e3.assignable = 2, F(e3, t, n, l, o, 0, i));
}
function Xt(e3, t, n, u) {
  (t & 2) === 0 && e3.report(169), r(e3, t);
  let o = e3.getToken();
  return o !== 209030 && e3.tokenValue !== "meta" ? e3.report(174) : o & -2147483648 && e3.report(175), e3.assignable = 2, e3.finishNode({ type: "MetaProperty", meta: n, property: N(e3, t) }, u);
}
function jt(e3, t, n, u, o) {
  y(e3, t | 32, 67174411), e3.getToken() === 14 && e3.report(143);
  let i = L(e3, t, n, 1, u, e3.tokenStart), l = null;
  if (e3.getToken() === 18) {
    if (y(e3, t, 18), e3.getToken() !== 16) {
      let c = (t | 131072) ^ 131072;
      l = L(e3, c, n, 1, u, e3.tokenStart);
    }
    C(e3, t, 18);
  }
  let f = { type: "ImportExpression", source: i, options: l };
  return y(e3, t, 16), e3.finishNode(f, o);
}
function He(e3, t) {
  if (!C(e3, t, 20579)) return [];
  y(e3, t, 2162700);
  let n = [], u = /* @__PURE__ */ new Set();
  for (; e3.getToken() !== 1074790415; ) {
    let o = e3.tokenStart, i = Hn(e3, t);
    y(e3, t, 21);
    let l = jn(e3, t), f = i.type === "Literal" ? i.value : i.name;
    u.has(f) && e3.report(145, `${f}`), u.add(f), n.push(e3.finishNode({ type: "ImportAttribute", key: i, value: l }, o)), e3.getToken() !== 1074790415 && y(e3, t, 18);
  }
  return y(e3, t, 1074790415), n;
}
function jn(e3, t) {
  if (e3.getToken() === 134283267) return O(e3, t);
  e3.report(30, B[e3.getToken() & 255]);
}
function Hn(e3, t) {
  if (e3.getToken() === 134283267) return O(e3, t);
  if (e3.getToken() & 143360) return N(e3, t);
  e3.report(30, B[e3.getToken() & 255]);
}
function Ce(e3, t) {
  if (e3.getToken() === 134283267) {
    let n = e3.tokenValue;
    return mt(0, n) || e3.report(171), O(e3, t);
  } else {
    if (e3.getToken() & 143360) return N(e3, t);
    e3.report(30, B[e3.getToken() & 255]);
  }
}
function Ht(e3, t) {
  let { tokenRaw: n, tokenValue: u, tokenStart: o } = e3;
  r(e3, t), e3.assignable = 2;
  let i = { type: "Literal", value: u, bigint: String(u) };
  return e3.options.raw && (i.raw = n), e3.finishNode(i, o);
}
function pe(e3, t) {
  e3.assignable = 2;
  let { tokenValue: n, tokenRaw: u, tokenStart: o } = e3;
  y(e3, t, 67174409);
  let i = [Ee(e3, n, u, o, !0)];
  return e3.finishNode({ type: "TemplateLiteral", expressions: [], quasis: i }, o);
}
function et(e3, t, n) {
  t = (t | 131072) ^ 131072;
  let { tokenValue: u, tokenRaw: o, tokenStart: i } = e3;
  y(e3, t & -65 | 32, 67174408);
  let l = [Ee(e3, u, o, i, !1)], f = [V(e3, t & -65, n, 0, 1, e3.tokenStart)];
  for (e3.getToken() !== 1074790415 && e3.report(83); e3.setToken(p2(e3, t), !0) !== 67174409; ) {
    let { tokenValue: c, tokenRaw: g, tokenStart: d } = e3;
    y(e3, t & -65 | 32, 67174408), l.push(Ee(e3, c, g, d, !1)), f.push(V(e3, t, n, 0, 1, e3.tokenStart)), e3.getToken() !== 1074790415 && e3.report(83);
  }
  {
    let { tokenValue: c, tokenRaw: g, tokenStart: d } = e3;
    y(e3, t, 67174409), l.push(Ee(e3, c, g, d, !0));
  }
  return e3.finishNode({ type: "TemplateLiteral", expressions: f, quasis: l }, i);
}
function Ee(e3, t, n, u, o) {
  let i = e3.finishNode({ type: "TemplateElement", value: { cooked: t, raw: n }, tail: o }, u), l = o ? 1 : 2;
  return e3.options.ranges && (i.start += 1, i.range[0] += 1, i.end -= l, i.range[1] -= l), e3.options.loc && (i.loc.start.column += 1, i.loc.end.column -= l), i;
}
function zn(e3, t, n) {
  let u = e3.tokenStart;
  t = (t | 131072) ^ 131072, y(e3, t | 32, 14);
  let o = L(e3, t, n, 1, 0, e3.tokenStart);
  return e3.assignable = 1, e3.finishNode({ type: "SpreadElement", argument: o }, u);
}
function tt(e3, t, n, u) {
  r(e3, t | 32);
  let o = [];
  if (e3.getToken() === 16) return r(e3, t | 64), o;
  for (; e3.getToken() !== 16 && (e3.getToken() === 14 ? o.push(zn(e3, t, n)) : o.push(L(e3, t, n, 1, u, e3.tokenStart)), !(e3.getToken() !== 18 || (r(e3, t | 32), e3.getToken() === 16))); ) ;
  return y(e3, t | 64, 16), o;
}
function N(e3, t) {
  let { tokenValue: n, tokenStart: u } = e3, o = n === "await" && (e3.getToken() & -2147483648) === 0;
  return r(e3, t | (o ? 32 : 0)), e3.finishNode({ type: "Identifier", name: n }, u);
}
function O(e3, t) {
  let { tokenValue: n, tokenRaw: u, tokenStart: o } = e3;
  if (e3.getToken() === 134283388) return Ht(e3, t);
  let i = { type: "Literal", value: n };
  return e3.options.raw && (i.raw = u), r(e3, t), e3.assignable = 2, e3.finishNode(i, o);
}
function Kn(e3, t) {
  let n = e3.tokenStart, u = B[e3.getToken() & 255], i = { type: "Literal", value: e3.getToken() === 86023 ? null : u === "true" };
  return e3.options.raw && (i.raw = u), r(e3, t), e3.assignable = 2, e3.finishNode(i, n);
}
function $n(e3, t) {
  let { tokenStart: n } = e3;
  return r(e3, t), e3.assignable = 2, e3.finishNode({ type: "ThisExpression" }, n);
}
function x(e3, t, n, u, o, i, l, f, c) {
  r(e3, t | 32);
  let g = i ? Qe(e3, t, 8391476) : 0, d = null, a, h = n ? e3.createScope() : void 0;
  if (e3.getToken() === 67174411) (l & 1) === 0 && e3.report(39, "Function");
  else {
    let T = o & 4 && ((t & 8) === 0 || (t & 2) === 0) ? 4 : 64 | (f ? 1024 : 0) | (g ? 1024 : 0);
    Ot(e3, t, e3.getToken()), n && (T & 4 ? n.addVarName(t, e3.tokenValue, T) : n.addBlockName(t, e3.tokenValue, T, o), h = h?.createChildScope(128), l && l & 2 && e3.declareUnboundVariable(e3.tokenValue)), a = e3.getToken(), e3.getToken() & 143360 ? d = N(e3, t) : e3.report(30, B[e3.getToken() & 255]);
  }
  t = (t | 28416) ^ 28416 | 65536 | (f ? 2048 : 0) | (g ? 1024 : 0) | (g ? 0 : 262144), h = h?.createChildScope(256);
  let A = $t(e3, (t | 8192) & -524289, h, u, 0, 1), b = 524428, w = Pe(e3, (t | b) ^ b | 32768 | 4096, h?.createChildScope(64), u, 8, a, h);
  return e3.finishNode({ type: "FunctionDeclaration", id: d, params: A, body: w, async: f === 1, generator: g === 1 }, c);
}
function zt(e3, t, n, u, o, i) {
  r(e3, t | 32);
  let l = Qe(e3, t, 8391476), f = (u ? 2048 : 0) | (l ? 1024 : 0), c = null, g, d = e3.createScopeIfLexical(), a = 552704;
  e3.getToken() & 143360 && (Ot(e3, (t | a) ^ a | f, e3.getToken()), d = d?.createChildScope(128), g = e3.getToken(), c = N(e3, t)), t = (t | a) ^ a | 65536 | f | (l ? 0 : 262144), d = d?.createChildScope(256);
  let h = $t(e3, (t | 8192) & -524289, d, n, o, 1), A = Pe(e3, t & -131229 | 32768 | 4096, d?.createChildScope(64), n, 0, g, d);
  return e3.assignable = 2, e3.finishNode({ type: "FunctionExpression", id: c, params: h, body: A, async: u === 1, generator: l === 1 }, i);
}
function Wn(e3, t, n, u, o) {
  let i = X(e3, t, void 0, n, u, o, 0, 2, 0);
  return e3.destructible & 64 && e3.report(63), e3.destructible & 8 && e3.report(62), i;
}
function X(e3, t, n, u, o, i, l, f, c) {
  let { tokenStart: g } = e3;
  r(e3, t | 32);
  let d = [], a = 0;
  for (t = (t | 131072) ^ 131072; e3.getToken() !== 20; ) if (C(e3, t | 32, 18)) d.push(null);
  else {
    let A, { tokenStart: b, tokenValue: w } = e3, T = e3.getToken();
    if (T & 143360) if (A = U(e3, t, u, f, 0, 1, i, 1, b), e3.getToken() === 1077936155) {
      e3.assignable & 2 && e3.report(26), r(e3, t | 32), n?.addVarOrBlock(t, w, f, c);
      let D = L(e3, t, u, 1, i, e3.tokenStart);
      A = e3.finishNode(l ? { type: "AssignmentPattern", left: A, right: D } : { type: "AssignmentExpression", operator: "=", left: A, right: D }, b), a |= e3.destructible & 256 ? 256 : 0 | e3.destructible & 128 ? 128 : 0;
    } else e3.getToken() === 18 || e3.getToken() === 20 ? (e3.assignable & 2 ? a |= 16 : n?.addVarOrBlock(t, w, f, c), a |= e3.destructible & 256 ? 256 : 0 | e3.destructible & 128 ? 128 : 0) : (a |= f & 1 ? 32 : (f & 2) === 0 ? 16 : 0, A = F(e3, t, u, A, i, 0, b), e3.getToken() !== 18 && e3.getToken() !== 20 ? (e3.getToken() !== 1077936155 && (a |= 16), A = I(e3, t, u, i, l, b, A)) : e3.getToken() !== 1077936155 && (a |= e3.assignable & 2 ? 16 : 32));
    else T & 2097152 ? (A = e3.getToken() === 2162700 ? j(e3, t, n, u, 0, i, l, f, c) : X(e3, t, n, u, 0, i, l, f, c), a |= e3.destructible, e3.assignable = e3.destructible & 16 ? 2 : 1, e3.getToken() === 18 || e3.getToken() === 20 ? e3.assignable & 2 && (a |= 16) : e3.destructible & 8 ? e3.report(71) : (A = F(e3, t, u, A, i, 0, b), a = e3.assignable & 2 ? 16 : 0, e3.getToken() !== 18 && e3.getToken() !== 20 ? A = I(e3, t, u, i, l, b, A) : e3.getToken() !== 1077936155 && (a |= e3.assignable & 2 ? 16 : 32))) : T === 14 ? (A = oe(e3, t, n, u, 20, f, c, 0, i, l), a |= e3.destructible, e3.getToken() !== 18 && e3.getToken() !== 20 && e3.report(30, B[e3.getToken() & 255])) : (A = _(e3, t, u, 1, 0, 1), e3.getToken() !== 18 && e3.getToken() !== 20 ? (A = I(e3, t, u, i, l, b, A), (f & 3) === 0 && T === 67174411 && (a |= 16)) : e3.assignable & 2 ? a |= 16 : T === 67174411 && (a |= e3.assignable & 1 && f & 3 ? 32 : 16));
    if (d.push(A), C(e3, t | 32, 18)) {
      if (e3.getToken() === 20) break;
    } else break;
  }
  y(e3, t, 20);
  let h = e3.finishNode({ type: l ? "ArrayPattern" : "ArrayExpression", elements: d }, g);
  return !o && e3.getToken() & 4194304 ? Kt(e3, t, u, a, i, l, g, h) : (e3.destructible = a, h);
}
function Kt(e3, t, n, u, o, i, l, f) {
  e3.getToken() !== 1077936155 && e3.report(26), r(e3, t | 32), u & 16 && e3.report(26), i || K(e3, f);
  let { tokenStart: c } = e3, g = L(e3, t, n, 1, o, c);
  return e3.destructible = (u | 64 | 8) ^ 72 | (e3.destructible & 128 ? 128 : 0) | (e3.destructible & 256 ? 256 : 0), e3.finishNode(i ? { type: "AssignmentPattern", left: f, right: g } : { type: "AssignmentExpression", left: f, operator: "=", right: g }, l);
}
function oe(e3, t, n, u, o, i, l, f, c, g) {
  let { tokenStart: d } = e3;
  r(e3, t | 32);
  let a = null, h = 0, { tokenValue: A, tokenStart: b } = e3, w = e3.getToken();
  if (w & 143360) e3.assignable = 1, a = U(e3, t, u, i, 0, 1, c, 1, b), w = e3.getToken(), a = F(e3, t, u, a, c, 0, b), e3.getToken() !== 18 && e3.getToken() !== o && (e3.assignable & 2 && e3.getToken() === 1077936155 && e3.report(71), h |= 16, a = I(e3, t, u, c, g, b, a)), e3.assignable & 2 ? h |= 16 : w === o || w === 18 ? n?.addVarOrBlock(t, A, i, l) : h |= 32, h |= e3.destructible & 128 ? 128 : 0;
  else if (w === o) e3.report(41);
  else if (w & 2097152) a = e3.getToken() === 2162700 ? j(e3, t, n, u, 1, c, g, i, l) : X(e3, t, n, u, 1, c, g, i, l), w = e3.getToken(), w !== 1077936155 && w !== o && w !== 18 ? (e3.destructible & 8 && e3.report(71), a = F(e3, t, u, a, c, 0, b), h |= e3.assignable & 2 ? 16 : 0, (e3.getToken() & 4194304) === 4194304 ? (e3.getToken() !== 1077936155 && (h |= 16), a = I(e3, t, u, c, g, b, a)) : ((e3.getToken() & 8388608) === 8388608 && (a = G(e3, t, u, 1, b, 4, w, a)), C(e3, t | 32, 22) && (a = ee(e3, t, u, a, b)), h |= e3.assignable & 2 ? 16 : 32)) : h |= o === 1074790415 && w !== 1077936155 ? 16 : e3.destructible;
  else {
    h |= 32, a = _(e3, t, u, 1, c, 1);
    let { tokenStart: T } = e3, D = e3.getToken();
    return D === 1077936155 ? (e3.assignable & 2 && e3.report(26), a = I(e3, t, u, c, g, T, a), h |= 16) : (D === 18 ? h |= 16 : D !== o && (a = I(e3, t, u, c, g, T, a)), h |= e3.assignable & 1 ? 32 : 16), e3.destructible = h, e3.getToken() !== o && e3.getToken() !== 18 && e3.report(161), e3.finishNode({ type: g ? "RestElement" : "SpreadElement", argument: a }, d);
  }
  if (e3.getToken() !== o) if (i & 1 && (h |= f ? 16 : 32), C(e3, t | 32, 1077936155)) {
    h & 16 && e3.report(26), K(e3, a);
    let T = L(e3, t, u, 1, c, e3.tokenStart);
    a = e3.finishNode(g ? { type: "AssignmentPattern", left: a, right: T } : { type: "AssignmentExpression", left: a, operator: "=", right: T }, b), h = 16;
  } else h |= 16;
  return e3.destructible = h, e3.finishNode({ type: g ? "RestElement" : "SpreadElement", argument: a }, d);
}
function z(e3, t, n, u, o, i) {
  let l = 11264 | ((u & 64) === 0 ? 16896 : 0);
  t = (t | l) ^ l | (u & 8 ? 1024 : 0) | (u & 16 ? 2048 : 0) | (u & 64 ? 16384 : 0) | 256 | 32768 | 65536;
  let f = e3.createScopeIfLexical(256), c = Qn(e3, (t | 8192) & -524289, f, n, u, 1, o);
  f = f?.createChildScope(64);
  let g = Pe(e3, t & -655373 | 32768 | 4096, f, n, 0, void 0, f?.parent);
  return e3.finishNode({ type: "FunctionExpression", params: c, body: g, async: (u & 16) > 0, generator: (u & 8) > 0, id: null }, i);
}
function Yn(e3, t, n, u, o) {
  let i = j(e3, t, void 0, n, u, o, 0, 2, 0);
  return e3.destructible & 64 && e3.report(63), e3.destructible & 8 && e3.report(62), i;
}
function j(e3, t, n, u, o, i, l, f, c) {
  let { tokenStart: g } = e3;
  r(e3, t);
  let d = [], a = 0, h = 0;
  for (t = (t | 131072) ^ 131072; e3.getToken() !== 1074790415; ) {
    let { tokenValue: b, tokenStart: w } = e3, T = e3.getToken();
    if (T === 14) d.push(oe(e3, t, n, u, 1074790415, f, c, 0, i, l));
    else {
      let D = 0, R = null, k;
      if (e3.getToken() & 143360 || e3.getToken() === -2147483528 || e3.getToken() === -2147483527) if (e3.getToken() === -2147483527 && (a |= 16), R = N(e3, t), e3.getToken() === 18 || e3.getToken() === 1074790415 || e3.getToken() === 1077936155) if (D |= 4, t & 1 && (T & 537079808) === 537079808 ? a |= 16 : Be(e3, t, f, T, 0), n?.addVarOrBlock(t, b, f, c), C(e3, t | 32, 1077936155)) {
        a |= 8;
        let E = L(e3, t, u, 1, i, e3.tokenStart);
        a |= e3.destructible & 256 ? 256 : 0 | e3.destructible & 128 ? 128 : 0, k = e3.finishNode({ type: "AssignmentPattern", left: e3.cloneIdentifier(R), right: E }, w);
      } else a |= (T === 209006 ? 128 : 0) | (T === -2147483528 ? 16 : 0), k = e3.cloneIdentifier(R);
      else if (C(e3, t | 32, 21)) {
        let { tokenStart: E } = e3;
        if (b === "__proto__" && h++, e3.getToken() & 143360) {
          let ce = e3.getToken(), Y = e3.tokenValue;
          k = U(e3, t, u, f, 0, 1, i, 1, E);
          let $ = e3.getToken();
          k = F(e3, t, u, k, i, 0, E), e3.getToken() === 18 || e3.getToken() === 1074790415 ? $ === 1077936155 || $ === 1074790415 || $ === 18 ? (a |= e3.destructible & 128 ? 128 : 0, e3.assignable & 2 ? a |= 16 : (ce & 143360) === 143360 && n?.addVarOrBlock(t, Y, f, c)) : a |= e3.assignable & 1 ? 32 : 16 : (e3.getToken() & 4194304) === 4194304 ? (e3.assignable & 2 ? a |= 16 : $ !== 1077936155 ? a |= 32 : n?.addVarOrBlock(t, Y, f, c), k = I(e3, t, u, i, l, E, k)) : (a |= 16, (e3.getToken() & 8388608) === 8388608 && (k = G(e3, t, u, 1, E, 4, $, k)), C(e3, t | 32, 22) && (k = ee(e3, t, u, k, E)));
        } else (e3.getToken() & 2097152) === 2097152 ? (k = e3.getToken() === 69271571 ? X(e3, t, n, u, 0, i, l, f, c) : j(e3, t, n, u, 0, i, l, f, c), a = e3.destructible, e3.assignable = a & 16 ? 2 : 1, e3.getToken() === 18 || e3.getToken() === 1074790415 ? e3.assignable & 2 && (a |= 16) : e3.destructible & 8 ? e3.report(71) : (k = F(e3, t, u, k, i, 0, E), a = e3.assignable & 2 ? 16 : 0, (e3.getToken() & 4194304) === 4194304 ? k = be(e3, t, u, i, l, E, k) : ((e3.getToken() & 8388608) === 8388608 && (k = G(e3, t, u, 1, E, 4, T, k)), C(e3, t | 32, 22) && (k = ee(e3, t, u, k, E)), a |= e3.assignable & 2 ? 16 : 32))) : (k = _(e3, t, u, 1, i, 1), a |= e3.assignable & 1 ? 32 : 16, e3.getToken() === 18 || e3.getToken() === 1074790415 ? e3.assignable & 2 && (a |= 16) : (k = F(e3, t, u, k, i, 0, E), a = e3.assignable & 2 ? 16 : 0, e3.getToken() !== 18 && T !== 1074790415 && (e3.getToken() !== 1077936155 && (a |= 16), k = I(e3, t, u, i, l, E, k))));
      } else e3.getToken() === 69271571 ? (a |= 16, T === 209005 && (D |= 16), D |= (T === 209008 ? 256 : T === 209009 ? 512 : 1) | 2, R = ne(e3, t, u, i), a |= e3.assignable, k = z(e3, t, u, D, i, e3.tokenStart)) : e3.getToken() & 143360 ? (a |= 16, T === -2147483528 && e3.report(95), T === 209005 ? (e3.flags & 1 && e3.report(132), D |= 17) : T === 209008 ? D |= 256 : T === 209009 ? D |= 512 : e3.report(0), R = N(e3, t), k = z(e3, t, u, D, i, e3.tokenStart)) : e3.getToken() === 67174411 ? (a |= 16, D |= 1, k = z(e3, t, u, D, i, e3.tokenStart)) : e3.getToken() === 8391476 ? (a |= 16, T === 209008 ? e3.report(42) : T === 209009 ? e3.report(43) : T !== 209005 && e3.report(30, B[52]), r(e3, t), D |= 9 | (T === 209005 ? 16 : 0), e3.getToken() & 143360 ? R = N(e3, t) : (e3.getToken() & 134217728) === 134217728 ? R = O(e3, t) : e3.getToken() === 69271571 ? (D |= 2, R = ne(e3, t, u, i), a |= e3.assignable) : e3.report(30, B[e3.getToken() & 255]), k = z(e3, t, u, D, i, e3.tokenStart)) : (e3.getToken() & 134217728) === 134217728 ? (T === 209005 && (D |= 16), D |= T === 209008 ? 256 : T === 209009 ? 512 : 1, a |= 16, R = O(e3, t), k = z(e3, t, u, D, i, e3.tokenStart)) : e3.report(133);
      else if ((e3.getToken() & 134217728) === 134217728) if (R = O(e3, t), e3.getToken() === 21) {
        y(e3, t | 32, 21);
        let { tokenStart: E } = e3;
        if (b === "__proto__" && h++, e3.getToken() & 143360) {
          k = U(e3, t, u, f, 0, 1, i, 1, E);
          let { tokenValue: ce } = e3, Y = e3.getToken();
          k = F(e3, t, u, k, i, 0, E), e3.getToken() === 18 || e3.getToken() === 1074790415 ? Y === 1077936155 || Y === 1074790415 || Y === 18 ? e3.assignable & 2 ? a |= 16 : n?.addVarOrBlock(t, ce, f, c) : a |= e3.assignable & 1 ? 32 : 16 : e3.getToken() === 1077936155 ? (e3.assignable & 2 && (a |= 16), k = I(e3, t, u, i, l, E, k)) : (a |= 16, k = I(e3, t, u, i, l, E, k));
        } else (e3.getToken() & 2097152) === 2097152 ? (k = e3.getToken() === 69271571 ? X(e3, t, n, u, 0, i, l, f, c) : j(e3, t, n, u, 0, i, l, f, c), a = e3.destructible, e3.assignable = a & 16 ? 2 : 1, e3.getToken() === 18 || e3.getToken() === 1074790415 ? e3.assignable & 2 && (a |= 16) : (e3.destructible & 8) !== 8 && (k = F(e3, t, u, k, i, 0, E), a = e3.assignable & 2 ? 16 : 0, (e3.getToken() & 4194304) === 4194304 ? k = be(e3, t, u, i, l, E, k) : ((e3.getToken() & 8388608) === 8388608 && (k = G(e3, t, u, 1, E, 4, T, k)), C(e3, t | 32, 22) && (k = ee(e3, t, u, k, E)), a |= e3.assignable & 2 ? 16 : 32))) : (k = _(e3, t, u, 1, 0, 1), a |= e3.assignable & 1 ? 32 : 16, e3.getToken() === 18 || e3.getToken() === 1074790415 ? e3.assignable & 2 && (a |= 16) : (k = F(e3, t, u, k, i, 0, E), a = e3.assignable & 1 ? 0 : 16, e3.getToken() !== 18 && e3.getToken() !== 1074790415 && (e3.getToken() !== 1077936155 && (a |= 16), k = I(e3, t, u, i, l, E, k))));
      } else e3.getToken() === 67174411 ? (D |= 1, k = z(e3, t, u, D, i, e3.tokenStart), a = e3.assignable | 16) : e3.report(134);
      else if (e3.getToken() === 69271571) if (R = ne(e3, t, u, i), a |= e3.destructible & 256 ? 256 : 0, D |= 2, e3.getToken() === 21) {
        r(e3, t | 32);
        let { tokenStart: E, tokenValue: ce } = e3, Y = e3.getToken();
        if (e3.getToken() & 143360) {
          k = U(e3, t, u, f, 0, 1, i, 1, E);
          let $ = e3.getToken();
          k = F(e3, t, u, k, i, 0, E), (e3.getToken() & 4194304) === 4194304 ? (a |= e3.assignable & 2 ? 16 : $ === 1077936155 ? 0 : 32, k = be(e3, t, u, i, l, E, k)) : e3.getToken() === 18 || e3.getToken() === 1074790415 ? $ === 1077936155 || $ === 1074790415 || $ === 18 ? e3.assignable & 2 ? a |= 16 : (Y & 143360) === 143360 && n?.addVarOrBlock(t, ce, f, c) : a |= e3.assignable & 1 ? 32 : 16 : (a |= 16, k = I(e3, t, u, i, l, E, k));
        } else (e3.getToken() & 2097152) === 2097152 ? (k = e3.getToken() === 69271571 ? X(e3, t, n, u, 0, i, l, f, c) : j(e3, t, n, u, 0, i, l, f, c), a = e3.destructible, e3.assignable = a & 16 ? 2 : 1, e3.getToken() === 18 || e3.getToken() === 1074790415 ? e3.assignable & 2 && (a |= 16) : a & 8 ? e3.report(62) : (k = F(e3, t, u, k, i, 0, E), a = e3.assignable & 2 ? a | 16 : 0, (e3.getToken() & 4194304) === 4194304 ? (e3.getToken() !== 1077936155 && (a |= 16), k = be(e3, t, u, i, l, E, k)) : ((e3.getToken() & 8388608) === 8388608 && (k = G(e3, t, u, 1, E, 4, T, k)), C(e3, t | 32, 22) && (k = ee(e3, t, u, k, E)), a |= e3.assignable & 2 ? 16 : 32))) : (k = _(e3, t, u, 1, 0, 1), a |= e3.assignable & 1 ? 32 : 16, e3.getToken() === 18 || e3.getToken() === 1074790415 ? e3.assignable & 2 && (a |= 16) : (k = F(e3, t, u, k, i, 0, E), a = e3.assignable & 1 ? 0 : 16, e3.getToken() !== 18 && e3.getToken() !== 1074790415 && (e3.getToken() !== 1077936155 && (a |= 16), k = I(e3, t, u, i, l, E, k))));
      } else e3.getToken() === 67174411 ? (D |= 1, k = z(e3, t, u, D, i, e3.tokenStart), a = 16) : e3.report(44);
      else if (T === 8391476) if (y(e3, t | 32, 8391476), D |= 8, e3.getToken() & 143360) {
        let E = e3.getToken();
        if (R = N(e3, t), D |= 1, e3.getToken() === 67174411) a |= 16, k = z(e3, t, u, D, i, e3.tokenStart);
        else throw new q(e3.tokenStart, e3.currentLocation, E === 209005 ? 46 : E === 209008 || e3.getToken() === 209009 ? 45 : 47, B[E & 255]);
      } else (e3.getToken() & 134217728) === 134217728 ? (a |= 16, R = O(e3, t), D |= 1, k = z(e3, t, u, D, i, e3.tokenStart)) : e3.getToken() === 69271571 ? (a |= 16, D |= 3, R = ne(e3, t, u, i), k = z(e3, t, u, D, i, e3.tokenStart)) : e3.report(126);
      else e3.report(30, B[T & 255]);
      a |= e3.destructible & 128 ? 128 : 0, e3.destructible = a, d.push(e3.finishNode({ type: "Property", key: R, value: k, kind: D & 768 ? D & 512 ? "set" : "get" : "init", computed: (D & 2) > 0, method: (D & 1) > 0, shorthand: (D & 4) > 0 }, w));
    }
    if (a |= e3.destructible, e3.getToken() !== 18) break;
    r(e3, t);
  }
  y(e3, t, 1074790415), h > 1 && (a |= 64);
  let A = e3.finishNode({ type: l ? "ObjectPattern" : "ObjectExpression", properties: d }, g);
  return !o && e3.getToken() & 4194304 ? Kt(e3, t, u, a, i, l, g, A) : (e3.destructible = a, A);
}
function Qn(e3, t, n, u, o, i, l) {
  y(e3, t, 67174411);
  let f = [];
  if (e3.flags = (e3.flags | 128) ^ 128, e3.getToken() === 16) return o & 512 && e3.report(37, "Setter", "one", ""), r(e3, t), f;
  o & 256 && e3.report(37, "Getter", "no", "s"), o & 512 && e3.getToken() === 14 && e3.report(38), t = (t | 131072) ^ 131072;
  let c = 0, g = 0;
  for (; e3.getToken() !== 18; ) {
    let d = null, { tokenStart: a } = e3;
    if (e3.getToken() & 143360 ? ((t & 1) === 0 && ((e3.getToken() & 36864) === 36864 && (e3.flags |= 256), (e3.getToken() & 537079808) === 537079808 && (e3.flags |= 512)), d = it(e3, t, n, o | 1, 0)) : (e3.getToken() === 2162700 ? d = j(e3, t, n, u, 1, l, 1, i, 0) : e3.getToken() === 69271571 ? d = X(e3, t, n, u, 1, l, 1, i, 0) : e3.getToken() === 14 && (d = oe(e3, t, n, u, 16, i, 0, 0, l, 1)), g = 1, e3.destructible & 48 && e3.report(50)), e3.getToken() === 1077936155) {
      r(e3, t | 32), g = 1;
      let h = L(e3, t, u, 1, 0, e3.tokenStart);
      d = e3.finishNode({ type: "AssignmentPattern", left: d, right: h }, a);
    }
    if (c++, f.push(d), !C(e3, t, 18) || e3.getToken() === 16) break;
  }
  return o & 512 && c !== 1 && e3.report(37, "Setter", "one", ""), n?.reportScopeError(), g && (e3.flags |= 128), y(e3, t, 16), f;
}
function ne(e3, t, n, u) {
  r(e3, t | 32);
  let o = L(e3, (t | 131072) ^ 131072, n, 1, u, e3.tokenStart);
  return y(e3, t, 20), o;
}
function Zn(e3, t, n, u, o, i, l) {
  e3.flags = (e3.flags | 128) ^ 128;
  let f = e3.tokenStart;
  r(e3, t | 32 | 262144);
  let c = e3.createScopeIfLexical()?.createChildScope(512);
  if (t = (t | 131072) ^ 131072, C(e3, t, 16)) return Ne(e3, t, c, n, [], u, 0, l);
  let g = 0;
  e3.destructible &= -385;
  let d, a = [], h = 0, A = 0, b = 0, w = e3.tokenStart;
  for (e3.assignable = 1; e3.getToken() !== 16; ) {
    let { tokenStart: T } = e3, D = e3.getToken();
    if (D & 143360) c?.addBlockName(t, e3.tokenValue, 1, 0), (D & 537079808) === 537079808 ? A = 1 : (D & 36864) === 36864 && (b = 1), d = U(e3, t, n, o, 0, 1, 1, 1, T), e3.getToken() === 16 || e3.getToken() === 18 ? e3.assignable & 2 && (g |= 16, A = 1) : (e3.getToken() === 1077936155 ? A = 1 : g |= 16, d = F(e3, t, n, d, 1, 0, T), e3.getToken() !== 16 && e3.getToken() !== 18 && (d = I(e3, t, n, 1, 0, T, d)));
    else if ((D & 2097152) === 2097152) d = D === 2162700 ? j(e3, t | 262144, c, n, 0, 1, 0, o, i) : X(e3, t | 262144, c, n, 0, 1, 0, o, i), g |= e3.destructible, A = 1, e3.assignable = 2, e3.getToken() !== 16 && e3.getToken() !== 18 && (g & 8 && e3.report(122), d = F(e3, t, n, d, 0, 0, T), g |= 16, e3.getToken() !== 16 && e3.getToken() !== 18 && (d = I(e3, t, n, 0, 0, T, d)));
    else if (D === 14) {
      d = oe(e3, t, c, n, 16, o, i, 0, 1, 0), e3.destructible & 16 && e3.report(74), A = 1, h && (e3.getToken() === 16 || e3.getToken() === 18) && a.push(d), g |= 8;
      break;
    } else {
      if (g |= 16, d = L(e3, t, n, 1, 1, T), h && (e3.getToken() === 16 || e3.getToken() === 18) && a.push(d), e3.getToken() === 18 && (h || (h = 1, a = [d])), h) {
        for (; C(e3, t | 32, 18); ) a.push(L(e3, t, n, 1, 1, e3.tokenStart));
        e3.assignable = 2, d = e3.finishNode({ type: "SequenceExpression", expressions: a }, w);
      }
      return y(e3, t, 16), e3.destructible = g, e3.options.preserveParens ? e3.finishNode({ type: "ParenthesizedExpression", expression: d }, f) : d;
    }
    if (h && (e3.getToken() === 16 || e3.getToken() === 18) && a.push(d), !C(e3, t | 32, 18)) break;
    if (h || (h = 1, a = [d]), e3.getToken() === 16) {
      g |= 8;
      break;
    }
  }
  return h && (e3.assignable = 2, d = e3.finishNode({ type: "SequenceExpression", expressions: a }, w)), y(e3, t, 16), g & 16 && g & 8 && e3.report(151), g |= e3.destructible & 256 ? 256 : 0 | e3.destructible & 128 ? 128 : 0, e3.getToken() === 10 ? (g & 48 && e3.report(49), t & 2050 && g & 128 && e3.report(31), t & 1025 && g & 256 && e3.report(32), A && (e3.flags |= 128), b && (e3.flags |= 256), Ne(e3, t, c, n, h ? a : [d], u, 0, l)) : (g & 64 && e3.report(63), g & 8 && e3.report(144), e3.destructible = (e3.destructible | 256) ^ 256 | g, e3.options.preserveParens ? e3.finishNode({ type: "ParenthesizedExpression", expression: d }, f) : d);
}
function nt(e3, t, n) {
  let { tokenStart: u } = e3, { tokenValue: o } = e3, i = 0, l = 0;
  (e3.getToken() & 537079808) === 537079808 ? i = 1 : (e3.getToken() & 36864) === 36864 && (l = 1);
  let f = N(e3, t);
  if (e3.assignable = 1, e3.getToken() === 10) {
    let c = e3.options.lexical ? qe(e3, t, o) : void 0;
    return i && (e3.flags |= 128), l && (e3.flags |= 256), ke(e3, t, c, n, [f], 0, u);
  }
  return f;
}
function Oe(e3, t, n, u, o, i, l, f, c) {
  l || e3.report(57), i && e3.report(51), e3.flags &= -129;
  let g = e3.options.lexical ? qe(e3, t, u) : void 0;
  return ke(e3, t, g, n, [o], f, c);
}
function Ne(e3, t, n, u, o, i, l, f) {
  i || e3.report(57);
  for (let c = 0; c < o.length; ++c) K(e3, o[c]);
  return ke(e3, t, n, u, o, l, f);
}
function ke(e3, t, n, u, o, i, l) {
  e3.flags & 1 && e3.report(48), y(e3, t | 32, 10);
  let f = 535552;
  t = (t | f) ^ f | (i ? 2048 : 0);
  let c = e3.getToken() !== 2162700, g;
  if (n?.reportScopeError(), c) e3.flags = (e3.flags | 512 | 256 | 64 | 4096) ^ 4928, g = L(e3, t, u, 1, 0, e3.tokenStart);
  else {
    n = n?.createChildScope(64);
    let d = 131084;
    switch (g = Pe(e3, (t | d) ^ d | 4096, n, u, 16, void 0, void 0), e3.getToken()) {
      case 69271571:
        (e3.flags & 1) === 0 && e3.report(116);
        break;
      case 67108877:
      case 67174409:
      case 22:
        e3.report(117);
      case 67174411:
        (e3.flags & 1) === 0 && e3.report(116), e3.flags |= 1024;
        break;
    }
    (e3.getToken() & 8388608) === 8388608 && (e3.flags & 1) === 0 && e3.report(30, B[e3.getToken() & 255]), (e3.getToken() & 33619968) === 33619968 && e3.report(125);
  }
  return e3.assignable = 2, e3.finishNode({ type: "ArrowFunctionExpression", params: o, body: g, async: i === 1, expression: c, generator: !1 }, l);
}
function $t(e3, t, n, u, o, i) {
  y(e3, t, 67174411), e3.flags = (e3.flags | 128) ^ 128;
  let l = [];
  if (C(e3, t, 16)) return l;
  t = (t | 131072) ^ 131072;
  let f = 0;
  for (; e3.getToken() !== 18; ) {
    let c, { tokenStart: g } = e3, d = e3.getToken();
    if (d & 143360 ? ((t & 1) === 0 && ((d & 36864) === 36864 && (e3.flags |= 256), (d & 537079808) === 537079808 && (e3.flags |= 512)), c = it(e3, t, n, i | 1, 0)) : (d === 2162700 ? c = j(e3, t, n, u, 1, o, 1, i, 0) : d === 69271571 ? c = X(e3, t, n, u, 1, o, 1, i, 0) : d === 14 ? c = oe(e3, t, n, u, 16, i, 0, 0, o, 1) : e3.report(30, B[d & 255]), f = 1, e3.destructible & 48 && e3.report(50)), e3.getToken() === 1077936155) {
      r(e3, t | 32), f = 1;
      let a = L(e3, t, u, 1, o, e3.tokenStart);
      c = e3.finishNode({ type: "AssignmentPattern", left: c, right: a }, g);
    }
    if (l.push(c), !C(e3, t, 18) || e3.getToken() === 16) break;
  }
  return f && (e3.flags |= 128), (f || t & 1) && n?.reportScopeError(), y(e3, t, 16), l;
}
function we(e3, t, n, u, o, i) {
  let l = e3.getToken();
  if (l & 67108864) {
    if (l === 67108877) {
      r(e3, t | 262144), e3.assignable = 1;
      let f = xe(e3, t, n);
      return we(e3, t, n, e3.finishNode({ type: "MemberExpression", object: u, computed: !1, property: f, optional: !1 }, i), 0, i);
    } else if (l === 69271571) {
      r(e3, t | 32);
      let { tokenStart: f } = e3, c = V(e3, t, n, o, 1, f);
      return y(e3, t, 20), e3.assignable = 1, we(e3, t, n, e3.finishNode({ type: "MemberExpression", object: u, computed: !0, property: c, optional: !1 }, i), 0, i);
    } else if (l === 67174408 || l === 67174409) return e3.assignable = 2, we(e3, t, n, e3.finishNode({ type: "TaggedTemplateExpression", tag: u, quasi: e3.getToken() === 67174408 ? et(e3, t | 64, n) : pe(e3, t | 64) }, i), 0, i);
  }
  return u;
}
function Gn(e3, t, n, u) {
  let { tokenStart: o } = e3, i = N(e3, t | 32), { tokenStart: l } = e3;
  if (C(e3, t, 67108877)) {
    if (t & 65536 && e3.getToken() === 209029) return e3.assignable = 2, xn(e3, t, i, o);
    e3.report(94);
  }
  e3.assignable = 2, (e3.getToken() & 16842752) === 16842752 && e3.report(65, B[e3.getToken() & 255]);
  let f = U(e3, t, n, 2, 1, 0, u, 1, l);
  t = (t | 131072) ^ 131072, e3.getToken() === 67108990 && e3.report(168);
  let c = we(e3, t, n, f, u, l);
  return e3.assignable = 2, e3.finishNode({ type: "NewExpression", callee: c, arguments: e3.getToken() === 67174411 ? tt(e3, t, n, u) : [] }, o);
}
function xn(e3, t, n, u) {
  let o = N(e3, t);
  return e3.finishNode({ type: "MetaProperty", meta: n, property: o }, u);
}
function Wt(e3, t, n, u, o) {
  return e3.getToken() === 209006 && e3.report(31), t & 1025 && e3.getToken() === 241771 && e3.report(32), Ie(e3, t, e3.getToken()), (e3.getToken() & 36864) === 36864 && (e3.flags |= 256), Oe(e3, t & -524289 | 2048, n, e3.tokenValue, N(e3, t), 0, u, 1, o);
}
function ut(e3, t, n, u, o, i, l, f, c) {
  r(e3, t | 32);
  let g = e3.createScopeIfLexical()?.createChildScope(512);
  if (t = (t | 131072) ^ 131072, C(e3, t, 16)) return e3.getToken() === 10 ? (f & 1 && e3.report(48), Ne(e3, t, g, n, [], o, 1, c)) : e3.finishNode({ type: "CallExpression", callee: u, arguments: [], optional: !1 }, c);
  let d = 0, a = null, h = 0;
  e3.destructible = (e3.destructible | 256 | 128) ^ 384;
  let A = [];
  for (; e3.getToken() !== 16; ) {
    let { tokenStart: b } = e3, w = e3.getToken();
    if (w & 143360) g?.addBlockName(t, e3.tokenValue, i, 0), (w & 537079808) === 537079808 ? e3.flags |= 512 : (w & 36864) === 36864 && (e3.flags |= 256), a = U(e3, t, n, i, 0, 1, 1, 1, b), e3.getToken() === 16 || e3.getToken() === 18 ? e3.assignable & 2 && (d |= 16, h = 1) : (e3.getToken() === 1077936155 ? h = 1 : d |= 16, a = F(e3, t, n, a, 1, 0, b), e3.getToken() !== 16 && e3.getToken() !== 18 && (a = I(e3, t, n, 1, 0, b, a)));
    else if (w & 2097152) a = w === 2162700 ? j(e3, t, g, n, 0, 1, 0, i, l) : X(e3, t, g, n, 0, 1, 0, i, l), d |= e3.destructible, h = 1, e3.getToken() !== 16 && e3.getToken() !== 18 && (d & 8 && e3.report(122), a = F(e3, t, n, a, 0, 0, b), d |= 16, (e3.getToken() & 8388608) === 8388608 && (a = G(e3, t, n, 1, c, 4, w, a)), C(e3, t | 32, 22) && (a = ee(e3, t, n, a, c)));
    else if (w === 14) a = oe(e3, t, g, n, 16, i, l, 1, 1, 0), d |= (e3.getToken() === 16 ? 0 : 16) | e3.destructible, h = 1;
    else {
      for (a = L(e3, t, n, 1, 0, b), d = e3.assignable, A.push(a); C(e3, t | 32, 18); ) A.push(L(e3, t, n, 1, 0, b));
      return d |= e3.assignable, y(e3, t, 16), e3.destructible = d | 16, e3.assignable = 2, e3.finishNode({ type: "CallExpression", callee: u, arguments: A, optional: !1 }, c);
    }
    if (A.push(a), !C(e3, t | 32, 18)) break;
  }
  return y(e3, t, 16), d |= e3.destructible & 256 ? 256 : 0 | e3.destructible & 128 ? 128 : 0, e3.getToken() === 10 ? (d & 48 && e3.report(27), (e3.flags & 1 || f & 1) && e3.report(48), d & 128 && e3.report(31), t & 1025 && d & 256 && e3.report(32), h && (e3.flags |= 128), Ne(e3, t | 2048, g, n, A, o, 1, c)) : (d & 64 && e3.report(63), d & 8 && e3.report(62), e3.assignable = 2, e3.finishNode({ type: "CallExpression", callee: u, arguments: A, optional: !1 }, c));
}
function pn(e3, t) {
  let { tokenRaw: n, tokenRegExp: u, tokenValue: o, tokenStart: i } = e3;
  r(e3, t), e3.assignable = 2;
  let l = { type: "Literal", value: o, regex: u };
  return e3.options.raw && (l.raw = n), e3.finishNode(l, i);
}
function ze(e3, t, n, u, o) {
  let i, l;
  e3.leadingDecorators.decorators.length ? (e3.getToken() === 132 && e3.report(30, "@"), i = e3.leadingDecorators.start, l = [...e3.leadingDecorators.decorators], e3.leadingDecorators.decorators.length = 0) : (i = e3.tokenStart, l = Ve(e3, t, u)), t = (t | 16384 | 1) ^ 16384, r(e3, t);
  let f = null, c = null, { tokenValue: g } = e3;
  e3.getToken() & 4096 && e3.getToken() !== 20565 ? (Vt(e3, t, e3.getToken()) && e3.report(118), (e3.getToken() & 537079808) === 537079808 && e3.report(119), n && (n.addBlockName(t, g, 32, 0), o && o & 2 && e3.declareUnboundVariable(g)), f = N(e3, t)) : (o & 1) === 0 && e3.report(39, "Class");
  let d = t;
  C(e3, t | 32, 20565) ? (c = _(e3, t, u, 0, 0, 0), d |= 512) : d = (d | 512) ^ 512;
  let a = Yt(e3, d, t, n, u, 2, 8, 0);
  return e3.finishNode({ type: "ClassDeclaration", id: f, superClass: c, body: a, ...e3.options.next ? { decorators: l } : null }, i);
}
function eu(e3, t, n, u, o) {
  let i = null, l = null, f = Ve(e3, t, n);
  t = (t | 1 | 16384) ^ 16384, r(e3, t), e3.getToken() & 4096 && e3.getToken() !== 20565 && (Vt(e3, t, e3.getToken()) && e3.report(118), (e3.getToken() & 537079808) === 537079808 && e3.report(119), i = N(e3, t));
  let c = t;
  C(e3, t | 32, 20565) ? (l = _(e3, t, n, 0, u, 0), c |= 512) : c = (c | 512) ^ 512;
  let g = Yt(e3, c, t, void 0, n, 2, 0, u);
  return e3.assignable = 2, e3.finishNode({ type: "ClassExpression", id: i, superClass: l, body: g, ...e3.options.next ? { decorators: f } : null }, o);
}
function Ve(e3, t, n) {
  let u = [];
  if (e3.options.next) for (; e3.getToken() === 132; ) u.push(tu(e3, t, n));
  return u;
}
function tu(e3, t, n) {
  let u = e3.tokenStart;
  r(e3, t | 32);
  let o = e3.tokenStart, i = U(e3, t, n, 2, 0, 1, 0, 1, u);
  return i = F(e3, t, n, i, 0, 0, o), e3.finishNode({ type: "Decorator", expression: i }, u);
}
function Yt(e3, t, n, u, o, i, l, f) {
  let { tokenStart: c } = e3, g = e3.createPrivateScopeIfLexical(o);
  y(e3, t | 32, 2162700);
  let d = 655360;
  t = (t | d) ^ d;
  let a = e3.flags & 32;
  e3.flags = (e3.flags | 32) ^ 32;
  let h = [];
  for (; e3.getToken() !== 1074790415; ) {
    let A = e3.tokenStart, b = Ve(e3, t, g);
    if (b.length > 0 && e3.tokenValue === "constructor" && e3.report(109), e3.getToken() === 1074790415 && e3.report(108), C(e3, t, 1074790417)) {
      b.length > 0 && e3.report(120);
      continue;
    }
    h.push(Qt(e3, t, u, g, n, i, b, 0, f, b.length > 0 ? A : e3.tokenStart));
  }
  return y(e3, l & 8 ? t | 32 : t, 1074790415), g?.validatePrivateIdentifierRefs(), e3.flags = e3.flags & -33 | a, e3.finishNode({ type: "ClassBody", body: h }, c);
}
function Qt(e3, t, n, u, o, i, l, f, c, g) {
  let d = f ? 32 : 0, a = null, h = e3.getToken();
  if (h & 176128 || h === -2147483528) switch (a = N(e3, t), h) {
    case 36970:
      if (!f && e3.getToken() !== 67174411 && (e3.getToken() & 1048576) !== 1048576 && e3.getToken() !== 1077936155) return Qt(e3, t, n, u, o, i, l, 1, c, g);
      break;
    case 209005:
      if (e3.getToken() !== 67174411 && (e3.flags & 1) === 0) {
        if ((e3.getToken() & 1073741824) === 1073741824) return se(e3, t, u, a, d, l, g);
        d |= 16 | (Qe(e3, t, 8391476) ? 8 : 0);
      }
      break;
    case 209008:
      if (e3.getToken() !== 67174411) {
        if ((e3.getToken() & 1073741824) === 1073741824) return se(e3, t, u, a, d, l, g);
        d |= 256;
      }
      break;
    case 209009:
      if (e3.getToken() !== 67174411) {
        if ((e3.getToken() & 1073741824) === 1073741824) return se(e3, t, u, a, d, l, g);
        d |= 512;
      }
      break;
    case 12402:
      if (e3.getToken() !== 67174411 && (e3.flags & 1) === 0) {
        if ((e3.getToken() & 1073741824) === 1073741824) return se(e3, t, u, a, d, l, g);
        e3.options.next && (d |= 1024);
      }
      break;
  }
  else if (h === 69271571) d |= 2, a = ne(e3, o, u, c);
  else if ((h & 134217728) === 134217728) a = O(e3, t);
  else if (h === 8391476) d |= 8, r(e3, t);
  else if (e3.getToken() === 130) d |= 8192, a = Le(e3, t | 16, u, 768);
  else if ((e3.getToken() & 1073741824) === 1073741824) d |= 128;
  else {
    if (f && h === 2162700) return Bn(e3, t | 16, n, u, g);
    h === -2147483527 ? (a = N(e3, t), e3.getToken() !== 67174411 && e3.report(30, B[e3.getToken() & 255])) : e3.report(30, B[e3.getToken() & 255]);
  }
  if (d & 1816 && (e3.getToken() & 143360 || e3.getToken() === -2147483528 || e3.getToken() === -2147483527 ? a = N(e3, t) : (e3.getToken() & 134217728) === 134217728 ? a = O(e3, t) : e3.getToken() === 69271571 ? (d |= 2, a = ne(e3, t, u, 0)) : e3.getToken() === 130 ? (d |= 8192, a = Le(e3, t, u, d)) : e3.report(135)), (d & 2) === 0 && (e3.tokenValue === "constructor" ? ((e3.getToken() & 1073741824) === 1073741824 ? e3.report(129) : (d & 32) === 0 && e3.getToken() === 67174411 && (d & 920 ? e3.report(53, "accessor") : (t & 512) === 0 && (e3.flags & 32 ? e3.report(54) : e3.flags |= 32)), d |= 64) : (d & 8192) === 0 && d & 32 && e3.tokenValue === "prototype" && e3.report(52)), d & 1024 || e3.getToken() !== 67174411 && (d & 768) === 0) return se(e3, t, u, a, d, l, g);
  let A = z(e3, t | 16, u, d, c, e3.tokenStart);
  return e3.finishNode({ type: "MethodDefinition", kind: (d & 32) === 0 && d & 64 ? "constructor" : d & 256 ? "get" : d & 512 ? "set" : "method", static: (d & 32) > 0, computed: (d & 2) > 0, key: a, value: A, ...e3.options.next ? { decorators: l } : null }, g);
}
function Le(e3, t, n, u) {
  let { tokenStart: o } = e3;
  r(e3, t);
  let { tokenValue: i } = e3;
  return i === "constructor" && e3.report(128), e3.options.lexical && (n || e3.report(4, i), u ? n.addPrivateIdentifier(i, u) : n.addPrivateIdentifierRef(i)), r(e3, t), e3.finishNode({ type: "PrivateIdentifier", name: i }, o);
}
function se(e3, t, n, u, o, i, l) {
  let f = null;
  if (o & 8 && e3.report(0), e3.getToken() === 1077936155) {
    r(e3, t | 32);
    let { tokenStart: c } = e3;
    e3.getToken() === 537079927 && e3.report(119);
    let g = 11264 | ((o & 64) === 0 ? 16896 : 0);
    t = (t | g) ^ g | (o & 8 ? 1024 : 0) | (o & 16 ? 2048 : 0) | (o & 64 ? 16384 : 0) | 256 | 65536, f = U(e3, t | 16, n, 2, 0, 1, 0, 1, c), ((e3.getToken() & 1073741824) !== 1073741824 || (e3.getToken() & 4194304) === 4194304) && (f = F(e3, t | 16, n, f, 0, 0, c), f = I(e3, t | 16, n, 0, 0, c, f));
  }
  return M(e3, t), e3.finishNode({ type: o & 1024 ? "AccessorProperty" : "PropertyDefinition", key: u, value: f, static: (o & 32) > 0, computed: (o & 2) > 0, ...e3.options.next ? { decorators: i } : null }, l);
}
function Zt(e3, t, n, u, o, i) {
  if (e3.getToken() & 143360 || (t & 1) === 0 && e3.getToken() === -2147483527) return it(e3, t, n, o, i);
  (e3.getToken() & 2097152) !== 2097152 && e3.report(30, B[e3.getToken() & 255]);
  let l = e3.getToken() === 69271571 ? X(e3, t, n, u, 1, 0, 1, o, i) : j(e3, t, n, u, 1, 0, 1, o, i);
  return e3.destructible & 16 && e3.report(50), e3.destructible & 32 && e3.report(50), l;
}
function it(e3, t, n, u, o) {
  let i = e3.getToken();
  t & 1 && ((i & 537079808) === 537079808 ? e3.report(119) : ((i & 36864) === 36864 || i === -2147483527) && e3.report(118)), (i & 20480) === 20480 && e3.report(102), i === 241771 && (t & 1024 && e3.report(32), t & 2 && e3.report(111)), (i & 255) === 73 && u & 24 && e3.report(100), i === 209006 && (t & 2048 && e3.report(176), t & 2 && e3.report(110));
  let { tokenValue: l, tokenStart: f } = e3;
  return r(e3, t), n?.addVarOrBlock(t, l, u, o), e3.finishNode({ type: "Identifier", name: l }, f);
}
function Re(e3, t, n, u, o) {
  if (u || y(e3, t, 8456256), e3.getToken() === 8390721) {
    let c = nu(e3, o), [g, d] = lu(e3, t, n, u);
    return e3.finishNode({ type: "JSXFragment", openingFragment: c, children: g, closingFragment: d }, o);
  }
  e3.getToken() === 8457014 && e3.report(30, B[e3.getToken() & 255]);
  let i = null, l = [], f = su(e3, t, n, u, o);
  if (!f.selfClosing) {
    [l, i] = ou(e3, t, n, u);
    let c = Fe(i.name);
    Fe(f.name) !== c && e3.report(155, c);
  }
  return e3.finishNode({ type: "JSXElement", children: l, openingElement: f, closingElement: i }, o);
}
function nu(e3, t) {
  return me(e3), e3.finishNode({ type: "JSXOpeningFragment" }, t);
}
function uu(e3, t, n, u) {
  y(e3, t, 8457014);
  let o = xt(e3, t);
  return e3.getToken() !== 8390721 && e3.report(25, B[65]), n ? me(e3) : r(e3, t), e3.finishNode({ type: "JSXClosingElement", name: o }, u);
}
function iu(e3, t, n, u) {
  return y(e3, t, 8457014), e3.getToken() !== 8390721 && e3.report(25, B[65]), n ? me(e3) : r(e3, t), e3.finishNode({ type: "JSXClosingFragment" }, u);
}
function ou(e3, t, n, u) {
  let o = [];
  for (; ; ) {
    let i = fu(e3, t, n, u);
    if (i.type === "JSXClosingElement") return [o, i];
    o.push(i);
  }
}
function lu(e3, t, n, u) {
  let o = [];
  for (; ; ) {
    let i = cu(e3, t, n, u);
    if (i.type === "JSXClosingFragment") return [o, i];
    o.push(i);
  }
}
function fu(e3, t, n, u) {
  if (e3.getToken() === 137) return Gt(e3, t);
  if (e3.getToken() === 2162700) return ot(e3, t, n, 1, 0);
  if (e3.getToken() === 8456256) {
    let { tokenStart: o } = e3;
    return r(e3, t), e3.getToken() === 8457014 ? uu(e3, t, u, o) : Re(e3, t, n, 1, o);
  }
  e3.report(0);
}
function cu(e3, t, n, u) {
  if (e3.getToken() === 137) return Gt(e3, t);
  if (e3.getToken() === 2162700) return ot(e3, t, n, 1, 0);
  if (e3.getToken() === 8456256) {
    let { tokenStart: o } = e3;
    return r(e3, t), e3.getToken() === 8457014 ? iu(e3, t, u, o) : Re(e3, t, n, 1, o);
  }
  e3.report(0);
}
function Gt(e3, t) {
  let n = e3.tokenStart;
  r(e3, t);
  let u = { type: "JSXText", value: e3.tokenValue };
  return e3.options.raw && (u.raw = e3.tokenRaw), e3.finishNode(u, n);
}
function su(e3, t, n, u, o) {
  (e3.getToken() & 143360) !== 143360 && (e3.getToken() & 4096) !== 4096 && e3.report(0);
  let i = xt(e3, t), l = au(e3, t, n), f = e3.getToken() === 8457014;
  return f && y(e3, t, 8457014), e3.getToken() !== 8390721 && e3.report(25, B[65]), u || !f ? me(e3) : r(e3, t), e3.finishNode({ type: "JSXOpeningElement", name: i, attributes: l, selfClosing: f }, o);
}
function xt(e3, t) {
  let { tokenStart: n } = e3;
  Ue(e3);
  let u = Me(e3, t);
  if (e3.getToken() === 21) return pt(e3, t, u, n);
  for (; C(e3, t, 67108877); ) Ue(e3), u = du(e3, t, u, n);
  return u;
}
function du(e3, t, n, u) {
  let o = Me(e3, t);
  return e3.finishNode({ type: "JSXMemberExpression", object: n, property: o }, u);
}
function au(e3, t, n) {
  let u = [];
  for (; e3.getToken() !== 8457014 && e3.getToken() !== 8390721 && e3.getToken() !== 1048576; ) u.push(mu(e3, t, n));
  return u;
}
function gu(e3, t, n) {
  let u = e3.tokenStart;
  r(e3, t), y(e3, t, 14);
  let o = L(e3, t, n, 1, 0, e3.tokenStart);
  return y(e3, t, 1074790415), e3.finishNode({ type: "JSXSpreadAttribute", argument: o }, u);
}
function mu(e3, t, n) {
  let { tokenStart: u } = e3;
  if (e3.getToken() === 2162700) return gu(e3, t, n);
  Ue(e3);
  let o = null, i = Me(e3, t);
  if (e3.getToken() === 21 && (i = pt(e3, t, i, u)), e3.getToken() === 1077936155) switch (un(e3, t)) {
    case 134283267:
      o = O(e3, t);
      break;
    case 8456256:
      o = Re(e3, t, n, 0, e3.tokenStart);
      break;
    case 2162700:
      o = ot(e3, t, n, 0, 1);
      break;
    default:
      e3.report(154);
  }
  return e3.finishNode({ type: "JSXAttribute", value: o, name: i }, u);
}
function pt(e3, t, n, u) {
  y(e3, t, 21);
  let o = Me(e3, t);
  return e3.finishNode({ type: "JSXNamespacedName", namespace: n, name: o }, u);
}
function ot(e3, t, n, u, o) {
  let { tokenStart: i } = e3;
  r(e3, t | 32);
  let { tokenStart: l } = e3;
  if (e3.getToken() === 14) return ru(e3, t, n, i);
  let f = null;
  return e3.getToken() === 1074790415 ? (o && e3.report(157), f = hu(e3, { index: e3.startIndex, line: e3.startLine, column: e3.startColumn })) : f = L(e3, t, n, 1, 0, l), e3.getToken() !== 1074790415 && e3.report(25, B[15]), u ? me(e3) : r(e3, t), e3.finishNode({ type: "JSXExpressionContainer", expression: f }, i);
}
function ru(e3, t, n, u) {
  y(e3, t, 14);
  let o = L(e3, t, n, 1, 0, e3.tokenStart);
  return y(e3, t, 1074790415), e3.finishNode({ type: "JSXSpreadChild", expression: o }, u);
}
function hu(e3, t) {
  return e3.finishNode({ type: "JSXEmptyExpression" }, t, e3.tokenStart);
}
function Me(e3, t) {
  let n = e3.tokenStart;
  e3.getToken() & 143360 || e3.report(30, B[e3.getToken() & 255]);
  let { tokenValue: u } = e3;
  return r(e3, t), e3.finishNode({ type: "JSXIdentifier", name: u }, n);
}
function e2(e3, t) {
  return sn(e3, t);
}
function ku(e3, t) {
  let n = new SyntaxError(e3 + " (" + t.loc.start.line + ":" + t.loc.start.column + ")");
  return Object.assign(n, t);
}
function n2(e3) {
  let t = [];
  for (let n of e3) try {
    return n();
  } catch (u) {
    t.push(u);
  }
  throw Object.assign(new Error("All combinations failed"), { errors: t });
}
function Tu(e3) {
  return this[e3 < 0 ? this.length + e3 : e3];
}
function H(e3) {
  let t = e3.range?.[0] ?? e3.start, n = (e3.declaration?.decorators ?? e3.decorators)?.[0];
  return n ? Math.min(H(n), t) : t;
}
function J(e3) {
  return e3.range?.[1] ?? e3.end;
}
function Du(e3) {
  let t = new Set(e3);
  return (n) => t.has(n?.type);
}
function wu(e3) {
  return lt.has(e3) || lt.set(e3, fe(e3) && e3.value[0] === "*" && /@(?:type|satisfies)\b/u.test(e3.value)), lt.get(e3);
}
function Su(e3) {
  if (!fe(e3)) return !1;
  let t = `*${e3.value}*`.split(`
`);
  return t.length > 1 && t.every((n) => n.trimStart()[0] === "*");
}
function Bu(e3) {
  return ft.has(e3) || ft.set(e3, Su(e3)), ft.get(e3);
}
function Fu(e3) {
  if (e3.length < 2) return;
  let t;
  for (let n = e3.length - 1; n >= 0; n--) {
    let u = e3[n];
    if (t && J(u) === H(t) && ct(u) && ct(t) && (e3.splice(n + 1, 1), u.value += "*//*" + t.value, u.range = [H(u), J(t)]), !o2(u) && !fe(u)) throw new TypeError(`Unknown comment type: "${u.type}".`);
    t = u;
  }
}
function Nu(e3) {
  return e3 !== null && typeof e3 == "object";
}
function Ae(e3) {
  if (ye !== null && typeof ye.property) {
    let t = ye;
    return ye = Ae.prototype = null, t;
  }
  return ye = Ae.prototype = e3 ?? /* @__PURE__ */ Object.create(null), new Ae();
}
function st(e3) {
  return Ae(e3);
}
function Iu(e3, t = "type") {
  st(e3);
  function n(u) {
    let o = u[t], i = e3[o];
    if (!Array.isArray(i)) throw Object.assign(new Error(`Missing visitor keys for '${o}'.`), { node: u });
    return i;
  }
  return n;
}
function ve(e3, t) {
  if (!c2(e3)) return e3;
  if (Array.isArray(e3)) {
    for (let u = 0; u < e3.length; u++) e3[u] = ve(e3[u], t);
    return e3;
  }
  if (t.onEnter) {
    let u = t.onEnter(e3) ?? e3;
    if (u !== e3) return ve(u, t);
    e3 = u;
  }
  let n = a2(e3);
  for (let u = 0; u < n.length; u++) e3[n[u]] = ve(e3[n[u]], t);
  return t.onLeave && (e3 = t.onLeave(e3) || e3), e3;
}
function Pu(e3, t) {
  let { parser: n, text: u } = t, { comments: o } = e3, i = n === "oxc" && t.oxcAstType === "ts";
  f2(o);
  let l = e3.type === "File" ? e3.program : e3;
  l.interpreter && (o.unshift(l.interpreter), delete l.interpreter), i && e3.hashbang && (o.unshift(e3.hashbang), delete e3.hashbang), e3.type === "Program" && (e3.range = [0, u.length]);
  let f;
  return e3 = g2(e3, { onEnter(c) {
    switch (c.type) {
      case "ParenthesizedExpression": {
        let { expression: g } = c, d = H(c);
        if (g.type === "TypeCastExpression") return g.range = [d, J(c)], g;
        let a = !1;
        if (!i) {
          if (!f) {
            f = [];
            for (let A of o) l2(A) && f.push(J(A));
          }
          let h = u2(0, f, (A) => A <= d);
          a = h && u.slice(h, d).trim().length === 0;
        }
        return a ? void 0 : (g.extra = { ...g.extra, parenthesized: !0 }, g);
      }
      case "TemplateLiteral":
        if (c.expressions.length !== c.quasis.length - 1) throw new Error("Malformed template literal.");
        break;
      case "TemplateElement":
        if (n === "flow" || n === "hermes" || n === "espree" || n === "typescript" || i) {
          let g = H(c) + 1, d = J(c) - (c.tail ? 1 : 2);
          c.range = [g, d];
        }
        break;
      case "VariableDeclaration": {
        let g = i2(0, c.declarations, -1);
        g?.init && u[J(g)] !== ";" && (c.range = [H(c), J(g)]);
        break;
      }
      case "TSParenthesizedType":
        return c.typeAnnotation;
      case "TopicReference":
        e3.extra = { ...e3.extra, __isUsingHackPipeline: !0 };
        break;
      case "TSUnionType":
      case "TSIntersectionType":
        if (c.types.length === 1) return c.types[0];
        break;
      case "ImportExpression":
        n === "hermes" && c.attributes && !c.options && (c.options = c.attributes);
        break;
    }
  }, onLeave(c) {
    switch (c.type) {
      case "LogicalExpression":
        if (m2(c)) return dt(c);
        break;
      case "TSImportType":
        !c.source && c.argument.type === "TSLiteralType" && (c.source = c.argument.literal, delete c.argument);
        break;
    }
  } }), e3;
}
function m2(e3) {
  return e3.type === "LogicalExpression" && e3.right.type === "LogicalExpression" && e3.operator === e3.right.operator;
}
function dt(e3) {
  return m2(e3) ? dt({ type: "LogicalExpression", operator: e3.operator, left: dt({ type: "LogicalExpression", operator: e3.operator, left: e3.left, right: e3.right.left, range: [H(e3.left), J(e3.right.left)] }), right: e3.right.right, range: [H(e3), J(e3)] }) : e3;
}
function y2(e3) {
  let t = e3.match(Ru);
  return t ? t[0].trimStart() : "";
}
function A2(e3) {
  e3 = p(0, e3.replace(Vu, "").replace(Ou, ""), Uu, "$1");
  let n = "";
  for (; n !== e3; ) n = e3, e3 = p(0, e3, vu, `
$1 $2
`);
  e3 = e3.replace(h2, "").trimEnd();
  let u = /* @__PURE__ */ Object.create(null), o = p(0, e3, k2, "").replace(h2, "").trimEnd(), i;
  for (; i = k2.exec(e3); ) {
    let l = p(0, i[2], Mu, "");
    if (typeof u[i[1]] == "string" || Array.isArray(u[i[1]])) {
      let f = u[i[1]];
      u[i[1]] = [...Ju, ...Array.isArray(f) ? f : [f], l];
    } else u[i[1]] = l;
  }
  return { comments: o, pragmas: u };
}
function _u(e3) {
  if (!e3.startsWith("#!")) return "";
  let t = e3.indexOf(`
`);
  return t === -1 ? e3 : e3.slice(0, t);
}
function C2(e3) {
  let t = D2(e3);
  t && (e3 = e3.slice(t.length + 1));
  let n = y2(e3), { pragmas: u, comments: o } = A2(n);
  return { shebang: t, text: e3, pragmas: u, comments: o };
}
function E2(e3) {
  let { pragmas: t } = C2(e3);
  return b2.some((n) => Object.prototype.hasOwnProperty.call(t, n));
}
function w2(e3) {
  let { pragmas: t } = C2(e3);
  return T2.some((n) => Object.prototype.hasOwnProperty.call(t, n));
}
function Xu(e3) {
  return e3 = typeof e3 == "function" ? { parse: e3 } : e3, { astFormat: "estree", hasPragma: E2, hasIgnorePragma: w2, locStart: H, locEnd: J, ...e3 };
}
function L2(e3) {
  if (typeof e3 == "string") {
    if (e3 = e3.toLowerCase(), /\.(?:mjs|mts)$/iu.test(e3)) return B2;
    if (/\.(?:cjs|cts)$/iu.test(e3)) return F2;
  }
}
function Hu(e3, t) {
  let n = [], u = e2(e3, { ...ju, sourceType: t, onComment: n });
  return u.comments = n, u;
}
function zu(e3) {
  let { description: t, loc: n } = e3;
  return n ? t2(t, { loc: { start: { line: n.start.line, column: n.start.column + 1 }, end: { line: n.end.line, column: n.end.column + 1 } }, cause: e3 }) : e3;
}
function Ku(e3, t) {
  let n = L2(t?.filepath), u = (n ? [n] : N2).map((i) => () => Hu(e3, i)), o;
  try {
    o = n2(u);
  } catch ({ errors: [i] }) {
    throw zu(i);
  }
  return r2(o, { parser: "meriyah", text: e3 });
}
var q2, gt, I2, at, Q, P2, O2, p, V2, R2, M2, v2, U2, mt, Dt, J2, Ct, S, _2, Et, wt, K2, q, B, Bt, Z, P, en, Je, _e, Xe, t2, yu, Au, u2, bu, i2, le, Cu, fe, Eu, o2, lt, l2, ft, ct, f2, c2, ye, Lu, s2, s, d2, qu, a2, g2, Xi, r2, Ou, Vu, Ru, Mu, h2, vu, k2, Uu, Ju, T2, b2, D2, S2, B2, F2, N2, ju, $u, init_meriyah = __esm({
  "../../node_modules/.pnpm/prettier@3.7.4/node_modules/prettier/plugins/meriyah.mjs"() {
    init_cjs_shims();
    q2 = Object.defineProperty, gt = (e3, t) => {
      for (var n in t) q2(e3, n, { get: t[n], enumerable: !0 });
    }, I2 = {};
    gt(I2, { parsers: () => at });
    at = {};
    gt(at, { meriyah: () => $u });
    Q = (e3, t) => (n, u, ...o) => n | 1 && u == null ? void 0 : (t.call(u) ?? u[e3]).apply(u, o), P2 = String.prototype.replaceAll ?? function(e3, t) {
      return e3.global ? this.replace(e3, t) : this.split(e3).join(t);
    }, O2 = Q("replaceAll", function() {
      if (typeof this == "string") return P2;
    }), p = O2, V2 = 55296, R2 = 56319, M2 = 56320, v2 = String.prototype.isWellFormed ?? function() {
      let { length: e3 } = this;
      for (let t = 0; t < e3; t++) {
        let n = this.charCodeAt(t);
        if ((n & 64512) === V2 && (n > R2 || ++t >= e3 || (this.charCodeAt(t) & 64512) !== M2)) return !1;
      }
      return !0;
    }, U2 = Q("isWellFormed", function() {
      if (typeof this == "string") return v2;
    }), mt = U2, Dt = ((e3, t) => {
      let n = new Uint32Array(69632), u = 0, o = 0;
      for (; u < 2597; ) {
        let i = e3[u++];
        if (i < 0) o -= i;
        else {
          let l = e3[u++];
          i & 2 && (l = t[l]), i & 1 ? n.fill(l, o, o += e3[u++]) : n[o++] = l;
        }
      }
      return n;
    })([-1, 2, 26, 2, 27, 2, 5, -1, 0, 77595648, 3, 44, 2, 3, 0, 14, 2, 61, 2, 62, 3, 0, 3, 0, 3168796671, 0, 4294956992, 2, 1, 2, 0, 2, 41, 3, 0, 4, 0, 4294966523, 3, 0, 4, 2, 16, 2, 63, 2, 0, 0, 4294836735, 0, 3221225471, 0, 4294901942, 2, 64, 0, 134152192, 3, 0, 2, 0, 4294951935, 3, 0, 2, 0, 2683305983, 0, 2684354047, 2, 17, 2, 0, 0, 4294961151, 3, 0, 2, 2, 19, 2, 0, 0, 608174079, 2, 0, 2, 58, 2, 7, 2, 6, 0, 4286643967, 3, 0, 2, 2, 1, 3, 0, 3, 0, 4294901711, 2, 40, 0, 4089839103, 0, 2961209759, 0, 1342439375, 0, 4294543342, 0, 3547201023, 0, 1577204103, 0, 4194240, 0, 4294688750, 2, 2, 0, 80831, 0, 4261478351, 0, 4294549486, 2, 2, 0, 2967484831, 0, 196559, 0, 3594373100, 0, 3288319768, 0, 8469959, 0, 65472, 2, 3, 0, 4093640191, 0, 929054175, 0, 65487, 0, 4294828015, 0, 4092591615, 0, 1885355487, 0, 982991, 2, 3, 2, 0, 0, 2163244511, 0, 4227923919, 0, 4236247022, 2, 69, 0, 4284449919, 0, 851904, 2, 4, 2, 12, 0, 67076095, -1, 2, 70, 0, 1073741743, 0, 4093607775, -1, 0, 50331649, 0, 3265266687, 2, 33, 0, 4294844415, 0, 4278190047, 2, 20, 2, 137, -1, 3, 0, 2, 2, 23, 2, 0, 2, 9, 2, 0, 2, 15, 2, 22, 3, 0, 10, 2, 72, 2, 0, 2, 73, 2, 74, 2, 75, 2, 0, 2, 76, 2, 0, 2, 11, 0, 261632, 2, 25, 3, 0, 2, 2, 13, 2, 4, 3, 0, 18, 2, 77, 2, 5, 3, 0, 2, 2, 78, 0, 2151677951, 2, 29, 2, 10, 0, 909311, 3, 0, 2, 0, 814743551, 2, 48, 0, 67090432, 3, 0, 2, 2, 42, 2, 0, 2, 6, 2, 0, 2, 30, 2, 8, 0, 268374015, 2, 108, 2, 51, 2, 0, 2, 79, 0, 134153215, -1, 2, 7, 2, 0, 2, 8, 0, 2684354559, 0, 67044351, 0, 3221160064, 2, 9, 2, 18, 3, 0, 2, 2, 53, 0, 1046528, 3, 0, 3, 2, 10, 2, 0, 2, 127, 0, 4294960127, 2, 9, 2, 6, 2, 11, 0, 4294377472, 2, 12, 3, 0, 16, 2, 13, 2, 0, 2, 80, 2, 9, 2, 0, 2, 81, 2, 82, 2, 83, 0, 12288, 2, 54, 0, 1048577, 2, 84, 2, 14, -1, 2, 14, 0, 131042, 2, 85, 2, 86, 2, 87, 2, 0, 2, 34, -83, 3, 0, 7, 0, 1046559, 2, 0, 2, 15, 2, 0, 0, 2147516671, 2, 21, 3, 88, 2, 2, 0, -16, 2, 89, 0, 524222462, 2, 4, 2, 0, 0, 4269801471, 2, 4, 3, 0, 2, 2, 28, 2, 16, 3, 0, 2, 2, 49, 2, 0, -1, 2, 17, -16, 3, 0, 206, -2, 3, 0, 692, 2, 71, -1, 2, 17, 2, 9, 3, 0, 8, 2, 91, 2, 18, 2, 0, 0, 3220242431, 3, 0, 3, 2, 19, 2, 92, 2, 93, 3, 0, 2, 2, 94, 2, 0, 2, 20, 2, 95, 2, 0, 0, 4351, 2, 0, 2, 10, 3, 0, 2, 0, 67043391, 0, 3909091327, 2, 0, 2, 24, 2, 10, 2, 20, 3, 0, 2, 0, 67076097, 2, 8, 2, 0, 2, 21, 0, 67059711, 0, 4236247039, 3, 0, 2, 0, 939524103, 0, 8191999, 2, 99, 2, 100, 2, 22, 2, 23, 3, 0, 3, 0, 67057663, 3, 0, 349, 2, 101, 2, 102, 2, 7, -264, 3, 0, 11, 2, 24, 3, 0, 2, 2, 32, -1, 0, 3774349439, 2, 103, 2, 104, 3, 0, 2, 2, 19, 2, 105, 3, 0, 10, 2, 9, 2, 17, 2, 0, 2, 46, 2, 0, 2, 31, 2, 106, 2, 25, 0, 1638399, 0, 57344, 2, 107, 3, 0, 3, 2, 20, 2, 26, 2, 27, 2, 5, 2, 28, 2, 0, 2, 8, 2, 109, -1, 2, 110, 2, 111, 2, 112, -1, 3, 0, 3, 2, 12, -2, 2, 0, 2, 29, -3, 0, 536870912, -4, 2, 20, 2, 0, 2, 36, 0, 1, 2, 0, 2, 65, 2, 6, 2, 12, 2, 9, 2, 0, 2, 113, -1, 3, 0, 4, 2, 9, 2, 23, 2, 114, 2, 7, 2, 0, 2, 115, 2, 0, 2, 116, 2, 117, 2, 118, 2, 0, 2, 10, 3, 0, 9, 2, 21, 2, 30, 2, 31, 2, 119, 2, 120, -2, 2, 121, 2, 122, 2, 30, 2, 21, 2, 8, -2, 2, 123, 2, 30, 3, 32, 2, -1, 2, 0, 2, 39, -2, 0, 4277137519, 0, 2269118463, -1, 3, 20, 2, -1, 2, 33, 2, 38, 2, 0, 3, 30, 2, 2, 35, 2, 19, -3, 3, 0, 2, 2, 34, -1, 2, 0, 2, 35, 2, 0, 2, 35, 2, 0, 2, 47, 2, 0, 0, 4294950463, 2, 37, -7, 2, 0, 0, 203775, 2, 125, 0, 4227858432, 2, 20, 2, 43, 2, 36, 2, 17, 2, 37, 2, 17, 2, 124, 2, 21, 3, 0, 2, 2, 38, 0, 2151677888, 2, 0, 2, 12, 0, 4294901764, 2, 145, 2, 0, 2, 56, 2, 55, 0, 5242879, 3, 0, 2, 0, 402644511, -1, 2, 128, 2, 39, 0, 3, -1, 2, 129, 2, 130, 2, 0, 0, 67045375, 2, 40, 0, 4226678271, 0, 3766565279, 0, 2039759, 2, 132, 2, 41, 0, 1046437, 0, 6, 3, 0, 2, 0, 3288270847, 0, 3, 3, 0, 2, 0, 67043519, -5, 2, 0, 0, 4282384383, 0, 1056964609, -1, 3, 0, 2, 0, 67043345, -1, 2, 0, 2, 42, 2, 23, 2, 50, 2, 11, 2, 59, 2, 38, -5, 2, 0, 2, 12, -3, 3, 0, 2, 0, 2147484671, 2, 133, 0, 4190109695, 2, 52, -2, 2, 134, 0, 4244635647, 0, 27, 2, 0, 2, 8, 2, 43, 2, 0, 2, 66, 2, 17, 2, 0, 2, 42, -3, 2, 31, -2, 2, 0, 2, 45, 2, 57, 2, 44, 2, 45, 2, 135, 2, 46, 0, 8388351, -2, 2, 136, 0, 3028287487, 2, 47, 2, 138, 0, 33259519, 2, 23, 2, 7, 2, 48, -7, 2, 21, 0, 4294836223, 0, 3355443199, 0, 134152199, -2, 2, 67, -2, 3, 0, 28, 2, 32, -3, 3, 0, 3, 2, 49, 3, 0, 6, 2, 50, -81, 2, 17, 3, 0, 2, 2, 36, 3, 0, 33, 2, 25, 2, 30, 3, 0, 124, 2, 12, 3, 0, 18, 2, 38, -213, 2, 0, 2, 32, -54, 3, 0, 17, 2, 42, 2, 8, 2, 23, 2, 0, 2, 8, 2, 23, 2, 51, 2, 0, 2, 21, 2, 52, 2, 139, 2, 25, -13, 2, 0, 2, 53, -6, 3, 0, 2, -1, 2, 140, 2, 10, -1, 3, 0, 2, 0, 4294936575, 2, 0, 0, 4294934783, -2, 0, 8323099, 3, 0, 230, 2, 30, 2, 54, 2, 8, -3, 3, 0, 3, 2, 35, -271, 2, 141, 3, 0, 9, 2, 142, 2, 143, 2, 55, 3, 0, 11, 2, 7, -72, 3, 0, 3, 2, 144, 0, 1677656575, -130, 2, 26, -16, 2, 0, 2, 24, 2, 38, -16, 0, 4161266656, 0, 4071, 0, 15360, -4, 0, 28, -13, 3, 0, 2, 2, 56, 2, 0, 2, 146, 2, 147, 2, 60, 2, 0, 2, 148, 2, 149, 2, 150, 3, 0, 10, 2, 151, 2, 152, 2, 22, 3, 56, 2, 3, 153, 2, 3, 57, 2, 0, 4294954999, 2, 0, -16, 2, 0, 2, 90, 2, 0, 0, 2105343, 0, 4160749584, 0, 65534, -34, 2, 8, 2, 155, -6, 0, 4194303871, 0, 4294903771, 2, 0, 2, 58, 2, 98, -3, 2, 0, 0, 1073684479, 0, 17407, -9, 2, 17, 2, 49, 2, 0, 2, 32, -14, 2, 17, 2, 32, -6, 2, 17, 2, 12, -6, 2, 8, 0, 3225419775, -7, 2, 156, 3, 0, 6, 0, 8323103, -1, 3, 0, 2, 2, 59, -37, 2, 60, 2, 157, 2, 158, 2, 159, 2, 160, 2, 161, -105, 2, 26, -32, 3, 0, 1335, -1, 3, 0, 136, 2, 9, 3, 0, 180, 2, 24, 3, 0, 233, 2, 162, 3, 0, 18, 2, 9, -77, 3, 0, 16, 2, 9, -47, 3, 0, 154, 2, 6, 3, 0, 264, 2, 32, -22116, 3, 0, 7, 2, 25, -6130, 3, 5, 2, -1, 0, 69207040, 3, 44, 2, 3, 0, 14, 2, 61, 2, 62, -3, 0, 3168731136, 0, 4294956864, 2, 1, 2, 0, 2, 41, 3, 0, 4, 0, 4294966275, 3, 0, 4, 2, 16, 2, 63, 2, 0, 2, 34, -1, 2, 17, 2, 64, -1, 2, 0, 0, 2047, 0, 4294885376, 3, 0, 2, 0, 3145727, 0, 2617294944, 0, 4294770688, 2, 25, 2, 65, 3, 0, 2, 0, 131135, 2, 96, 0, 70256639, 0, 71303167, 0, 272, 2, 42, 2, 6, 0, 65279, 2, 0, 2, 48, -1, 2, 97, 2, 66, 0, 4278255616, 0, 4294836227, 0, 4294549473, 0, 600178175, 0, 2952806400, 0, 268632067, 0, 4294543328, 0, 57540095, 0, 1577058304, 0, 1835008, 0, 4294688736, 2, 68, 2, 67, 0, 33554435, 2, 131, 2, 68, 0, 2952790016, 0, 131075, 0, 3594373096, 0, 67094296, 2, 67, -1, 0, 4294828e3, 0, 603979263, 0, 922746880, 0, 3, 0, 4294828001, 0, 602930687, 0, 1879048192, 0, 393219, 0, 4294828016, 0, 671088639, 0, 2154840064, 0, 4227858435, 0, 4236247008, 2, 69, 2, 38, -1, 2, 4, 0, 917503, 2, 38, -1, 2, 70, 0, 537788335, 0, 4026531935, -1, 0, 1, -1, 2, 33, 2, 71, 0, 7936, -3, 2, 0, 0, 2147485695, 0, 1010761728, 0, 4292984930, 0, 16387, 2, 0, 2, 15, 2, 22, 3, 0, 10, 2, 72, 2, 0, 2, 73, 2, 74, 2, 75, 2, 0, 2, 76, 2, 0, 2, 12, -1, 2, 25, 3, 0, 2, 2, 13, 2, 4, 3, 0, 18, 2, 77, 2, 5, 3, 0, 2, 2, 78, 0, 2147745791, 3, 19, 2, 0, 122879, 2, 0, 2, 10, 0, 276824064, -2, 3, 0, 2, 2, 42, 2, 0, 0, 4294903295, 2, 0, 2, 30, 2, 8, -1, 2, 17, 2, 51, 2, 0, 2, 79, 2, 48, -1, 2, 21, 2, 0, 2, 29, -2, 0, 128, -2, 2, 28, 2, 10, 0, 8160, -1, 2, 126, 0, 4227907585, 2, 0, 2, 37, 2, 0, 2, 50, 0, 4227915776, 2, 9, 2, 6, 2, 11, -1, 0, 74440192, 3, 0, 6, -2, 3, 0, 8, 2, 13, 2, 0, 2, 80, 2, 9, 2, 0, 2, 81, 2, 82, 2, 83, -3, 2, 84, 2, 14, -3, 2, 85, 2, 86, 2, 87, 2, 0, 2, 34, -83, 3, 0, 7, 0, 817183, 2, 0, 2, 15, 2, 0, 0, 33023, 2, 21, 3, 88, 2, -17, 2, 89, 0, 524157950, 2, 4, 2, 0, 2, 90, 2, 4, 2, 0, 2, 22, 2, 28, 2, 16, 3, 0, 2, 2, 49, 2, 0, -1, 2, 17, -16, 3, 0, 206, -2, 3, 0, 692, 2, 71, -1, 2, 17, 2, 9, 3, 0, 8, 2, 91, 0, 3072, 2, 0, 0, 2147516415, 2, 9, 3, 0, 2, 2, 25, 2, 92, 2, 93, 3, 0, 2, 2, 94, 2, 0, 2, 20, 2, 95, 0, 4294965179, 0, 7, 2, 0, 2, 10, 2, 93, 2, 10, -1, 0, 1761345536, 2, 96, 0, 4294901823, 2, 38, 2, 20, 2, 97, 2, 35, 2, 98, 0, 2080440287, 2, 0, 2, 34, 2, 154, 0, 3296722943, 2, 0, 0, 1046675455, 0, 939524101, 0, 1837055, 2, 99, 2, 100, 2, 22, 2, 23, 3, 0, 3, 0, 7, 3, 0, 349, 2, 101, 2, 102, 2, 7, -264, 3, 0, 11, 2, 24, 3, 0, 2, 2, 32, -1, 0, 2700607615, 2, 103, 2, 104, 3, 0, 2, 2, 19, 2, 105, 3, 0, 10, 2, 9, 2, 17, 2, 0, 2, 46, 2, 0, 2, 31, 2, 106, -3, 2, 107, 3, 0, 3, 2, 20, -1, 3, 5, 2, 2, 108, 2, 0, 2, 8, 2, 109, -1, 2, 110, 2, 111, 2, 112, -1, 3, 0, 3, 2, 12, -2, 2, 0, 2, 29, -8, 2, 20, 2, 0, 2, 36, -1, 2, 0, 2, 65, 2, 6, 2, 30, 2, 9, 2, 0, 2, 113, -1, 3, 0, 4, 2, 9, 2, 17, 2, 114, 2, 7, 2, 0, 2, 115, 2, 0, 2, 116, 2, 117, 2, 118, 2, 0, 2, 10, 3, 0, 9, 2, 21, 2, 30, 2, 31, 2, 119, 2, 120, -2, 2, 121, 2, 122, 2, 30, 2, 21, 2, 8, -2, 2, 123, 2, 30, 3, 32, 2, -1, 2, 0, 2, 39, -2, 0, 4277075969, 2, 30, -1, 3, 20, 2, -1, 2, 33, 2, 124, 2, 0, 3, 30, 2, 2, 35, 2, 19, -3, 3, 0, 2, 2, 34, -1, 2, 0, 2, 35, 2, 0, 2, 35, 2, 0, 2, 50, 2, 96, 0, 4294934591, 2, 37, -7, 2, 0, 0, 197631, 2, 125, -1, 2, 20, 2, 43, 2, 37, 2, 17, 0, 3, 2, 17, 2, 124, 2, 21, 2, 126, 2, 127, -1, 0, 2490368, 2, 126, 2, 25, 2, 17, 2, 34, 2, 126, 2, 38, 0, 4294901904, 0, 4718591, 2, 126, 2, 35, 0, 335544350, -1, 2, 128, 0, 2147487743, 0, 1, -1, 2, 129, 2, 130, 2, 8, -1, 2, 131, 2, 68, 0, 3758161920, 0, 3, 2, 132, 0, 12582911, 0, 655360, -1, 2, 0, 2, 29, 0, 2147485568, 0, 3, 2, 0, 2, 25, 0, 176, -5, 2, 0, 2, 49, 0, 251658240, -1, 2, 0, 2, 25, 0, 16, -1, 2, 0, 0, 16779263, -2, 2, 12, -1, 2, 38, -5, 2, 0, 2, 18, -3, 3, 0, 2, 2, 54, 2, 133, 0, 2147549183, 0, 2, -2, 2, 134, 2, 36, 0, 10, 0, 4294965249, 0, 67633151, 0, 4026597376, 2, 0, 0, 536871935, 2, 17, 2, 0, 2, 42, -6, 2, 0, 0, 1, 2, 57, 2, 49, 0, 1, 2, 135, 2, 25, -3, 2, 136, 2, 36, 2, 137, 2, 138, 0, 16778239, 2, 17, 2, 7, -8, 2, 35, 0, 4294836212, 2, 10, -3, 2, 67, -2, 3, 0, 28, 2, 32, -3, 3, 0, 3, 2, 49, 3, 0, 6, 2, 50, -81, 2, 17, 3, 0, 2, 2, 36, 3, 0, 33, 2, 25, 0, 126, 3, 0, 124, 2, 12, 3, 0, 18, 2, 38, -213, 2, 9, -55, 3, 0, 17, 2, 42, 2, 8, 2, 17, 2, 0, 2, 8, 2, 17, 2, 58, 2, 0, 2, 25, 2, 50, 2, 139, 2, 25, -13, 2, 0, 2, 71, -6, 3, 0, 2, -1, 2, 140, 2, 10, -1, 3, 0, 2, 0, 67583, -1, 2, 105, -2, 0, 8126475, 3, 0, 230, 2, 30, 2, 54, 2, 8, -3, 3, 0, 3, 2, 35, -271, 2, 141, 3, 0, 9, 2, 142, 2, 143, 2, 55, 3, 0, 11, 2, 7, -72, 3, 0, 3, 2, 144, 2, 145, -187, 3, 0, 2, 2, 56, 2, 0, 2, 146, 2, 147, 2, 60, 2, 0, 2, 148, 2, 149, 2, 150, 3, 0, 10, 2, 151, 2, 152, 2, 22, 3, 56, 2, 3, 153, 2, 3, 57, 2, 2, 154, -57, 2, 8, 2, 155, -7, 2, 17, 2, 0, 2, 58, -4, 2, 0, 0, 1065361407, 0, 16384, -9, 2, 17, 2, 58, 2, 0, 2, 18, -14, 2, 17, 2, 18, -6, 2, 17, 0, 81919, -6, 2, 8, 0, 3223273399, -7, 2, 156, 3, 0, 6, 2, 124, -1, 3, 0, 2, 0, 2063, -37, 2, 60, 2, 157, 2, 158, 2, 159, 2, 160, 2, 161, -138, 3, 0, 1335, -1, 3, 0, 136, 2, 9, 3, 0, 180, 2, 24, 3, 0, 233, 2, 162, 3, 0, 18, 2, 9, -77, 3, 0, 16, 2, 9, -47, 3, 0, 154, 2, 6, 3, 0, 264, 2, 32, -28252], [4294967295, 4294967291, 4092460543, 4294828031, 4294967294, 134217726, 4294903807, 268435455, 2147483647, 1073741823, 1048575, 3892314111, 134217727, 1061158911, 536805376, 4294910143, 4294901759, 4294901760, 4095, 262143, 536870911, 8388607, 4160749567, 4294902783, 4294918143, 65535, 67043328, 2281701374, 4294967264, 2097151, 4194303, 255, 67108863, 4294967039, 511, 524287, 131071, 63, 127, 3238002687, 4294549487, 4290772991, 33554431, 4294901888, 4286578687, 67043329, 4294770687, 67043583, 1023, 32767, 15, 2047999, 67043343, 67051519, 2147483648, 4294902e3, 4292870143, 4294966783, 16383, 67047423, 4294967279, 262083, 20511, 41943039, 493567, 4294959104, 603979775, 65536, 602799615, 805044223, 4294965206, 8191, 1031749119, 4294917631, 2134769663, 4286578493, 4282253311, 4294942719, 33540095, 4294905855, 2868854591, 1608515583, 265232348, 534519807, 2147614720, 1060109444, 4093640016, 17376, 2139062143, 224, 4169138175, 4294909951, 4286578688, 4294967292, 4294965759, 4294836224, 4294966272, 4294967280, 32768, 8289918, 4294934399, 4294901775, 4294965375, 1602223615, 4294967259, 4294443008, 268369920, 4292804608, 4294967232, 486341884, 4294963199, 3087007615, 1073692671, 4128527, 4279238655, 4294902015, 4160684047, 4290246655, 469499899, 4294967231, 134086655, 4294966591, 2445279231, 3670015, 31, 252, 4294967288, 16777215, 4294705151, 3221208447, 4294902271, 4294549472, 4294921215, 4285526655, 4294966527, 4294705152, 4294966143, 64, 4294966719, 3774873592, 4194303999, 1877934080, 262151, 2555904, 536807423, 67043839, 3758096383, 3959414372, 3755993023, 2080374783, 4294835295, 4294967103, 4160749565, 4294934527, 4087, 2016, 2147446655, 184024726, 2862017156, 1593309078, 268434431, 268434414, 4294901761]), J2 = (e3) => (Dt[(e3 >>> 5) + 0] >>> e3 & 31 & 1) !== 0, Ct = (e3) => (Dt[(e3 >>> 5) + 34816] >>> e3 & 31 & 1) !== 0, S = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1032, 0, 0, 2056, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 8192, 0, 3, 0, 0, 8192, 0, 0, 0, 256, 0, 33024, 0, 0, 242, 242, 114, 114, 114, 114, 114, 114, 594, 594, 0, 0, 16384, 0, 0, 0, 0, 67, 67, 67, 67, 67, 67, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 0, 1, 0, 0, 4099, 0, 71, 71, 71, 71, 71, 71, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 16384, 0, 0, 0, 0], _2 = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0], Et = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0];
    wt = ["SingleLine", "MultiLine", "HTMLOpen", "HTMLClose", "HashbangComment"];
    K2 = { 0: "Unexpected token", 30: "Unexpected token: '%0'", 1: "Octal escape sequences are not allowed in strict mode", 2: "Octal escape sequences are not allowed in template strings", 3: "\\8 and \\9 are not allowed in template strings", 4: "Private identifier #%0 is not defined", 5: "Illegal Unicode escape sequence", 6: "Invalid code point %0", 7: "Invalid hexadecimal escape sequence", 9: "Octal literals are not allowed in strict mode", 8: "Decimal integer literals with a leading zero are forbidden in strict mode", 10: "Expected number in radix %0", 151: "Invalid left-hand side assignment to a destructible right-hand side", 11: "Non-number found after exponent indicator", 12: "Invalid BigIntLiteral", 13: "No identifiers allowed directly after numeric literal", 14: "Escapes \\8 or \\9 are not syntactically valid escapes", 15: "Escapes \\8 or \\9 are not allowed in strict mode", 16: "Unterminated string literal", 17: "Unterminated template literal", 18: "Multiline comment was not closed properly", 19: "The identifier contained dynamic unicode escape that was not closed", 20: "Illegal character '%0'", 21: "Missing hexadecimal digits", 22: "Invalid implicit octal", 23: "Invalid line break in string literal", 24: "Only unicode escapes are legal in identifier names", 25: "Expected '%0'", 26: "Invalid left-hand side in assignment", 27: "Invalid left-hand side in async arrow", 28: 'Calls to super must be in the "constructor" method of a class expression or class declaration that has a superclass', 29: "Member access on super must be in a method", 31: "Await expression not allowed in formal parameter", 32: "Yield expression not allowed in formal parameter", 95: "Unexpected token: 'escaped keyword'", 33: "Unary expressions as the left operand of an exponentiation expression must be disambiguated with parentheses", 123: "Async functions can only be declared at the top level or inside a block", 34: "Unterminated regular expression", 35: "Unexpected regular expression flag", 36: "Duplicate regular expression flag '%0'", 37: "%0 functions must have exactly %1 argument%2", 38: "Setter function argument must not be a rest parameter", 39: "%0 declaration must have a name in this context", 40: "Function name may not contain any reserved words or be eval or arguments in strict mode", 41: "The rest operator is missing an argument", 42: "A getter cannot be a generator", 43: "A setter cannot be a generator", 44: "A computed property name must be followed by a colon or paren", 134: "Object literal keys that are strings or numbers must be a method or have a colon", 46: "Found `* async x(){}` but this should be `async * x(){}`", 45: "Getters and setters can not be generators", 47: "'%0' can not be generator method", 48: "No line break is allowed after '=>'", 49: "The left-hand side of the arrow can only be destructed through assignment", 50: "The binding declaration is not destructible", 51: "Async arrow can not be followed by new expression", 52: "Classes may not have a static property named 'prototype'", 53: "Class constructor may not be a %0", 54: "Duplicate constructor method in class", 55: "Invalid increment/decrement operand", 56: "Invalid use of `new` keyword on an increment/decrement expression", 57: "`=>` is an invalid assignment target", 58: "Rest element may not have a trailing comma", 59: "Missing initializer in %0 declaration", 60: "'for-%0' loop head declarations can not have an initializer", 61: "Invalid left-hand side in for-%0 loop: Must have a single binding", 62: "Invalid shorthand property initializer", 63: "Property name __proto__ appears more than once in object literal", 64: "Let is disallowed as a lexically bound name", 65: "Invalid use of '%0' inside new expression", 66: "Illegal 'use strict' directive in function with non-simple parameter list", 67: 'Identifier "let" disallowed as left-hand side expression in strict mode', 68: "Illegal continue statement", 69: "Illegal break statement", 70: "Cannot have `let[...]` as a var name in strict mode", 71: "Invalid destructuring assignment target", 72: "Rest parameter may not have a default initializer", 73: "The rest argument must the be last parameter", 74: "Invalid rest argument", 76: "In strict mode code, functions can only be declared at top level or inside a block", 77: "In non-strict mode code, functions can only be declared at top level, inside a block, or as the body of an if statement", 78: "Without web compatibility enabled functions can not be declared at top level, inside a block, or as the body of an if statement", 79: "Class declaration can't appear in single-statement context", 80: "Invalid left-hand side in for-%0", 81: "Invalid assignment in for-%0", 82: "for await (... of ...) is only valid in async functions and async generators", 83: "The first token after the template expression should be a continuation of the template", 85: "`let` declaration not allowed here and `let` cannot be a regular var name in strict mode", 84: "`let \n [` is a restricted production at the start of a statement", 86: "Catch clause requires exactly one parameter, not more (and no trailing comma)", 87: "Catch clause parameter does not support default values", 88: "Missing catch or finally after try", 89: "More than one default clause in switch statement", 90: "Illegal newline after throw", 91: "Strict mode code may not include a with statement", 92: "Illegal return statement", 93: "The left hand side of the for-header binding declaration is not destructible", 94: "new.target only allowed within functions or static blocks", 96: "'#' not followed by identifier", 102: "Invalid keyword", 101: "Can not use 'let' as a class name", 100: "'A lexical declaration can't define a 'let' binding", 99: "Can not use `let` as variable name in strict mode", 97: "'%0' may not be used as an identifier in this context", 98: "Await is only valid in async functions", 103: "The %0 keyword can only be used with the module goal", 104: "Unicode codepoint must not be greater than 0x10FFFF", 105: "%0 source must be string", 106: "Only a identifier or string can be used to indicate alias", 107: "Only '*' or '{...}' can be imported after default", 108: "Trailing decorator may be followed by method", 109: "Decorators can't be used with a constructor", 110: "Can not use `await` as identifier in module or async func", 111: "Can not use `await` as identifier in module", 112: "HTML comments are only allowed with web compatibility (Annex B)", 113: "The identifier 'let' must not be in expression position in strict mode", 114: "Cannot assign to `eval` and `arguments` in strict mode", 115: "The left-hand side of a for-of loop may not start with 'let'", 116: "Block body arrows can not be immediately invoked without a group", 117: "Block body arrows can not be immediately accessed without a group", 118: "Unexpected strict mode reserved word", 119: "Unexpected eval or arguments in strict mode", 120: "Decorators must not be followed by a semicolon", 121: "Calling delete on expression not allowed in strict mode", 122: "Pattern can not have a tail", 124: "Can not have a `yield` expression on the left side of a ternary", 125: "An arrow function can not have a postfix update operator", 126: "Invalid object literal key character after generator star", 127: "Private fields can not be deleted", 129: "Classes may not have a field called constructor", 128: "Classes may not have a private element named constructor", 130: "A class field initializer or static block may not contain arguments", 131: "Generators can only be declared at the top level or inside a block", 132: "Async methods are a restricted production and cannot have a newline following it", 133: "Unexpected character after object literal property name", 135: "Invalid key token", 136: "Label '%0' has already been declared", 137: "continue statement must be nested within an iteration statement", 138: "Undefined label '%0'", 139: "Trailing comma is disallowed inside import(...) arguments", 140: "Invalid binding in JSON import", 141: "import() requires exactly one argument", 142: "Cannot use new with import(...)", 143: "... is not allowed in import()", 144: "Expected '=>'", 145: "Duplicate binding '%0'", 146: "Duplicate private identifier #%0", 147: "Cannot export a duplicate name '%0'", 150: "Duplicate %0 for-binding", 148: "Exported binding '%0' needs to refer to a top-level declared variable", 149: "Unexpected private field", 153: "Numeric separators are not allowed at the end of numeric literals", 152: "Only one underscore is allowed as numeric separator", 154: "JSX value should be either an expression or a quoted JSX text", 155: "Expected corresponding JSX closing tag for %0", 156: "Adjacent JSX elements must be wrapped in an enclosing tag", 157: "JSX attributes must only be assigned a non-empty 'expression'", 158: "'%0' has already been declared", 159: "'%0' shadowed a catch clause binding", 160: "Dot property must be an identifier", 161: "Encountered invalid input after spread/rest argument", 162: "Catch without try", 163: "Finally without try", 164: "Expected corresponding closing tag for JSX fragment", 165: "Coalescing and logical operators used together in the same expression must be disambiguated with parentheses", 166: "Invalid tagged template on optional chain", 167: "Invalid optional chain from super property", 168: "Invalid optional chain from new expression", 169: 'Cannot use "import.meta" outside a module', 170: "Leading decorators must be attached to a class declaration", 171: "An export name cannot include a lone surrogate", 172: "A string literal cannot be used as an exported binding without `from`", 173: "Private fields can't be accessed on super", 174: "The only valid meta property for import is 'import.meta'", 175: "'import.meta' must not contain escaped characters", 176: 'cannot use "await" as identifier inside an async function', 177: 'cannot use "await" in static blocks' }, q = class extends SyntaxError {
      start;
      end;
      range;
      loc;
      description;
      constructor(t, n, u, ...o) {
        let i = K2[u].replace(/%(\d+)/g, (f, c) => o[c]), l = "[" + t.line + ":" + t.column + "-" + n.line + ":" + n.column + "]: " + i;
        super(l), this.start = t.index, this.end = n.index, this.range = [t.index, n.index], this.loc = { start: { line: t.line, column: t.column }, end: { line: n.line, column: n.column } }, this.description = i;
      }
    };
    B = ["end of source", "identifier", "number", "string", "regular expression", "false", "true", "null", "template continuation", "template tail", "=>", "(", "{", ".", "...", "}", ")", ";", ",", "[", "]", ":", "?", "'", '"', "++", "--", "=", "<<=", ">>=", ">>>=", "**=", "+=", "-=", "*=", "/=", "%=", "^=", "|=", "&=", "||=", "&&=", "??=", "typeof", "delete", "void", "!", "~", "+", "-", "in", "instanceof", "*", "%", "/", "**", "&&", "||", "===", "!==", "==", "!=", "<=", ">=", "<", ">", "<<", ">>", ">>>", "&", "|", "^", "var", "let", "const", "break", "case", "catch", "class", "continue", "debugger", "default", "do", "else", "export", "extends", "finally", "for", "function", "if", "import", "new", "return", "super", "switch", "this", "throw", "try", "while", "with", "implements", "interface", "package", "private", "protected", "public", "static", "yield", "as", "async", "await", "constructor", "get", "set", "accessor", "from", "of", "enum", "eval", "arguments", "escaped keyword", "escaped future reserved keyword", "reserved if strict", "#", "BigIntLiteral", "??", "?.", "WhiteSpace", "Illegal", "LineTerminator", "PrivateField", "Template", "@", "target", "meta", "LineFeed", "Escaped", "JSXText"], Bt = { this: 86111, function: 86104, if: 20569, return: 20572, var: 86088, else: 20563, for: 20567, new: 86107, in: 8673330, typeof: 16863275, while: 20578, case: 20556, break: 20555, try: 20577, catch: 20557, delete: 16863276, throw: 86112, switch: 86110, continue: 20559, default: 20561, instanceof: 8411187, do: 20562, void: 16863277, finally: 20566, async: 209005, await: 209006, class: 86094, const: 86090, constructor: 12399, debugger: 20560, export: 20564, extends: 20565, false: 86021, from: 209011, get: 209008, implements: 36964, import: 86106, interface: 36965, let: 241737, null: 86023, of: 471156, package: 36966, private: 36967, protected: 36968, public: 36969, set: 209009, static: 36970, super: 86109, true: 86022, with: 20579, yield: 241771, enum: 86133, eval: 537079926, as: 77932, arguments: 537079927, target: 209029, meta: 209030, accessor: 12402 };
    (function(e3) {
      e3[e3.Empty = 0] = "Empty", e3[e3.Escape = 1] = "Escape", e3[e3.Class = 2] = "Class";
    })(Z || (Z = {}));
    (function(e3) {
      e3[e3.Empty = 0] = "Empty", e3[e3.IgnoreCase = 1] = "IgnoreCase", e3[e3.Global = 2] = "Global", e3[e3.Multiline = 4] = "Multiline", e3[e3.Unicode = 16] = "Unicode", e3[e3.Sticky = 8] = "Sticky", e3[e3.DotAll = 32] = "DotAll", e3[e3.Indices = 64] = "Indices", e3[e3.UnicodeSets = 128] = "UnicodeSets";
    })(P || (P = {}));
    en = [128, 128, 128, 128, 128, 128, 128, 128, 128, 127, 135, 127, 127, 129, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 127, 16842798, 134283267, 130, 208897, 8391477, 8390213, 134283267, 67174411, 16, 8391476, 25233968, 18, 25233969, 67108877, 8457014, 134283266, 134283266, 134283266, 134283266, 134283266, 134283266, 134283266, 134283266, 134283266, 134283266, 21, 1074790417, 8456256, 1077936155, 8390721, 22, 132, 208897, 208897, 208897, 208897, 208897, 208897, 208897, 208897, 208897, 208897, 208897, 208897, 208897, 208897, 208897, 208897, 208897, 208897, 208897, 208897, 208897, 208897, 208897, 208897, 208897, 208897, 69271571, 136, 20, 8389959, 208897, 131, 4096, 4096, 4096, 4096, 4096, 4096, 4096, 208897, 4096, 208897, 208897, 4096, 208897, 4096, 208897, 4096, 208897, 4096, 4096, 4096, 208897, 4096, 4096, 208897, 4096, 4096, 2162700, 8389702, 1074790415, 16842799, 128];
    Je = class {
      parser;
      parent;
      refs = /* @__PURE__ */ Object.create(null);
      privateIdentifiers = /* @__PURE__ */ new Map();
      constructor(t, n) {
        this.parser = t, this.parent = n;
      }
      addPrivateIdentifier(t, n) {
        let { privateIdentifiers: u } = this, o = n & 800;
        o & 768 || (o |= 768);
        let i = u.get(t);
        this.hasPrivateIdentifier(t) && ((i & 32) !== (o & 32) || i & o & 768) && this.parser.report(146, t), u.set(t, this.hasPrivateIdentifier(t) ? i | o : o);
      }
      addPrivateIdentifierRef(t) {
        var n;
        (n = this.refs)[t] ?? (n[t] = []), this.refs[t].push(this.parser.tokenStart);
      }
      isPrivateIdentifierDefined(t) {
        return this.hasPrivateIdentifier(t) || !!this.parent?.isPrivateIdentifierDefined(t);
      }
      validatePrivateIdentifierRefs() {
        for (let t in this.refs) if (!this.isPrivateIdentifierDefined(t)) {
          let { index: n, line: u, column: o } = this.refs[t][0];
          throw new q({ index: n, line: u, column: o }, { index: n + t.length, line: u, column: o + t.length }, 4, t);
        }
      }
      hasPrivateIdentifier(t) {
        return this.privateIdentifiers.has(t);
      }
    }, _e = class e {
      parser;
      type;
      parent;
      scopeError;
      variableBindings = /* @__PURE__ */ new Map();
      constructor(t, n = 2, u) {
        this.parser = t, this.type = n, this.parent = u;
      }
      createChildScope(t) {
        return new e(this.parser, t, this);
      }
      addVarOrBlock(t, n, u, o) {
        u & 4 ? this.addVarName(t, n, u) : this.addBlockName(t, n, u, o), o & 64 && this.parser.declareUnboundVariable(n);
      }
      addVarName(t, n, u) {
        let { parser: o } = this, i = this;
        for (; i && (i.type & 128) === 0; ) {
          let { variableBindings: l } = i, f = l.get(n);
          f && f & 248 && (o.options.webcompat && (t & 1) === 0 && (u & 128 && f & 68 || f & 128 && u & 68) || o.report(145, n)), i === this && f && f & 1 && u & 1 && i.recordScopeError(145, n), f && (f & 256 || f & 512 && !o.options.webcompat) && o.report(145, n), i.variableBindings.set(n, u), i = i.parent;
        }
      }
      hasVariable(t) {
        return this.variableBindings.has(t);
      }
      addBlockName(t, n, u, o) {
        let { parser: i } = this, l = this.variableBindings.get(n);
        l && (l & 2) === 0 && (u & 1 ? this.recordScopeError(145, n) : i.options.webcompat && (t & 1) === 0 && o & 2 && l === 64 && u === 64 || i.report(145, n)), this.type & 64 && this.parent?.hasVariable(n) && (this.parent.variableBindings.get(n) & 2) === 0 && i.report(145, n), this.type & 512 && l && (l & 2) === 0 && u & 1 && this.recordScopeError(145, n), this.type & 32 && this.parent.variableBindings.get(n) & 768 && i.report(159, n), this.variableBindings.set(n, u);
      }
      recordScopeError(t, ...n) {
        this.scopeError = { type: t, params: n, start: this.parser.tokenStart, end: this.parser.currentLocation };
      }
      reportScopeError() {
        let { scopeError: t } = this;
        if (t) throw new q(t.start, t.end, t.type, ...t.params);
      }
    };
    Xe = class {
      source;
      lastOnToken = null;
      options;
      token = 1048576;
      flags = 0;
      index = 0;
      line = 1;
      column = 0;
      startIndex = 0;
      end = 0;
      tokenIndex = 0;
      startColumn = 0;
      tokenColumn = 0;
      tokenLine = 1;
      startLine = 1;
      tokenValue = "";
      tokenRaw = "";
      tokenRegExp = void 0;
      currentChar = 0;
      exportedNames = /* @__PURE__ */ new Set();
      exportedBindings = /* @__PURE__ */ new Set();
      assignable = 1;
      destructible = 0;
      leadingDecorators = { decorators: [] };
      constructor(t, n = {}) {
        this.source = t, this.end = t.length, this.currentChar = t.charCodeAt(0), this.options = ln(n), Array.isArray(this.options.onComment) && (this.options.onComment = fn(this.options.onComment, this.options)), Array.isArray(this.options.onToken) && (this.options.onToken = cn(this.options.onToken, this.options));
      }
      getToken() {
        return this.token;
      }
      setToken(t, n = !1) {
        this.token = t;
        let { onToken: u } = this.options;
        if (u) if (t !== 1048576) {
          let o = { start: { line: this.tokenLine, column: this.tokenColumn }, end: { line: this.line, column: this.column } };
          !n && this.lastOnToken && u(...this.lastOnToken), this.lastOnToken = [j2(t), this.tokenIndex, this.index, o];
        } else this.lastOnToken && (u(...this.lastOnToken), this.lastOnToken = null);
        return t;
      }
      get tokenStart() {
        return { index: this.tokenIndex, line: this.tokenLine, column: this.tokenColumn };
      }
      get currentLocation() {
        return { index: this.index, line: this.line, column: this.column };
      }
      finishNode(t, n, u) {
        if (this.options.ranges) {
          t.start = n.index;
          let o = u ? u.index : this.startIndex;
          t.end = o, t.range = [n.index, o];
        }
        return this.options.loc && (t.loc = { start: { line: n.line, column: n.column }, end: u ? { line: u.line, column: u.column } : { line: this.startLine, column: this.startColumn } }, this.options.source && (t.loc.source = this.options.source)), t;
      }
      addBindingToExports(t) {
        this.exportedBindings.add(t);
      }
      declareUnboundVariable(t) {
        let { exportedNames: n } = this;
        n.has(t) && this.report(147, t), n.add(t);
      }
      report(t, ...n) {
        throw new q(this.tokenStart, this.currentLocation, t, ...n);
      }
      createScopeIfLexical(t, n) {
        if (this.options.lexical) return this.createScope(t, n);
      }
      createScope(t, n) {
        return new _e(this, t, n);
      }
      createPrivateScopeIfLexical(t) {
        if (this.options.lexical) return new Je(this, t);
      }
      cloneIdentifier(t) {
        return this.cloneLocationInformation({ ...t }, t);
      }
      cloneStringLiteral(t) {
        return this.cloneLocationInformation({ ...t }, t);
      }
      cloneLocationInformation(t, n) {
        return this.options.ranges && (t.range = [...n.range]), this.options.loc && (t.loc = { ...n.loc, start: { ...n.loc.start }, end: { ...n.loc.end } }), t;
      }
    };
    t2 = ku;
    yu = Array.prototype.findLast ?? function(e3) {
      for (let t = this.length - 1; t >= 0; t--) {
        let n = this[t];
        if (e3(n, t, this)) return n;
      }
    }, Au = Q("findLast", function() {
      if (Array.isArray(this)) return yu;
    }), u2 = Au;
    bu = Q("at", function() {
      if (Array.isArray(this) || typeof this == "string") return Tu;
    }), i2 = bu;
    le = Du, Cu = le(["Block", "CommentBlock", "MultiLine"]), fe = Cu, Eu = le(["Line", "CommentLine", "SingleLine", "HashbangComment", "HTMLOpen", "HTMLClose", "Hashbang", "InterpreterDirective"]), o2 = Eu, lt = /* @__PURE__ */ new WeakMap();
    l2 = wu;
    ft = /* @__PURE__ */ new WeakMap();
    ct = Bu;
    f2 = Fu;
    c2 = Nu, ye = null;
    Lu = 10;
    for (let e3 = 0; e3 <= Lu; e3++) Ae();
    s2 = Iu, s = [["decorators", "key", "typeAnnotation", "value"], [], ["elementType"], ["expression"], ["expression", "typeAnnotation"], ["left", "right"], ["argument"], ["directives", "body"], ["label"], ["callee", "typeArguments", "arguments"], ["body"], ["decorators", "id", "typeParameters", "superClass", "superTypeArguments", "mixins", "implements", "body", "superTypeParameters"], ["id", "typeParameters"], ["decorators", "key", "typeParameters", "params", "returnType", "body"], ["decorators", "variance", "key", "typeAnnotation", "value"], ["name", "typeAnnotation"], ["test", "consequent", "alternate"], ["checkType", "extendsType", "trueType", "falseType"], ["value"], ["id", "body"], ["declaration", "specifiers", "source", "attributes"], ["id"], ["id", "typeParameters", "extends", "body"], ["typeAnnotation"], ["id", "typeParameters", "right"], ["body", "test"], ["members"], ["id", "init"], ["exported"], ["left", "right", "body"], ["id", "typeParameters", "params", "predicate", "returnType", "body"], ["id", "params", "body", "typeParameters", "returnType"], ["key", "value"], ["local"], ["objectType", "indexType"], ["typeParameter"], ["types"], ["node"], ["object", "property"], ["argument", "cases"], ["pattern", "body", "guard"], ["literal"], ["decorators", "key", "value"], ["expressions"], ["qualification", "id"], ["decorators", "key", "typeAnnotation"], ["typeParameters", "params", "returnType"], ["expression", "typeArguments"], ["params"], ["parameterName", "typeAnnotation"]], d2 = { AccessorProperty: s[0], AnyTypeAnnotation: s[1], ArgumentPlaceholder: s[1], ArrayExpression: ["elements"], ArrayPattern: ["elements", "typeAnnotation", "decorators"], ArrayTypeAnnotation: s[2], ArrowFunctionExpression: ["typeParameters", "params", "predicate", "returnType", "body"], AsConstExpression: s[3], AsExpression: s[4], AssignmentExpression: s[5], AssignmentPattern: ["left", "right", "decorators", "typeAnnotation"], AwaitExpression: s[6], BigIntLiteral: s[1], BigIntLiteralTypeAnnotation: s[1], BigIntTypeAnnotation: s[1], BinaryExpression: s[5], BindExpression: ["object", "callee"], BlockStatement: s[7], BooleanLiteral: s[1], BooleanLiteralTypeAnnotation: s[1], BooleanTypeAnnotation: s[1], BreakStatement: s[8], CallExpression: s[9], CatchClause: ["param", "body"], ChainExpression: s[3], ClassAccessorProperty: s[0], ClassBody: s[10], ClassDeclaration: s[11], ClassExpression: s[11], ClassImplements: s[12], ClassMethod: s[13], ClassPrivateMethod: s[13], ClassPrivateProperty: s[14], ClassProperty: s[14], ComponentDeclaration: ["id", "params", "body", "typeParameters", "rendersType"], ComponentParameter: ["name", "local"], ComponentTypeAnnotation: ["params", "rest", "typeParameters", "rendersType"], ComponentTypeParameter: s[15], ConditionalExpression: s[16], ConditionalTypeAnnotation: s[17], ContinueStatement: s[8], DebuggerStatement: s[1], DeclareClass: ["id", "typeParameters", "extends", "mixins", "implements", "body"], DeclareComponent: ["id", "params", "rest", "typeParameters", "rendersType"], DeclaredPredicate: s[18], DeclareEnum: s[19], DeclareExportAllDeclaration: ["source", "attributes"], DeclareExportDeclaration: s[20], DeclareFunction: ["id", "predicate"], DeclareHook: s[21], DeclareInterface: s[22], DeclareModule: s[19], DeclareModuleExports: s[23], DeclareNamespace: s[19], DeclareOpaqueType: ["id", "typeParameters", "supertype", "lowerBound", "upperBound"], DeclareTypeAlias: s[24], DeclareVariable: s[21], Decorator: s[3], Directive: s[18], DirectiveLiteral: s[1], DoExpression: s[10], DoWhileStatement: s[25], EmptyStatement: s[1], EmptyTypeAnnotation: s[1], EnumBigIntBody: s[26], EnumBigIntMember: s[27], EnumBooleanBody: s[26], EnumBooleanMember: s[27], EnumDeclaration: s[19], EnumDefaultedMember: s[21], EnumNumberBody: s[26], EnumNumberMember: s[27], EnumStringBody: s[26], EnumStringMember: s[27], EnumSymbolBody: s[26], ExistsTypeAnnotation: s[1], ExperimentalRestProperty: s[6], ExperimentalSpreadProperty: s[6], ExportAllDeclaration: ["source", "attributes", "exported"], ExportDefaultDeclaration: ["declaration"], ExportDefaultSpecifier: s[28], ExportNamedDeclaration: s[20], ExportNamespaceSpecifier: s[28], ExportSpecifier: ["local", "exported"], ExpressionStatement: s[3], File: ["program"], ForInStatement: s[29], ForOfStatement: s[29], ForStatement: ["init", "test", "update", "body"], FunctionDeclaration: s[30], FunctionExpression: s[30], FunctionTypeAnnotation: ["typeParameters", "this", "params", "rest", "returnType"], FunctionTypeParam: s[15], GenericTypeAnnotation: s[12], HookDeclaration: s[31], HookTypeAnnotation: ["params", "returnType", "rest", "typeParameters"], Identifier: ["typeAnnotation", "decorators"], IfStatement: s[16], ImportAttribute: s[32], ImportDeclaration: ["specifiers", "source", "attributes"], ImportDefaultSpecifier: s[33], ImportExpression: ["source", "options"], ImportNamespaceSpecifier: s[33], ImportSpecifier: ["imported", "local"], IndexedAccessType: s[34], InferredPredicate: s[1], InferTypeAnnotation: s[35], InterfaceDeclaration: s[22], InterfaceExtends: s[12], InterfaceTypeAnnotation: ["extends", "body"], InterpreterDirective: s[1], IntersectionTypeAnnotation: s[36], JsExpressionRoot: s[37], JsonRoot: s[37], JSXAttribute: ["name", "value"], JSXClosingElement: ["name"], JSXClosingFragment: s[1], JSXElement: ["openingElement", "children", "closingElement"], JSXEmptyExpression: s[1], JSXExpressionContainer: s[3], JSXFragment: ["openingFragment", "children", "closingFragment"], JSXIdentifier: s[1], JSXMemberExpression: s[38], JSXNamespacedName: ["namespace", "name"], JSXOpeningElement: ["name", "typeArguments", "attributes"], JSXOpeningFragment: s[1], JSXSpreadAttribute: s[6], JSXSpreadChild: s[3], JSXText: s[1], KeyofTypeAnnotation: s[6], LabeledStatement: ["label", "body"], Literal: s[1], LogicalExpression: s[5], MatchArrayPattern: ["elements", "rest"], MatchAsPattern: ["pattern", "target"], MatchBindingPattern: s[21], MatchExpression: s[39], MatchExpressionCase: s[40], MatchIdentifierPattern: s[21], MatchLiteralPattern: s[41], MatchMemberPattern: ["base", "property"], MatchObjectPattern: ["properties", "rest"], MatchObjectPatternProperty: ["key", "pattern"], MatchOrPattern: ["patterns"], MatchRestPattern: s[6], MatchStatement: s[39], MatchStatementCase: s[40], MatchUnaryPattern: s[6], MatchWildcardPattern: s[1], MemberExpression: s[38], MetaProperty: ["meta", "property"], MethodDefinition: s[42], MixedTypeAnnotation: s[1], ModuleExpression: s[10], NeverTypeAnnotation: s[1], NewExpression: s[9], NGChainedExpression: s[43], NGEmptyExpression: s[1], NGMicrosyntax: s[10], NGMicrosyntaxAs: ["key", "alias"], NGMicrosyntaxExpression: ["expression", "alias"], NGMicrosyntaxKey: s[1], NGMicrosyntaxKeyedExpression: ["key", "expression"], NGMicrosyntaxLet: s[32], NGPipeExpression: ["left", "right", "arguments"], NGRoot: s[37], NullableTypeAnnotation: s[23], NullLiteral: s[1], NullLiteralTypeAnnotation: s[1], NumberLiteralTypeAnnotation: s[1], NumberTypeAnnotation: s[1], NumericLiteral: s[1], ObjectExpression: ["properties"], ObjectMethod: s[13], ObjectPattern: ["decorators", "properties", "typeAnnotation"], ObjectProperty: s[42], ObjectTypeAnnotation: ["properties", "indexers", "callProperties", "internalSlots"], ObjectTypeCallProperty: s[18], ObjectTypeIndexer: ["variance", "id", "key", "value"], ObjectTypeInternalSlot: ["id", "value"], ObjectTypeMappedTypeProperty: ["keyTparam", "propType", "sourceType", "variance"], ObjectTypeProperty: ["key", "value", "variance"], ObjectTypeSpreadProperty: s[6], OpaqueType: ["id", "typeParameters", "supertype", "impltype", "lowerBound", "upperBound"], OptionalCallExpression: s[9], OptionalIndexedAccessType: s[34], OptionalMemberExpression: s[38], ParenthesizedExpression: s[3], PipelineBareFunction: ["callee"], PipelinePrimaryTopicReference: s[1], PipelineTopicExpression: s[3], Placeholder: s[1], PrivateIdentifier: s[1], PrivateName: s[21], Program: s[7], Property: s[32], PropertyDefinition: s[14], QualifiedTypeIdentifier: s[44], QualifiedTypeofIdentifier: s[44], RegExpLiteral: s[1], RestElement: ["argument", "typeAnnotation", "decorators"], ReturnStatement: s[6], SatisfiesExpression: s[4], SequenceExpression: s[43], SpreadElement: s[6], StaticBlock: s[10], StringLiteral: s[1], StringLiteralTypeAnnotation: s[1], StringTypeAnnotation: s[1], Super: s[1], SwitchCase: ["test", "consequent"], SwitchStatement: ["discriminant", "cases"], SymbolTypeAnnotation: s[1], TaggedTemplateExpression: ["tag", "typeArguments", "quasi"], TemplateElement: s[1], TemplateLiteral: ["quasis", "expressions"], ThisExpression: s[1], ThisTypeAnnotation: s[1], ThrowStatement: s[6], TopicReference: s[1], TryStatement: ["block", "handler", "finalizer"], TSAbstractAccessorProperty: s[45], TSAbstractKeyword: s[1], TSAbstractMethodDefinition: s[32], TSAbstractPropertyDefinition: s[45], TSAnyKeyword: s[1], TSArrayType: s[2], TSAsExpression: s[4], TSAsyncKeyword: s[1], TSBigIntKeyword: s[1], TSBooleanKeyword: s[1], TSCallSignatureDeclaration: s[46], TSClassImplements: s[47], TSConditionalType: s[17], TSConstructorType: s[46], TSConstructSignatureDeclaration: s[46], TSDeclareFunction: s[31], TSDeclareKeyword: s[1], TSDeclareMethod: ["decorators", "key", "typeParameters", "params", "returnType"], TSEmptyBodyFunctionExpression: ["id", "typeParameters", "params", "returnType"], TSEnumBody: s[26], TSEnumDeclaration: s[19], TSEnumMember: ["id", "initializer"], TSExportAssignment: s[3], TSExportKeyword: s[1], TSExternalModuleReference: s[3], TSFunctionType: s[46], TSImportEqualsDeclaration: ["id", "moduleReference"], TSImportType: ["options", "qualifier", "typeArguments", "source"], TSIndexedAccessType: s[34], TSIndexSignature: ["parameters", "typeAnnotation"], TSInferType: s[35], TSInstantiationExpression: s[47], TSInterfaceBody: s[10], TSInterfaceDeclaration: s[22], TSInterfaceHeritage: s[47], TSIntersectionType: s[36], TSIntrinsicKeyword: s[1], TSJSDocAllType: s[1], TSJSDocNonNullableType: s[23], TSJSDocNullableType: s[23], TSJSDocUnknownType: s[1], TSLiteralType: s[41], TSMappedType: ["key", "constraint", "nameType", "typeAnnotation"], TSMethodSignature: ["key", "typeParameters", "params", "returnType"], TSModuleBlock: s[10], TSModuleDeclaration: s[19], TSNamedTupleMember: ["label", "elementType"], TSNamespaceExportDeclaration: s[21], TSNeverKeyword: s[1], TSNonNullExpression: s[3], TSNullKeyword: s[1], TSNumberKeyword: s[1], TSObjectKeyword: s[1], TSOptionalType: s[23], TSParameterProperty: ["parameter", "decorators"], TSParenthesizedType: s[23], TSPrivateKeyword: s[1], TSPropertySignature: ["key", "typeAnnotation"], TSProtectedKeyword: s[1], TSPublicKeyword: s[1], TSQualifiedName: s[5], TSReadonlyKeyword: s[1], TSRestType: s[23], TSSatisfiesExpression: s[4], TSStaticKeyword: s[1], TSStringKeyword: s[1], TSSymbolKeyword: s[1], TSTemplateLiteralType: ["quasis", "types"], TSThisType: s[1], TSTupleType: ["elementTypes"], TSTypeAliasDeclaration: ["id", "typeParameters", "typeAnnotation"], TSTypeAnnotation: s[23], TSTypeAssertion: s[4], TSTypeLiteral: s[26], TSTypeOperator: s[23], TSTypeParameter: ["name", "constraint", "default"], TSTypeParameterDeclaration: s[48], TSTypeParameterInstantiation: s[48], TSTypePredicate: s[49], TSTypeQuery: ["exprName", "typeArguments"], TSTypeReference: ["typeName", "typeArguments"], TSUndefinedKeyword: s[1], TSUnionType: s[36], TSUnknownKeyword: s[1], TSVoidKeyword: s[1], TupleTypeAnnotation: ["types", "elementTypes"], TupleTypeLabeledElement: ["label", "elementType", "variance"], TupleTypeSpreadElement: ["label", "typeAnnotation"], TypeAlias: s[24], TypeAnnotation: s[23], TypeCastExpression: s[4], TypeofTypeAnnotation: ["argument", "typeArguments"], TypeOperator: s[23], TypeParameter: ["bound", "default", "variance"], TypeParameterDeclaration: s[48], TypeParameterInstantiation: s[48], TypePredicate: s[49], UnaryExpression: s[6], UndefinedTypeAnnotation: s[1], UnionTypeAnnotation: s[36], UnknownTypeAnnotation: s[1], UpdateExpression: s[6], V8IntrinsicIdentifier: s[1], VariableDeclaration: ["declarations"], VariableDeclarator: s[27], Variance: s[1], VoidPattern: s[1], VoidTypeAnnotation: s[1], WhileStatement: s[25], WithStatement: ["object", "body"], YieldExpression: s[6] }, qu = s2(d2), a2 = qu;
    g2 = ve, Xi = le(["RegExpLiteral", "BigIntLiteral", "NumericLiteral", "StringLiteral", "DirectiveLiteral", "Literal", "JSXText", "TemplateElement", "StringLiteralTypeAnnotation", "NumberLiteralTypeAnnotation", "BigIntLiteralTypeAnnotation"]);
    r2 = Pu, Ou = /\*\/$/, Vu = /^\/\*\*?/, Ru = /^\s*(\/\*\*?(.|\r?\n)*?\*\/)/, Mu = /(^|\s+)\/\/([^\n\r]*)/g, h2 = /^(\r?\n)+/, vu = /(?:^|\r?\n) *(@[^\n\r]*?) *\r?\n *(?![^\n\r@]*\/\/[^]*)([^\s@][^\n\r@]+?) *\r?\n/g, k2 = /(?:^|\r?\n) *@(\S+) *([^\n\r]*)/g, Uu = /(\r?\n|^) *\* ?/g, Ju = [];
    T2 = ["noformat", "noprettier"], b2 = ["format", "prettier"];
    D2 = _u;
    S2 = Xu, B2 = "module", F2 = "commonjs", N2 = [B2, F2];
    ju = { next: !0, ranges: !0, webcompat: !0, loc: !1, raw: !0, directives: !0, impliedStrict: !1, preserveParens: !0, lexical: !1, jsx: !0, validateRegex: !1 };
    $u = S2(Ku);
  }
});
init_meriyah();
export {
  I2 as default,
  at as parsers
};
